---
title: "Daria HNSCC Analysis"
output:
  html_document:
    toc: yes
    toc_float: yes
---

```{r}

library(dplyr)
library(ggplot2)
library(plotly)
library(DT)
library(writexl)

```


# Establishing Target Genes for Tumor Samples

## Reading in Gene Metric and Group Data

```{r}

gene_metric_mean_file <- "/media/theron/One_Touch/head_and_neck_DARIA/data/splicemutr_05_26_2021/GENE_METRIC_01032022/HNSCC_gene_metric_mean.rds"
gene_metric_mean <- readRDS(gene_metric_mean_file)
groups_file <- "/media/theron/One_Touch/head_and_neck_DARIA/data/splicemutr_05_26_2021/GENE_METRIC_01032022/groups_file.txt"
groups <- read.table(groups_file)
colnames(groups) <- c("sample","group")

```

## Gene Metric Distribution for Tumor Samples

```{r}

tumor_samples <- groups %>% dplyr::filter(group == "Tumor")
tumor_gene_metric <- gene_metric_mean[,tumor_samples$sample]
tumor_gene_metric_mean <- apply(tumor_gene_metric,1,mean)
tumor_gene_metric_mean <- as.data.frame(tumor_gene_metric_mean)
colnames(tumor_gene_metric_mean)<-"gene_metric"

tumor_gene_metric_save <- tumor_gene_metric
tumor_gene_metric_save$genes <- rownames(tumor_gene_metric)
tumor_gene_metric_mean_save <- tumor_gene_metric_mean
tumor_gene_metric_mean_save$genes <- rownames(tumor_gene_metric_mean) 
write_xlsx(tumor_gene_metric_save[order(tumor_gene_metric_save$`DGay4-37924`,decreasing=T),],
            path="/media/theron/One_Touch/head_and_neck_DARIA/data/splicemutr_05_26_2021/GENE_METRIC_01032022/tumor_gene_metric_all.xlsx")
write_xlsx(tumor_gene_metric_mean_save[order(tumor_gene_metric_mean_save$gene_metric,decreasing=T),,drop = FALSE],
            path="/media/theron/One_Touch/head_and_neck_DARIA/data/splicemutr_05_26_2021/GENE_METRIC_01032022/tumor_gene_metric_mean.xlsx")

ggplot(tumor_gene_metric_mean,aes(x=gene_metric))+
  geom_histogram()+xlab("Gene Metric")+labs(title="Tumor Samples")

```

## Gene metric distribution for 090 Cell Line

```{r}

gene_metric_090 <- gene_metric_mean[,"DGay21-090-1_1_val",drop = FALSE]
colnames(gene_metric_090)<-"gene_metric"
ggplot(gene_metric_090,aes(x=gene_metric_090$gene_metric))+
  geom_histogram()+xlab("Gene Metric")+labs(title="090 Cell Line")

```

## Gene metric distribution for 047 Cell Line

```{r}

gene_metric_047 <- gene_metric_mean[,"DGay18-047-1_1_val",drop = FALSE]
colnames(gene_metric_047)<-"gene_metric"
ggplot(gene_metric_047,aes(x=gene_metric_047$gene_metric))+
  geom_histogram()+xlab("Gene Metric")+labs(title="047 Cell Line")

```

## Target Genes Across Samples

```{r}

genes <- rownames(gene_metric_mean)
gene_metric_all <- rbind(tumor_gene_metric_mean,gene_metric_090,gene_metric_047)
gene_metric_all <- cbind(gene_metric_all,
                         c(rep("tumor",nrow(gene_metric_mean)),
                           rep("090",nrow(gene_metric_090)),
                           rep("047",nrow(gene_metric_047))),
                         c(rep(genes,3)))
colnames(gene_metric_all) <- c("gene_metric","sample","gene")

gene_metric_nonlinear <- cbind(tumor_gene_metric_mean,gene_metric_090,gene_metric_047)
colnames(gene_metric_nonlinear) <- c("tumor","090","047")
gene_metric_nonlinear$sdev <- apply(gene_metric_nonlinear[,seq(3)],1,sd)
gene_metric_nonlinear$xlab <- seq(nrow(gene_metric_nonlinear))

gene_metric_nonlinear <- gene_metric_nonlinear %>% dplyr::filter(`090` > 0 & `047` > 0 & tumor > 0)
ggplot(gene_metric_nonlinear,aes(y=log10(`090`+1),x=log10(tumor+1)))+geom_point()
ggplot(gene_metric_nonlinear,aes(y=`090`,x=tumor))+geom_point()

ggplot(gene_metric_nonlinear,aes(y=log10(`047`+1),x=log10(tumor+1)))+geom_point()
ggplot(gene_metric_nonlinear,aes(y=`047`,x=tumor))+geom_point()

```
# Number of Annotated Transcripts found with binders by splicemutr that are formed by transcript formation using stringtie2

## Extracting only reconstructed annotated transcripts

```{bash, eval=F}

grep ENST merged_transcripts.gtf | awk '{print $12}' | tr -d '""' | tr -d ';' > recon_trans.txt

```

## Looking for those splicemutr formed genes/transcripts that have reconstructed transcripts

```{r}

splicemutr_dat_file <- "/media/theron/One_Touch/head_and_neck_DARIA/data/splicemutr_05_26_2021/GENE_METRIC_01032022/full_splicemutr_dat.rds"
splicemutr_dat <- readRDS(splicemutr_dat_file)
splicemutr_dat_annotated <- splicemutr_dat %>% dplyr::filter(verdict == "annotated" & modified == "normal")
splicemutr_dat_unannotated <- splicemutr_dat %>% dplyr::filter(modified == "changed")
annotated_tx <- unique(splicemutr_dat_annotated$tx_id)
unannotated_tx <- unique(splicemutr_dat_unannotated$tx_id)
recon_trans_file <- "/media/theron/One_Touch/head_and_neck_DARIA/data/splicemutr_05_26_2021/GENE_METRIC_01032022/recon_trans.txt"
recon_trans <- read.table(recon_trans_file,sep="\t")
recon_trans <- unique(recon_trans$V1)

ann_length <- length(annotated_tx)
unann_length <- length(unannotated_tx)
reconstructed_transcripts <- data.frame(ann_length,
                                        length(which(annotated_tx %in% recon_trans))/ann_length*100,
                                        unann_length,
                                        length(which(unannotated_tx %in% recon_trans))/unann_length*100)

colnames(reconstructed_transcripts) <- c("annotated transcripts","percent ann reconstructed","unannotated transcripts","percent unann represented")
datatable(reconstructed_transcripts)

```


## Looking at top 10% ranked genes in tumor samples

```{r}

gene_metric_nonlinear <- gene_metric_nonlinear[order(gene_metric_nonlinear$tumor,decreasing=T),]
gene_metric_nonlin_top_1_perc <- gene_metric_nonlinear[seq(round(nrow(gene_metric_nonlinear)*0.01)),]

ggplot(gene_metric_nonlin_top_1_perc,aes(y=log10(`090`+1),x=log10(tumor+1)))+geom_point()
ggplot(gene_metric_nonlin_top_1_perc,aes(y=log10(`047`+1),x=log10(tumor+1)))+geom_point()

genes_top_1_perc <- rownames(gene_metric_nonlin_top_1_perc)

```

# Looking at splicemutr data for tope 1% ranked genes

```{r}

splicemutr_dat_file <- "/media/theron/One_Touch/head_and_neck_DARIA/data/splicemutr_05_26_2021/GENE_METRIC_01032022/full_splicemutr_dat.rds"
splicemutr_dat <- readRDS(splicemutr_dat_file)
splicemutr_dat_top_1_perc <- splicemutr_dat %>% dplyr::filter(gene %in% genes_top_1_perc)
table(splicemutr_dat_top_1_perc$verdict)
table(splicemutr_dat_top_1_perc$error)
table(splicemutr_dat_top_1_perc$modified)

splicemutr_dat_annotated <- splicemutr_dat_top_1_perc %>% dplyr::filter(verdict == "annotated" & modified == "normal")
splicemutr_dat_unannotated <- splicemutr_dat_top_1_perc %>% dplyr::filter(modified == "changed")
annotated_tx <- unique(splicemutr_dat_annotated$tx_id)
unannotated_tx <- unique(splicemutr_dat_unannotated$tx_id)
recon_trans_file <- "/media/theron/One_Touch/head_and_neck_DARIA/data/splicemutr_05_26_2021/GENE_METRIC_01032022/recon_trans.txt"
recon_trans <- read.table(recon_trans_file,sep="\t")
recon_trans <- unique(recon_trans$V1)

ann_length <- length(annotated_tx)
unann_length <- length(unannotated_tx)
reconstructed_transcripts <- data.frame(ann_length,
                                        length(which(annotated_tx %in% recon_trans))/ann_length*100,
                                        unann_length,
                                        length(which(unannotated_tx %in% recon_trans))/unann_length*100)

colnames(reconstructed_transcripts) <- c("annotated transcripts","percent ann reconstructed","unannotated transcripts","percent unann represented")
datatable(reconstructed_transcripts)

```




