---
title: "Valsamo_analysis"
author: "Theron Palmer"
date: "8/26/2021"
output:
  html_document:
    toc: yes
    toc_float: yes
---


# Load in the necessary libraries

```{r}

library(readxl)
library(ggplot2)
library(DT)
library(dplyr)
library(plyr)
library(stringr)
library(pheatmap)
library(rjson)

```

# Exploring the sample metadata via the study manifest

## finding samples that did not get aligned on JHPCE for some reason

```{r}

manifest <- read_excel("/media/theron/My_Passport/Valsamo/manifest.xlsx")
files_to_remove <- c("hg19MTERCC-ensembl75-genes-Q21777-Plate-1-E06_L65",
"hg19MTERCC-ensembl75-genes-Q21777-Plate-1-F12_L1.D707_508",
"hg19MTERCC-ensembl75-genes-Q23152+B4+H2+AG710464_L1.D705")
manifest <- manifest %>% dplyr::filter(!(Sample %in% files_to_remove))
manifest$Sample <- str_replace_all(manifest$Sample,"hg19MTERCC-ensembl75-genes-","")
fastq_files <- read.table("/media/theron/My_Passport/Valsamo/fastq_files.txt")
SJ_files <- read.table("/media/theron/My_Passport/Valsamo/SJ_files.txt")
SJ_files$sample_name <- vapply(SJ_files$V1,function(file){
  str_remove(file,"SJ.out.tab")
},character(1))
fastq_files$sample_name <- vapply(fastq_files$V1,function(file){
  str_remove(file,"_1.clipped.fastq.gz")
},character(1))

```


```{r}

missing_files <- data.frame(sprintf("/dcs04/fertig/data/theron/share/%s",fastq_files[which(!(fastq_files$sample_name %in% SJ_files$sample_name)),"V1"]))
write.table(missing_files,file="/media/theron/My_Passport/Valsamo/missing_fasta.txt",
            quote=F, col.names = F, row.names = F, sep = "\t")

```

## Looking at response rates across treatment group

RESIST 1.1 Progression Terms form: https://project.eortc.org/recist/wp-content/uploads/sites/4/2015/03/RECISTGuidelines.pdf

CR = Complete Response  
PD = Progressive Disease  
PR = Partial Response  
SD = Stable Disease  
NE = ?  

```{r}

ggplot(manifest,aes(x=TRTGRP,y=log2(PFS),fill=TRTGRP))+geom_violin()+geom_point(position = position_jitter(seed = 1, width = 0.2))
ggplot(manifest,aes(x=TRTGRP,y=log2(OS),fill=TRTGRP))+geom_violin()+geom_point(position = position_jitter(seed = 1, width = 0.2))
ggplot(manifest,aes(x=TRTGRP,fill=AX_BOR))+geom_bar(position='dodge')
ggplot(manifest,aes(x=TRTGRP,fill=AX_BOR3))+geom_bar(position='dodge')

```

## Leafcutter Analysis Preparation

Comparisons:  
a) NIV3.NAIVE(PR) vs NIV3.NAIVE(PD)  
b) NIV3.NAIVE(CR) vs NIV3.NAIVE(PD)  
c) NIV3.NAIVE(SD) vs NIV3.NAIVE(PD)  
d) NIV3.PROG(PR) vs NIV3.PROG(PD)  
e) NIV3.PROG(CR) vs NIV3.PROG(PD)  
f) NIV3.PROG(SD) vs NIV3.PROG(PD)  
g) NIV1.IPI3(PR) vs NIV1.IPI3(PD)  
h) NIV1.IPI3(SD) vs NIV3.PROG(PD)  
i) NIV1.IPI3(PR) vs NIV3.PROG(PD)  
j) NIV1.IPI3(SD) vs NIV3.PROG(PD)  
k) NIV1.IPI3(SD) vs NIV3.NAIVE(PD)  
l) NIV1.IPI3(PR) vs NIV3.NAIVE(PD)  
m) NIV1.IPI3(SD) vs NIV3.NAIVE(PD)  
n) NIV3.PROG(CR) vs NIV3.PROG(SD)  
o) NIV3.PROG(PR) vs NIV3.PROG(SD)  
p) NIV3.PROG(CR) vs NIV3.PROG(PR)  
q) NIV3.NAIVE(CR) vs NIV3.NAIVE(SD)  
r) NIV3.NAIVE(PR) vs NIV3.NAIVE(SD)  
s) NIV3.NAIVE(CR) vs NIV3.NAIVE(PR)  

# Forming groups_file.txt and junc_file.txt for each comparison

```{r}

NIV3_NAIVE <- manifest[which(manifest$TRTGRP == "NIV3-NAIVE"),]
NIV3_PROG <- manifest[which(manifest$TRTGRP == "NIV3-PROG"),]
NIV1_IPI3 <- manifest[which(manifest$TRTGRP %in% c("NIV1+IPI3 P2","NIV1+IPI3 P3")),]
groups_and_junc_dir <- "/media/theron/My_Passport/Valsamo/leafcutter_prep"
JHPCE_dir <- "/dcs04/fertig/data/theron/share/juncs"
comparisons <- list()

```


## NIV3.NAIVE(PR) vs NIV3.NAIVE(PD) 

```{r}

tar<-"PR"
comp<-"PD"
tar_samp<-"NIV3_NAIVE"
comp_samp<-"NIV3_NAIVE"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == tar]
comparators<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a


for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV3.NAIVE(CR) vs NIV3.NAIVE(PD) 

```{r}

tar<-"CR"
comp<-"PD"
tar_samp<-"NIV3_NAIVE"
comp_samp<-"NIV3_NAIVE"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == tar]
comparators<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV3.NAIVE(SD) vs NIV3.NAIVE(PD) 

```{r}

tar<-"SD"
comp<-"PD"
tar_samp<-"NIV3_NAIVE"
comp_samp<-"NIV3_NAIVE"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == tar]
comparators<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV3.PROG(PR) vs NIV3.PROG(PD)  

```{r}

tar<-"PR"
comp<-"PD"
tar_samp <- "NIV3_PROG"
comp_samp<-"NIV3_PROG"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == tar]
comparators<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV3.PROG(CR) vs NIV3.PROG(PD)  

```{r}

tar<-"CR"
comp<-"PD"
tar_samp <- "NIV3_PROG"
comp_samp<-"NIV3_PROG"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == tar]
comparators<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV3.PROG(SD) vs NIV3.PROG(PD)  

```{r}

tar<-"SD"
comp<-"PD"
tar_samp <- "NIV3_PROG"
comp_samp<-"NIV3_PROG"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == tar]
comparators<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV1.IPI3(PR) vs NIV1.IPI3(PD)  

```{r}

tar<-"PR"
comp<-"PD"
tar_samp <- "NIV1_IPI3"
comp_samp<-"NIV1_IPI3"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV1_IPI3$Sample[NIV1_IPI3$AX_BOR == tar]
comparators<-NIV1_IPI3$Sample[NIV1_IPI3$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV1.IPI3(SD) vs NIV3.PROG(PD)  

```{r}

tar<-"SD"
comp<-"PD"
tar_samp <- "NIV1_IPI3"
comp_samp<-"NIV3_PROG"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV1_IPI3$Sample[NIV1_IPI3$AX_BOR == tar]
comparators<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV1.IPI3(PR) vs NIV3.PROG(PD)  

```{r}

tar<-"PR"
comp<-"PD"
tar_samp <- "NIV1_IPI3"
comp_samp<-"NIV3_PROG"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV1_IPI3$Sample[NIV1_IPI3$AX_BOR == tar]
comparators<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV1.IPI3(SD) vs NIV3.PROG(PD)  

```{r}

tar<-"SD"
comp<-"PD"
tar_samp <- "NIV1_IPI3"
comp_samp<-"NIV3_PROG"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV1_IPI3$Sample[NIV1_IPI3$AX_BOR == tar]
comparators<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV1.IPI3(SD) vs NIV3.NAIVE(PD)  

```{r}

tar<-"SD"
comp<-"PD"
tar_samp <- "NIV1_IPI3"
comp_samp<-"NIV3_NAIVE"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV1_IPI3$Sample[NIV1_IPI3$AX_BOR == tar]
comparators<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV1.IPI3(PR) vs NIV3.NAIVE(PD)  

```{r}

tar<-"PR"
comp<-"PD"
tar_samp <- "NIV1_IPI3"
comp_samp<-"NIV3_NAIVE"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV1_IPI3$Sample[NIV1_IPI3$AX_BOR == tar]
comparators<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV1.IPI3(SD) vs NIV3.NAIVE(PD)  

```{r}

tar<-"SD"
comp<-"PD"
tar_samp <- "NIV1_IPI3"
comp_samp<-"NIV3_NAIVE"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV1_IPI3$Sample[NIV1_IPI3$AX_BOR == tar]
comparators<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == comp]

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV3.PROG(CR) vs NIV3.PROG(SD)  

```{r}

tar<-"CR"
comp<-"SD"
tar_samp <- "NIV3_PROG"
comp_samp<-"NIV3_PROG"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == tar]
comparators<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```

## NIV3.PROG(PR) vs NIV3.PROG(SD)  

```{r}

tar<-"PR"
comp<-"SD"
tar_samp <- "NIV3_PROG"
comp_samp<-"NIV3_PROG"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == tar]
comparators<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```


## NIV3.PROG(CR) vs NIV3.PROG(PR)  

```{r}

tar<-"CR"
comp<-"PR"
tar_samp <- "NIV3_PROG"
comp_samp<-"NIV3_PROG"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == tar]
comparators<-NIV3_PROG$Sample[NIV3_PROG$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```


## NIV3.NAIVE(CR) vs NIV3.NAIVE(SD)  

```{r}

tar<-"CR"
comp<-"SD"
tar_samp <- "NIV3_NAIVE"
comp_samp<-"NIV3_NAIVE"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == tar]
comparators<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```


## NIV3.NAIVE(PR) vs NIV3.NAIVE(SD)  

```{r}

tar<-"PR"
comp<-"SD"
tar_samp <- "NIV3_NAIVE"
comp_samp<-"NIV3_NAIVE"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == tar]
comparators<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

```


## NIV3.NAIVE(CR) vs NIV3.NAIVE(PR)  

```{r}
  
tar<-"CR"
comp<-"PR"
tar_samp <- "NIV3_NAIVE"
comp_samp<-"NIV3_NAIVE"
comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
targets<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == tar]
comparators<-NIV3_NAIVE$Sample[NIV3_NAIVE$AX_BOR == comp]
a<-list()
a[["targets"]] <- targets
a[["comparators"]] <- comparators
comparisons[[comparison]] <- a

for (target in targets){
  target_junc_file <- sprintf("%s/%s_juncs_%s.txt",groups_and_junc_dir,target,comparison)
  file_contents <- sprintf("%s.filt.junc",c(target,comparators))
  file_contents <- data.frame(file_contents)
  write.table(file_contents,target_junc_file,
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
}

saveRDS(comparisons,file="/media/theron/My_Passport/Valsamo/analysis/leafcutterMD/comparisons.rds")

```

## Prepping comparisons for splicemutr transcript formation

```{r}

comparison_junctions <- list()
leafcutter_files <- read.table("/media/theron/My_Passport/Valsamo/analysis/leafcutterMD/filenames.txt")
for (i in seq(nrow(leafcutter_files))){
  print(i)
  outlier_file <- leafcutter_files[i,]
  outlier_juncs <- read.table(outlier_file)
  outlier_junc_cols <- str_replace_all(colnames(outlier_juncs),".filt","")
  colnames(outlier_juncs) <- str_replace_all(outlier_junc_cols,"[-_+.]","_")
  file_split <- strsplit(outlier_file,"_juncs_")[[1]]
  sample <- basename(file_split[1])
  sample <- substr(sample,1,nchar(sample))
  sample<-str_replace_all(sample,"[-_+.]","_")
  arm_info <- strsplit(file_split[2],"_")[[1]]
  tar_samp <- sprintf("%s_%s",arm_info[1],arm_info[2])
  comp_samp <- sprintf("%s_%s",arm_info[3],arm_info[4])
  tar<-arm_info[5]
  comp<-arm_info[6]
  comparison <- sprintf("%s_%s_%s_%s",tar_samp,comp_samp,tar,comp)
  sig_juncs <- rownames(outlier_juncs)[as.numeric(outlier_juncs[,sample]) <=0.05]
  comparison_junctions[[comparison]] <- unique(c(comparison_junctions[[comparison]],sig_juncs))
}

```

```{r,eval=F}

comparison_juncs_linear <- data.frame(t(vapply(unname(unlist(comparison_junctions)),function(junc){
  junc_vals<-strsplit(junc,":")[[1]]
  strand<-strsplit(junc_vals[4],"_")[[1]][3]
  c(junc_vals[1],junc_vals[2],junc_vals[3],strand)
},character(4))))
rownames(comparison_juncs_linear)<-seq(nrow(comparison_juncs_linear))
colnames(comparison_juncs_linear)<-c("chr","start","end","strand")

iter<-1
total_juncs <- nrow(comparison_juncs_linear)
for (i in seq(1,total_juncs,10000)){
  if (i+9999 >= total_juncs){
    end = total_juncs
  } else {
    end = i+9999
  }
  comparison_juncs_small <- comparison_juncs_linear[seq(i,end),]
  saveRDS(comparison_juncs_small,
          sprintf("/media/theron/My_Passport/Valsamo/analysis/leafcutterMD/comparison_juncs_linear_%d.rds",iter))
  iter<-iter+1
}


saveRDS(comparison_juncs_linear,file="/media/theron/My_Passport/Valsamo/analysis/leafcutterMD/comparison_juncs_linear.rds")
saveRDS(comparison_junctions,file="/media/theron/My_Passport/Valsamo/analysis/leafcutterMD/comparison_juncs.rds")


```

## Looking at junction overlap between comparison groups

```{r}

comparison_junctions <- readRDS("/media/theron/My_Passport/Valsamo/analysis/leafcutterMD/comparison_juncs.rds")

for (i in seq(length(comparison_junctions))){
    comparison_junctions[[i]]<-vapply(comparison_junctions[[i]],function(junc){
      junc_vals<-strsplit(junc,":")[[1]]
      strand<-strsplit(junc_vals[4],"_")[[1]][3]
      sprintf("%s:%s:%s:%s",junc_vals[1],junc_vals[2],junc_vals[3],strand)
    },character(1))
}

for (i in seq(length(comparison_junctions))){
  comparison_junctions[[i]]<-unique(comparison_junctions[[i]])
}

```

```{r}

comp_juncs <- lapply(seq(length(comparison_junctions)),function(val){
  current<-comparison_junctions[[val]]
  comparisons<-vapply(seq(length(comparison_junctions)),function(comp_val){
    length(which(comparison_junctions[[comp_val]] %in% current))/length(comparison_junctions[[comp_val]])
  },as.numeric(1))
})
comp_juncs<-as.data.frame(matrix(unlist(comp_juncs),byrow=T,nrow=length(comp_juncs)))
colnames(comp_juncs)<-names(comparison_junctions)
rownames(comp_juncs)<-names(comparison_junctions)

pheatmap::pheatmap(comp_juncs,cluster_cols=F,cluster_rows = F)
pheatmap::pheatmap(comp_juncs,cluster_cols=F,cluster_rows = T)

```

# Compiling the genotype information per sample

```{r}

genotype_files <- read.table("/media/theron/My_Passport/Valsamo/genotypes/filenames.txt")
genotypes<-lapply(seq(nrow(genotype_files)),function(row_val){
  spec_genotype_dir <- genotype_files[row_val,]
  file <- sprintf("%s/%s",spec_genotype_dir,list.files(spec_genotype_dir,"*genotype.json"))
  spec_genotype_json <- fromJSON(file=file)
  spec_genotype_json <- unname(unlist(spec_genotype_json))
  vapply(spec_genotype_json,function(allele){
    allele<-str_split(allele,":")[[1]]
    allele<-paste(allele[seq(2)],collapse=":")
    sprintf("HLA-%s",str_replace(allele,"[*]",""))
  },character(1))
})
names(genotypes) <- vapply(seq(nrow(genotype_files)),function(row_val){
  spec_genotype_dir <- genotype_files[row_val,]
  sample <- str_replace(basename(spec_genotype_dir),"Aligned.out.bam.sorted_dir","")
},character(1))

unique_alleles <- unique(unname(unlist(genotypes)))
class <- str_detect(unique_alleles,"HLA-A") | str_detect(unique_alleles,"HLA-B") | str_detect(unique_alleles,"HLA-C")
class_1_alleles <- data.frame(unique_alleles[class])
class_2_alleles <- data.frame(unique_alleles[!class])

write.table(class_1_alleles,
            file="/media/theron/My_Passport/Valsamo/genotypes/class_1_alleles.txt",
            quote=F, col.names = F, row.names = F, sep = "\t")
write.table(class_2_alleles,
            file="/media/theron/My_Passport/Valsamo/genotypes/class_2_alleles.txt",
            quote=F, col.names = F, row.names = F, sep = "\t")
saveRDS(genotypes,file="/media/theron/My_Passport/Valsamo/genotypes/genotypes.rds")

```

# Creating specific junction files per comparison with respect to data_splicemutr

```{r}

comparison_junctions <- readRDS("/media/theron/My_Passport/Valsamo/analysis/leafcutterMD/comparison_juncs.rds")
data_splicemutr <- read.table("/media/theron/My_Passport/Valsamo/analysis/splicemutr_output/data_splicemutr.txt",header=T)
data_splicemutr$juncs <- sprintf("%s:%s:%s:%s",data_splicemutr$chr,data_splicemutr$start,data_splicemutr$end,data_splicemutr$strand)
data_splicemutr$orig_row <- seq(nrow(data_splicemutr))-1
genotypes <- readRDS(file="/media/theron/My_Passport/Valsamo/genotypes/genotypes.rds")

```

## NIV3_PROG_NIV3_PROG_SD_PD

```{r}

comparisons <- names(comparison_junctions)
for (comparison in comparisons){
  output_loc <- "/media/theron/My_Passport/Valsamo/analysis/splicemutr_output"
  juncs_str <- comparison_junctions[[comparison]]
  juncs <- data.frame(matrix(unlist(strsplit(juncs_str,"[:]")),
                                            byrow=T,
                                            nrow=length(juncs_str)))
  colnames(juncs) <- c("chr","start","end","cluster")
  juncs$strand <- vapply(juncs$cluster,function(clu){strsplit(clu,"_")[[1]][3]},character(1))
  juncs$juncs <- sprintf("%s:%s:%s:%s",juncs$chr,juncs$start,juncs$end,juncs$strand)
  data_splicemutr_specific <- data_splicemutr[which(data_splicemutr$juncs %in% juncs$juncs),]
  
  specific_splice_file <- sprintf("%s/%s_data_splicemutr.rds",output_loc,comparison)
  
  saveRDS(data_splicemutr_specific,file=specific_splice_file)
}

```

