# SpliceMutr: Calculating splicing-derived neoantigen burden from splice-junctions affecting protein-coding transcripts in RNA-seq data.

------------------------------------------------------------------------

SpliceMutr is a tool used for the analysis of splicing-derived antigen burden. SpliceMutr uses differential intron usage analysis through LeafCutter or LeafCutterMD to first determine differential usage of splice-junctions between sets of RNA-seq samples. LeafCutter uses a Dirichlet-Multinomial generalized linear model to fit splice-junction counts within clusters of splice-junctions grouped due to overlapping splice site usage between pairs of sample types (tumor and normal or pre and post treatment in the case of the original SpliceMutr paper). Two separate Dirichlet-Multinomial models are fit to the splice-junction counts per splice-junction cluster based on the sample type stratification and a likelihood ratio test is used to evaluate differential usage within the splice-junction cluster as a whole. Each individual differentially-used splice junction per cluster is then used to modify the reference transcriptome. The modified transcripts are searched for canonical and/or modified open reading frames, translated based on the best identified open reading frame, and kmerized into 9-mers. During kmerization, a map between the peptide kmer and modified protein of interest is generated for searching after MHC-peptide binding prediction. MHC-peptide binding prediction is carried out using the arcasHLA generated HLA genotype per sample analyzed, the set of generated peptide 9-mers, and MHCnuggets or NetMHCpan. Peptide 9-mers found to bind to any of the HLA molecules per sample are then mapped back to their transcript of origin using the kmer-transcript map and peptide binding summaries are generated that contain the number of unique binding kmers per transcript and per sample. If differential intron usage analysis is performed, the number of unique and sample-type specific kmers are generated by filtering out all kmers shared between sample types within splice-junction clusters. The number of these unique peptide kmers per splice-junction-modified transcript is used to generate a splicing antigenicity score per sample and per gene that is higher if the splice-junctions associated with a gene produce more peptide kmers able to bind to an MHC complex and have higher RNA-seq expression. This score can then be associated to other quantitative measures.

------------------------------------------------------------------------

# Installation

Requirements

[STAR](https://github.com/alexdobin/STAR)\
[arcasHLA](https://anaconda.org/bioconda/arcas-hla)\
[MHCnuggets](https://karchinlab.org/apps/appMHCnuggets.html) or [NetMHCpan](https://github.com/tzina97/netMHCpan/tree/main)\
[LeafCutter](https://davidaknowles.github.io/leafcutter/)\
R \>= 4.0.2\
python \>= 3.6.10\

To complete installation install the splicemutr package within your installation of R. This will install the packages necessary and prompt you to install the associated bioconductor packages.

``` r
install_github("FertigLab/splicemute")
```

To build the SpliceMutr environment used in the SpliceMutr paper, install [miniconda](https://docs.conda.io/projects/miniconda/en/latest/), create a conda environment, and build the conda package contained within splicemutr.yml

``` bash
conda env create -f environment.yml
```

------------------------------------------------------------------------

SpliceMutr is licensed under the [GNU General Public License v3](http://www.gnu.org/licenses/gpl-3.0.html)

If you find SpliceMutr useful, please cite as follows:

Palmer, Theron, et al. "SpliceMutr Enables Pan-Cancer Analysis of Splicing-Derived Neoantigen Burden in Tumors." bioRxiv, 2023, <https://doi.org/10.1101/2023.05.26.542165>.

Examples of how the SpliceMutr pipeline scripts have been used for the analysis associated with the manuscript "SpliceMutr enables pan-cancer analysis of splicing-derived neoantigen burden in tumors" can be found in the folder "analysis" under the TCGA and CM038 headings and within the pipeline section. There is a jupyter notebook explaining usage of the appropriate scripts.
