---
title: "simulated_reads"
author: "Theron Palmer"
date: "2024-01-19"
output: html_document
---

```{r setup, include=FALSE}
cran_packages <- c("BiocManager", "tidyverse", "dplyr")

lapply(cran_packages, function(x){
  if (!require(x, character.only = TRUE)) {
    install.packages(x, dependencies = TRUE, repos = "https://cran.r-project.org")
    library(x, character.only = TRUE)
  }
})

BiocManager::install(c("BiocFileCache", "polyester", "Biostrings"))

library(Biostrings)
library(polyester)

knitr::opts_knit$set(root.dir = './')

set.seed(27)
```

Polyester simulations

```{r}



# FASTA annotation
bfc <- BiocFileCache::BiocFileCache(ask = FALSE)
url <- "https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_39/gencode.v39.pc_transcripts.fa.gz"
fasta_file <- BiocFileCache::bfcrpath(bfc, url)
fasta <- readDNAStringSet(fasta_file)

# generating transcript gene map
fasta_names <- names(fasta)
transcripts <- unname(vapply(fasta_names,function(fname){str_split(fname,"[|]")[[1]][1]},character(1)))
genes <- unname(vapply(fasta_names,function(fname){str_split(fname,"[|]")[[1]][2]},character(1)))
genes_trans_map <- data.frame(gene=genes,transcript=transcripts,index=seq(length(transcripts)))
gene_counts <- genes_trans_map %>% group_by(gene) %>% dplyr::count()

# counting transcripts per gene
gene_counts_filt <- gene_counts %>% dplyr::filter(n>1) 
gene_counts_filt <- gene_counts_filt[order(gene_counts_filt$n,decreasing=F),] 
gene_counts_filt <- gene_counts_filt[seq(200),]

genes_trans_map_target <- genes_trans_map %>% dplyr::filter(gene %in% gene_counts_filt$gene)

# 50 genes differentially expressed, 50 differentially spliced, 50 both, the rest neither

gene_counts_filt[seq(1,200),"type"] <- "diff_splice" # isoform usage varies across groups
rownames(gene_counts_filt) <- gene_counts_filt$gene
genes_trans_map_target$type <- "diff_splice"

# ~20x coverage ----> reads per transcript = transcriptlength/readlength * 20
# here all transcripts will have ~equal FPKM
readspertx <- round(30 * width(fasta) / 100)

num_timepoints <- 2
countmat <- data.frame(matrix(0, nrow = length(fasta), ncol = num_timepoints))

a <- lapply(seq(nrow(gene_counts_filt)),function(row_val){
  gene_val <- rownames(gene_counts_filt)[row_val]
  genes_trans_map_small <- genes_trans_map_target %>% dplyr::filter(gene==gene_val)
  transcripts <- genes_trans_map_small$transcript
  indeces <- genes_trans_map_small$index
  type <- unique(genes_trans_map_small$type)
  up_side <- as.numeric(vapply(runif(1),function(rand_val){
    if (rand_val > 0.5) {
      return(2)
    } else {
      return(1)
    }
  },numeric(1)))
  down_side <- seq(2)[!(seq(2) %in% up_side)]
  up_indeces <- sample(indeces,floor(length(indeces)/2))
  down_indeces <- indeces[!(indeces %in% up_indeces)]
  if (type == "diff_splice") {
    total_reads_distr_1 <- sum(countmat[down_indeces, 1]) / length(up_indeces)
    total_reads_distr_2 <- sum(countmat[up_indeces, 1]) / length(down_indeces)

    countmat[up_indeces, up_side] <<- countmat[up_indeces, up_side]+4
    countmat[down_indeces, up_side] <<- 1
    countmat[down_indeces, down_side] <<- countmat[down_indeces, down_side] + 4
    countmat[up_indeces, down_side] <<- 1
  }
  return(c(indeces, type))
})

countmat_filt <- countmat[genes_trans_map_target$index, ]
countmat_labelled <- countmat_filt
countmat_labelled$gene <- genes_trans_map_target$gene
countmat_labelled$transcript <- genes_trans_map_target$transcript
countmat_filt <- as.matrix(countmat_filt)

readspertx_filt <- readspertx[genes_trans_map_target$index]

# subset the FASTA file to first 20 transcripts
small_fasta <- fasta[genes_trans_map_target$index]
writeXStringSet(small_fasta, "./gencode.v39.pc_transcripts_small.fa")

simulate_experiment("./gencode.v39.pc_transcripts_small.fa",
                    reads_per_transcript = readspertx_filt,
                    num_reps = c(4, 4),
                    fold_changes = countmat_filt,
                    outdir = './simulated_reads')

```


Forming groups file

```{r}

library(tidyverse)

sim_rep_file <- "./simulated_reads/sim_rep_info.txt"
sim_rep <- read.table(sim_rep_file, sep = "\t", header = T)

groups_file <-  sim_rep %>% select(rep_id, group)

write.table(groups_file,
            file = "./groups_file.txt",
            sep = "\t", col.names = FALSE, quote = FALSE, row.names = FALSE)

```

Forming genotypes_specific.txt file


HLA-A01-01,HLA-B02-01,HLA-C03-01  sample_1
HLA-A01-01,HLA-B02-01,HLA-C03-01  sample_2
HLA-A01-01,HLA-B02-01,HLA-C03-01  sample_3

```{r}

library(tidyverse)

sim_rep_file <- "./simulated_reads/sim_rep_info.txt"
sim_rep <- read.table(sim_rep_file, sep = "\t", header = TRUE)

hla <- paste(rep("HLA-A01-01", 6), collapse = ",")

sim_rep$HLA <- hla

genotypes_specific <- sim_rep %>% select(HLA, rep_id)
genotypes_specific$rep_id <- sprintf("%s.filt", genotypes_specific$rep_id)

write.table(
  genotypes_specific,
  sep = "\t",
  quote = FALSE,
  col.names = FALSE,
  row.names = FALSE,
  file = "./simulated_reads/genotypes_specific.txt"
)
```

# Forming fasta_files.txt

```{r}

library(dplyr)

sim_rep_file <- "./simulated_reads/sim_rep_info.txt"
sim_rep <- read.table(sim_rep_file, sep = "\t", header = TRUE)

fasta_directory <- sprintf("%s/simulated_reads",getwd())
sim_rep$samples <- vapply(sim_rep$rep_id,function(sample){sprintf("%s/%s_01.fasta",fasta_directory,sample)},character(1))

fasta_files <-  sim_rep %>% dplyr::select(samples)

write.table(
  fasta_files,
  sep = "\t",
  quote = FALSE,
  col.names = FALSE,
  row.names = FALSE,
  file = "./simulated_reads/fasta_files.txt"
)

```

