---

title: "analysis_Theron_08262021"
author: "Theron Palmer"
date: "8/26/2021"
output:
  html_document:
    toc: yes
    toc_float: yes
    
---

# Analysis Pipeline

On ZAPPA. recount3_tcga_juncs_prep.sh -> recount3_tcga_juncs.sh -> splicemutr_leafcutter.sh -> save_introns.sh -> cong_tcga_junc.R -> tcga_form_transcripts.sh -> process_peps.sh -> runMHCnuggets.sh -> process_percentile.sh -> create_genotypes.sh (TCGA_assign_imm.sh) -> TCGA_assign_imm_py.sh -> TCGA_combine_kmers.sh -> create_junc_expr.sh -> TCGA_gene_expr_per_sample.sh -> calc_gene_metric.sh

create_TCGA_TMB_maf_summary.sh

# Loading in necessary libraries

```{r,echo=FALSE}

library(dplyr)
library(knitr)
library(ggplot2)
library(TCGAutils)
library(maftools)
library(ggpubr)
library(stringr)
library(readxl)
library(ComplexHeatmap)
library(recount3)
library(DT)
library(rstatix)
library(ggrepel)
library(ggprism)

```

```{r}

mod_path <- function(path){ # requires path to be pulled from external mounted through wsl
  str_replace_all(path,"/mnt/f","F:")
}

```

# Accumulating mutation data per sample only annotated

## Preparing TCGA cibersort data [1]

```{r}

TCGA_cibersort_all <- read.table(mod_path("/mnt/f/TCGA_junctions/ext_dat/the_Immune_Landscape_of_Cancer/TCGA.Kallisto.fullIDs.cibersort.relative.tsv"),header=T)
TCGA_cibersort_all$barcode <- str_replace_all(TCGA_cibersort_all$SampleID,"[.]","-")
TCGA_cibersort_all$sample_ID <- vapply(TCGAbarcode(TCGA_cibersort_all$barcode,sample=T),
                                       function(val){substr(val,1,nchar(val)-1)},character(1))
TCGA_cibersort_all$patient_ID <- TCGAbarcode(TCGA_cibersort_all$barcode)
TCGA_cibersort_all <- TCGA_cibersort_all[as.numeric(TCGA_cibersort_all$P.value)<=0.05,]
cibersort_cells <- c("SampleID","B.cells.naive","B.cells.memory","Plasma.cells",
                     "T.cells.CD8","T.cells.CD4.naive","T.cells.CD4.memory.resting",
                     "T.cells.CD4.memory.activated","T.cells.follicular.helper",
                     "T.cells.regulatory..Tregs.","T.cells.gamma.delta","NK.cells.resting",
                     "NK.cells.activated","Monocytes","Macrophages.M0","Macrophages.M1",
                     "Macrophages.M2","Dendritic.cells.resting","Dendritic.cells.activated",
                     "Mast.cells.resting","Mast.cells.activated","Eosinophils","Neutrophils","P.value","Correlation","RMSE",
                     "barcode","sample_ID","patient_ID")
actual_cibersort_cells <- c("B.cells.naive","B.cells.memory","Plasma.cells",
                     "T.cells.CD8","T.cells.CD4.naive","T.cells.CD4.memory.resting",
                     "T.cells.CD4.memory.activated","T.cells.follicular.helper",
                     "T.cells.regulatory..Tregs.","T.cells.gamma.delta","NK.cells.resting",
                     "NK.cells.activated","Monocytes","Macrophages.M0","Macrophages.M1",
                     "Macrophages.M2","Dendritic.cells.resting","Dendritic.cells.activated",
                     "Mast.cells.resting","Mast.cells.activated","Eosinophils","Neutrophils")

```

## Preparing non-silent mutation load [1]

```{r}

# contains sample id minus the last letter, or rather the vial code

mutation_load_file <- mod_path("/mnt/f/TCGA_junctions/ext_dat/the_Immune_Landscape_of_Cancer/mutation-load_updated.txt")
mutation_load <- read.table(mutation_load_file,sep="\t",header=T)
colnames(mutation_load)<-c("cohort","patient_ID","sample_ID","silent_per_mb","non_silent_per_mb")

ggplot(mutation_load,aes(x=cohort,y=log10(non_silent_per_mb+1)))+
  geom_boxplot()+
  theme(axis.text.x = element_text(angle = 90))+
  ylab("log10(Non silent mutations per Mb)")

num_samples <- data.frame(table(mutation_load$cohort))
datatable(num_samples)

```

## Preparing Leuokocyte Fraction Data [1]

```{r}

leukocyte_fraction_file <- mod_path("/mnt/f/TCGA_junctions/ext_dat/the_Immune_Landscape_of_Cancer/TCGA_all_leuk_estimate.masked.20170107.tsv")
leukocyte_fraction <- read.table(leukocyte_fraction_file,sep="\t",header=F)
colnames(leukocyte_fraction)<-c("cohort","barcode","fraction")
leukocyte_fraction$patient_ID <- TCGAbarcode(leukocyte_fraction$barcode)
leukocyte_fraction$sample_ID <- vapply(TCGAbarcode(leukocyte_fraction$barcode,sample=T),
                                       function(val){substr(val,1,nchar(val)-1)},character(1))

```

## Preparing Immunogenic SNV Data [1]

```{r}

binding_snv_file <- mod_path("/mnt/f/TCGA_junctions/ext_dat/the_Immune_Landscape_of_Cancer/TCGA_pMHC_SNV_sampleSummary_MC3_v0.2.8.CONTROLLED_170404.tsv")
binding_snv <- read.table(binding_snv_file,sep="\t",header=T)
binding_snv$patient_ID <- TCGAbarcode(binding_snv$barcode)
binding_snv$sample_ID <- vapply(TCGAbarcode(binding_snv$barcode,sample=T),
                                       function(val){substr(val,1,nchar(val)-1)},character(1))

```

## Preparing TCR clonality data [1]

```{r}

tcr_clonality_file <-  mod_path("/mnt/f/TCGA_junctions/ext_dat/the_Immune_Landscape_of_Cancer/mitcr_sampleStatistics_20160714.tsv")
tcr_clonality <-  read.table(tcr_clonality_file,sep="\t",header=T)
tcr_clonality$sample_ID <- vapply(TCGAbarcode(tcr_clonality$AliquotBarcode,sample=T),
                                       function(val){substr(val,1,nchar(val)-1)},character(1))

```


## Preparing splicemutr data

```{r}

tumor_data_file <- mod_path("/mnt/f/TCGA_junctions/TCGA_cancers/JHPCE/cancer_dirs_filt.txt")
tumor_data <- read.table(tumor_data_file)
cancers <- tumor_data$V1

splicing_antigenicity_tum <- data.frame(cancers)

TMB_all_cor <- data.frame(cancers)
rownames(TMB_all_cor) <- cancers
TMB_all_pvals <- data.frame(cancers)
rownames(TMB_all_pvals) <- cancers
DA_all_cor <- data.frame(cancers)
rownames(DA_all_cor) <- cancers
DA_all_pvals <- data.frame(cancers)
rownames(DA_all_pvals) <- cancers
cibersort_all_pvals <- data.frame(cancers)
rownames(cibersort_all_pvals)<-cancers
cibersort_all_cor <- data.frame(cancers)
rownames(cibersort_all_cor)<-cancers
all_genes <- c()

TCGA_ROOT_DIR=mod_path("/mnt/f/TCGA_junctions/TCGA_cancers")
dups <- c()
for (i in seq(length(cancers))){
  cancer <- cancers[i]
  print(sprintf("%s: %d out of %d",cancer,i,length(cancers)))
  tumor_geno <- read.table(sprintf("%s/%s/JHPCE/%s_genotypes_specific.txt",TCGA_ROOT_DIR,cancer,cancer),header=T)
  tumor_geno$sample_ID <- vapply(TCGAbarcode(tumor_geno$aliquot_id,sample=T),
                                       function(val){substr(val,1,nchar(val)-1)},character(1))
  tumor_geno$patient_ID <- TCGAbarcode(tumor_geno$aliquot_id)
  tumor_geno <- tumor_geno %>% dplyr::filter(type=="T")
  tumor_geno <- tumor_geno[complete.cases(tumor_geno),]

  splice_mut_file <- sprintf("%s/%s/JHPCE/GENE_METRIC_CP/SA_tumor.txt",TCGA_ROOT_DIR,cancer)
  splice_mut_files <- read.table(splice_mut_file)
  for (j in seq(length(splice_mut_files$V1))){ # splicing antigenicity
    if (j == 1){
      combined_gene_metric <- readRDS(mod_path(splice_mut_files$V1[j]))
    } else {
      a<-readRDS(mod_path(splice_mut_files$V1[j]))
      combined_gene_metric <- cbind(combined_gene_metric,a)
    }
  }
  CP_file <- str_replace(splice_mut_files$V1[1],"gene_metric_mean_len_norm_no_gene_norm_tumor","coding_potential_LGC_tumor")
  coding_potential_LGC_tumor <- readRDS(mod_path(CP_file))
  HIGH_CP_GENES <- rownames(coding_potential_LGC_tumor)[coding_potential_LGC_tumor$coding_potential_LGC>0]

  splice_mut_file <- sprintf("%s/%s/JHPCE/GENE_METRIC_CP/SA_normal.txt",TCGA_ROOT_DIR,cancer)
  splice_mut_files <- read.table(splice_mut_file)
  for (j in seq(length(splice_mut_files$V1))){ 
    if (j == 1){
      combined_gene_metric_normal <- readRDS(mod_path(splice_mut_files$V1[j]))
    } else {
      a<-readRDS(mod_path(splice_mut_files$V1[j]))
      combined_gene_metric_normal <- cbind(combined_gene_metric_normal,a)
    }
  }
  

  tumor_geno <- tumor_geno[tumor_geno$external_id %in% colnames(combined_gene_metric),]
  external_ids <- tumor_geno$external_id
  combined_gene_metric <- combined_gene_metric[,external_ids]
  combined_gene_metric_normal <- combined_gene_metric_normal[,external_ids]
  conglomerate_genes <- unique(rownames(combined_gene_metric),rownames(combined_gene_metric_normal))
  combined_gene_metric_tumor <- combined_gene_metric[conglomerate_genes,]
  combined_gene_metric_tumor[is.na(combined_gene_metric_tumor)]<-0
  combined_gene_metric_normal <- combined_gene_metric_normal[conglomerate_genes,]
  combined_gene_metric_normal[is.na(combined_gene_metric_normal)]<-0
  combined_gene_metric_tumor <- apply(combined_gene_metric_tumor,1,mean,na.rm=T)
  if (cancer=="HNSC"){
    combined_gene_metric_tumor_HNSC <- data.frame(combined_gene_metric_tumor)
  }
  combined_gene_metric_normal <- apply(combined_gene_metric_normal,1,mean,na.rm=T)
  combined_gene_metric_DA <- data.frame(DA=(combined_gene_metric_tumor)/(combined_gene_metric_normal),
                                        DA_inv=(combined_gene_metric_normal)/(combined_gene_metric_tumor))
  
  combined_gene_metric_DA$cancer <- cancer
  combined_gene_metric_DA$DA[combined_gene_metric_DA$DA==Inf & !is.na(combined_gene_metric_DA$DA)]<-sample(combined_gene_metric_DA$DA[combined_gene_metric_DA$DA!=Inf & !is.na(combined_gene_metric_DA$DA) & combined_gene_metric_DA$DA > 1],length(which(combined_gene_metric_DA$DA==Inf & !is.na(combined_gene_metric_DA$DA))),replace=T)
  
  combined_gene_metric_DA$DA[combined_gene_metric_DA$DA_inv==Inf & !is.na(combined_gene_metric_DA$DA_inv)]<-sample(combined_gene_metric_DA$DA[combined_gene_metric_DA$DA_inv!=Inf & !is.na(combined_gene_metric_DA$DA_inv) & combined_gene_metric_DA$DA_inv > 1],length(which(combined_gene_metric_DA$DA_inv==Inf & !is.na(combined_gene_metric_DA$DA_inv))),replace=T)
  combined_gene_metric_DA$genes <- rownames(combined_gene_metric_DA)
  
  if (cancer=="HNSC"){
    combined_gene_metric_DA_HNSC <- combined_gene_metric_DA
  }
  
  if (i == 1){
    combined_gene_metric_DA_all <- combined_gene_metric_DA
  } else {
    combined_gene_metric_DA_all <- rbind(combined_gene_metric_DA_all,combined_gene_metric_DA)
  }

  high_DA_genes <-  combined_gene_metric_DA[combined_gene_metric_DA$DA>0,"genes"]
  high_DA_genes <- high_DA_genes[!is.na(high_DA_genes)]
  all_genes <- unique(c(all_genes,rownames(combined_gene_metric)))
  HIGH_DA_HIGH_CP_genes <- intersect(high_DA_genes,HIGH_CP_GENES)
  combined_gene_metric<-combined_gene_metric[HIGH_DA_HIGH_CP_genes,]

  colnames_combined_gene_metric <-colnames(combined_gene_metric)
  colnames(combined_gene_metric) <- vapply(colnames_combined_gene_metric,function(col_name){
    tumor_geno$sample_ID[which(tumor_geno$external_id == col_name)[1]]
  },character(1))
  combined_gene_metric_barcode <- vapply(colnames_combined_gene_metric,function(col_name){
    tumor_geno$aliquot_id[which(tumor_geno$external_id == col_name)[1]]
  },character(1))

  mutation_load_small <- mutation_load %>% dplyr::filter(sample_ID %in% colnames(combined_gene_metric))
  leukocyte_fraction_small <- leukocyte_fraction %>% dplyr::filter(sample_ID %in% colnames(combined_gene_metric))
  binding_snv_small <- binding_snv %>% dplyr::filter(sample_ID %in% colnames(combined_gene_metric))
  tcr_clonality_small <- tcr_clonality %>% dplyr::filter(sample_ID %in% colnames(combined_gene_metric))
  TCGA_cibersort_small <- TCGA_cibersort_all %>% dplyr::filter(sample_ID %in% colnames(combined_gene_metric))

  combined_gene_metric_per_sample<-data.frame(sample_ID=colnames(combined_gene_metric),barcode=combined_gene_metric_barcode)
  combined_gene_metric_per_sample$splicing_antigenicity <- vapply(seq(ncol(combined_gene_metric)),
                                               function(col_val){mean(as.numeric(combined_gene_metric[,col_val]),na.rm=T)},
                                               numeric(1))
  
  # I need to handle duplicate samples
  # print("mutation_load")
  combined_gene_metric_per_sample$non_silent_mutations_per_mb <- unlist(lapply(combined_gene_metric_per_sample$sample_ID,function(ID){
    a<-which(mutation_load_small$sample_ID==ID)
    if (length(a)==1){
      return(mutation_load_small$non_silent_per_mb[a])
    } else if (length(a)>1){
      dups<<-c(dups,ID)
      return(mutation_load_small$non_silent_per_mb[a[1]])
    } else {
      return(NA)
    }
  }))
  # print("leukocyte_fraction")
  combined_gene_metric_per_sample$leukocyte_fraction <- unlist(lapply(seq(length(combined_gene_metric_per_sample$sample_ID)),function(ID_num){
    ID<-combined_gene_metric_per_sample$sample_ID[ID_num]
    barcode<-combined_gene_metric_per_sample$barcode[ID_num]
    a<-which(leukocyte_fraction_small$sample_ID==ID)
    if (length(a)==1){
      return(leukocyte_fraction_small$fraction[a])
    } else if (length(a)>1){
      aa<-which(leukocyte_fraction$barcode==barcode)
      if (length(aa)==1){
        return(leukocyte_fraction_small$fraction[aa]) 
      } else if (length(aa)>1){
        dups<<-c(dups,ID)
        return(leukocyte_fraction_small$fraction[aa[1]])
      } else {
        return(NA)
      }
    } else {
      return(NA)
    }
  }))
  # print("numberOfImmunogenicMutation")
  combined_gene_metric_per_sample$numberOfImmunogenicMutation <- unlist(lapply(seq(length(combined_gene_metric_per_sample$sample_ID)),function(ID_num){
    ID<-combined_gene_metric_per_sample$sample_ID[ID_num]
    barcode<-combined_gene_metric_per_sample$barcode[ID_num]
    a<-which(binding_snv_small$sample_ID==ID)
    if (length(a)==1){
      return(binding_snv_small$numberOfImmunogenicMutation[a])
    } else if (length(a)>1){
      aa<-which(binding_snv_small$barcode==barcode)
      if (length(aa)==1){
        return(binding_snv_small$numberOfImmunogenicMutation[aa]) 
      } else if (length(aa)>1){
        dups<<-c(dups,ID)
        return(binding_snv_small$numberOfImmunogenicMutation[aa[1]])
      } else {
        return(NA)
      }
    } else  {
      return(NA)
    }
  }))
  # print("numberOfBindingExpresedPMHC")
  combined_gene_metric_per_sample$numberOfBindingExpressedPMHC <- unlist(lapply(seq(length(combined_gene_metric_per_sample$sample_ID)),function(ID_num){
    ID<-combined_gene_metric_per_sample$sample_ID[ID_num]
    barcode<-combined_gene_metric_per_sample$barcode[ID_num]
    a<-which(binding_snv_small$sample_ID==ID)
    if (length(a)==1){
      return(binding_snv_small$numberOfBindingExpressedPMHC[a])
    } else if (length(a)>1){
      aa<-which(binding_snv_small$barcode==barcode)
      if (length(aa)==1){
        return(binding_snv_small$numberOfBindingExpressedPMHC[aa]) 
      } else if (length(aa)>1){
        dups<<-c(dups,ID)
        return(binding_snv_small$numberOfBindingExpressedPMHC[aa[1]])
      } else {
        return(NA)
      }
    } else {
      return(NA)
    }
  }))
  # print("numClones")
  combined_gene_metric_per_sample$numClones <- unlist(lapply(seq(length(combined_gene_metric_per_sample$sample_ID)),function(ID_num){
    ID<-combined_gene_metric_per_sample$sample_ID[ID_num]
    barcode<-combined_gene_metric_per_sample$barcode[ID_num]
    a<-which(tcr_clonality$sample_ID==ID)
    if (length(a)==1){
      return(tcr_clonality$numClones[a])
    } else if (length(a)>1){
      aa<-which(tcr_clonality$AliquotBarcode==barcode)
      if (length(aa)==1){
        return(tcr_clonality$numClones[aa]) 
      } else if (length(aa)>1){
        dups<<-c(dups,ID)
        return(tcr_clonality$numClones[aa[1]])
      } else {
        return(NA)
      }
    } else {
      return(NA)
    }
  }))
  combined_gene_metric_per_sample$splicing_antigenicity_norm <- combined_gene_metric_per_sample$splicing_antigenicity/max(combined_gene_metric_per_sample$splicing_antigenicity)
  combined_gene_metric_per_sample$cancer <- cancer
  colnames(combined_gene_metric_per_sample) <- c("sample_ID","barcode","splicing_antigenicity","non_silent_mutations_per_mb",
                                                 "leukocyte_fraction","numberOfImmunogenicMutation","numberOfBindingExpressedPMHC",
                                                 "num_TCR_Clones","splicing_antigenicity_norm","cancer")
  combined_gene_metric_per_sample <- combined_gene_metric_per_sample[!(combined_gene_metric_per_sample$sample_ID %in% dups),]
  combined_gene_metric_per_sample$TMB_TYPE <- "LOW"
  combined_gene_metric_per_sample$TMB_TYPE <- unlist(lapply(combined_gene_metric_per_sample$non_silent_mutations_per_mb,function(val){
    if (is.na(val)){
      return(NA)
    } else if(val>=10){
      return("TMB HIGH")
    } else {
      return("TMB LOW")
    }}))
  # print("cibersort_per_samp")
  cibersort_per_samp <- lapply(seq(length(combined_gene_metric_per_sample$sample_ID)),
                             function(samp_num){
                               samp<-TCGA_cibersort_small$sample_ID[samp_num]
                               barcode <- TCGA_cibersort_small$barcode[samp_num]
                               a<-which(TCGA_cibersort_small$sample_ID==samp)
                               if (length(a)==1){
                                 return(TCGA_cibersort_small[a,cibersort_cells,drop=F])
                               } else if (length(a)>1){
                                 aa<-which(TCGA_cibersort_small$barcode==barcode)
                                 if (length(aa)==1){
                                   return(TCGA_cibersort_small[aa,cibersort_cells,drop=F])
                                 } else if (length(aa)>1){
                                   dups<<-c(dups,samp)
                                   return(TCGA_cibersort_small[aa[1],cibersort_cells,drop=F])
                                 } else {
                                   return(NA)
                                 }
                               } else {
                                a<-data.frame(t(rep(NA,length(cibersort_cells))))
                                colnames(a)<-cibersort_cells
                                return(a)
                              }
                             })
  cibersort_per_samp_df<-cibersort_per_samp[[1]]
  for (k in seq(2,length(cibersort_per_samp))){
    cibersort_per_samp_df <- rbind(cibersort_per_samp_df,cibersort_per_samp[[k]])
  }
  cibersort_per_samp_df <- cibersort_per_samp_df[complete.cases(cibersort_per_samp_df),]
  
  cibersort_cols_to_add <- c("splicing_antigenicity_norm","non_silent_mutations_per_mb","leukocyte_fraction",
                           "numberOfImmunogenicMutation","numberOfBindingExpressedPMHC","num_TCR_Clones",
                           "cancer")
  cibersort_per_samp_df_cols <- colnames(cibersort_per_samp_df)
  # print("cibersort_per_samp_df")
  cibersort_per_samp_df <- cbind(cibersort_per_samp_df,as.data.frame(matrix(unlist(lapply(seq(length(cibersort_per_samp_df$sample_ID)),function(ID_num){
                             ID <- cibersort_per_samp_df$sample_ID[ID_num]
                             barcode <- cibersort_per_samp_df$barcode[ID_num]
                             a<-which(combined_gene_metric_per_sample$sample_ID==ID)
                             if (length(a)==1){
                               return(as.character(combined_gene_metric_per_sample[a,cibersort_cols_to_add]))
                             } else if (length(a)>1){
                               aa<-which(combined_gene_metric_per_sample$barcode==barcode)
                               if (length(aa)==1){
                                 return(combined_gene_metric_per_sample[aa,cibersort_cols_to_add,drop=F])
                               } else if (length(aa)>1){
                                 dups<<-c(dups,ID)
                                 return(combined_gene_metric_per_sample[aa[1],cibersort_cols_to_add,drop=F])
                               } else {
                                 return(rep(NA,length(cibersort_cols_to_add)))
                               }
                             } else {
                               return(rep(NA,length(cibersort_cols_to_add)))
                             }
  })),byrow=T,nrow=nrow(cibersort_per_samp_df))))
  colnames(cibersort_per_samp_df) <- c(cibersort_per_samp_df_cols,cibersort_cols_to_add)
  cibersort_per_samp_df <- cibersort_per_samp_df[!(cibersort_per_samp_df$SampleID %in% dups),]
  
  a<-cor.test(combined_gene_metric_per_sample$splicing_antigenicity,log10(combined_gene_metric_per_sample$non_silent_mutations_per_mb+1),method="kendall")
  TMB_all_cor[cancer,"SA_vs_TMB_cor"]<-a$estimate
  TMB_all_pvals[cancer,"SA_vs_TMB_pval"]<-a$p.value

  a<-cor.test(combined_gene_metric_per_sample$splicing_antigenicity,combined_gene_metric_per_sample$leukocyte_fraction,method="kendall")
  TMB_all_cor[cancer,"SA_vs_leuk_frac_cor"]<-a$estimate
  TMB_all_pvals[cancer,"SA_vs_leuk_frac_pval"]<-a$p.value

  a<-cor.test(combined_gene_metric_per_sample$splicing_antigenicity,log10(combined_gene_metric_per_sample$numberOfImmunogenicMutation+1),method="kendall")
  TMB_all_cor[cancer,"SA_vs_numImmMut_cor"]<-a$estimate
  TMB_all_pvals[cancer,"SA_vs_numImmMut_pval"]<-a$p.value
  
  a<-cor.test(combined_gene_metric_per_sample$splicing_antigenicity,log10(combined_gene_metric_per_sample$numberOfBindingExpressedPMHC+1),method="kendall")
  TMB_all_cor[cancer,"SA_vs_numBindExprPMHC_cor"]<-a$estimate
  TMB_all_pvals[cancer,"SA_vs_numBindExprPMHC_pval"]<-a$p.value

  a<-cor.test(combined_gene_metric_per_sample$splicing_antigenicity,log2(combined_gene_metric_per_sample$num_TCR_Clones+1),method="kendall")
  TMB_all_cor[cancer,"SA_vs_num_TCR_Clones_cor"]<-a$estimate
  TMB_all_pvals[cancer,"SA_vs_num_TCR_Clones_pval"]<-a$p.value
  
  for (cell in actual_cibersort_cells){
    a<-cor.test(as.numeric(cibersort_per_samp_df$splicing_antigenicity),as.numeric(cibersort_per_samp_df[,cell]),method="kendall")
    cibersort_all_cor[cancer,sprintf("%s",cell)]<-as.numeric(a$estimate)
    cibersort_all_pvals[cancer,sprintf("%s",cell)]<-as.numeric(a$p.value)
  }
  
  if (i == 1){
    cibersort_per_samp_df_all <- cibersort_per_samp_df
  } else {
    cibersort_per_samp_df_all <- rbind(cibersort_per_samp_df_all,cibersort_per_samp_df)
  }
  
    if (i == 1){
    combined_gene_metric_per_sample_all <- combined_gene_metric_per_sample
  } else {
    combined_gene_metric_per_sample_all <- rbind(combined_gene_metric_per_sample_all,combined_gene_metric_per_sample)
  }
  print("-------------------------------------------------------")
}

```

### Reading in Gene Metric and Group Data for HNSCC Cohort

```{r}

gene_metric_mean_file_tumor <- "F:/head_and_neck_DARIA/data/splicemutr_02_13_2022/calc_gene_metric_out/gaykalova_gene_metric_mean_len_norm_no_gene_norm_tumor.rds"
gene_metric_mean_tumor <- readRDS(gene_metric_mean_file_tumor)
gene_metric_mean_file_normal <- "F:/head_and_neck_DARIA/data/splicemutr_02_13_2022/calc_gene_metric_out/gaykalova_gene_metric_mean_len_norm_no_gene_norm_normal.rds"
gene_metric_mean_normal <- readRDS(gene_metric_mean_file_normal)
# junction_expression_file <- "F:/head_and_neck_DARIA/data/splicemutr_02_13_2022/create_junc_expression/junc_expr_combined_vst.rds"
# junction_expression <- readRDS(junction_expression_file)
splicemutr_data_file <- "F:/head_and_neck_DARIA/data/splicemutr_02_13_2022/create_comparisons_out/gaykalova_splice_dat_all.rds"
splicemutr_data <- readRDS(splicemutr_data_file)
splicemutr_data <- splicemutr_data[as.numeric(splicemutr_data$deltapsi)>0,]
groups_file <- "F:/head_and_neck_DARIA/data/splicemutr_02_13_2022/groups_file.txt"
groups <- read.table(groups_file)
colnames(groups) <- c("sample","group")

```

### Processing Tumor and normal differential agretopicity for HNSCC cohort

```{r}

tumor_samples <- groups$sample[groups$group=="Tumor"]
normal_samples <- groups$sample[groups$group=="Normal_Mucosa"]
all_genes <- unique(c(rownames(gene_metric_mean_tumor,gene_metric_mean_normal)))
normal_summary <- data.frame(normal_SA=apply(gene_metric_mean_normal[,normal_samples],1,mean,na.rm=T))
normal_summary <- data.frame(normal_summary[all_genes,])
normal_summary[is.na(normal_summary)]<-0
rownames(normal_summary)<-all_genes
tumor_summary <- data.frame(tumor_SA=apply(gene_metric_mean_tumor[,tumor_samples],1,mean,na.rm=T))
tumor_summary <- data.frame(tumor_summary[all_genes,])
tumor_summary[is.na(tumor_summary)]<-0
rownames(tumor_summary) <- all_genes

DA <- tumor_summary/normal_summary
DA_inv <- normal_summary/tumor_summary
DA[is.na(DA)]<-1
DA[DA_inv==Inf] <- mean(DA[DA_inv!=Inf & DA!=Inf & DA < 1])
DA[DA==Inf] <- mean(DA[DA_inv!=Inf & DA!=Inf & DA > 1])
colnames(DA)<-"DA"
DA$type <- vapply(DA$DA,function(val){
  if (val < 1){
    return("normal high")
  } else if (val > 1){
    "tumor high"
  } else {
    return("equal")
  }
},character(1))

DA$DA_filt <- vapply(log2(DA$DA),function(val){
  if(val>0){
    return(val)
  } else if (val<0){
    return(-1*val)
  } else {return(val)}
},numeric(1))

```

### Calculating correlation between HNSC Differential Agretopicity and Tumor splicing antigenicity

```{r}

colnames(DA)<-c("DA_cohort","type_cohort","DA_filt_cohort")
colnames(tumor_summary)<-"SA_cohort"
genes_first_part <- unname(vapply(intersect(rownames(combined_gene_metric_tumor_HNSC),rownames(combined_gene_metric_DA_HNSC)),function(gene){
  strsplit(gene,"[.]")[[1]][1]
},character(1)))
all_genes <- intersect(genes_first_part,intersect(rownames(DA),rownames(tumor_summary)))

HNSC_SA <- cbind(data.frame(combined_gene_metric_tumor_HNSC[all_genes,]),
                            data.frame(combined_gene_metric_DA_HNSC[all_genes,]),data.frame(DA[all_genes,]),
                            data.frame(tumor_summary[all_genes,]))
rownames(HNSC_SA)<-all_genes
colnames(HNSC_SA)<-c("SA_TCGA","DA_TCGA","DA_inv_TCGA","cacner_TCGA","genes","DA_cohort","type_cohort","DA_filt_cohort","SA_cohort")
HNSC_SA$DA_TCGA[is.na(HNSC_SA$DA_TCGA)]<-1
HNSC_SA$DA_cohort[is.na(HNSC_SA$DA_cohort)]<-1
HNSC_SA$DA_inv_TCGA[is.na(HNSC_SA$DA_inv_TCGA)]<-1
HNSC_SA$DA_filt_cohort[is.na(HNSC_SA$DA_filt_cohort)]<-1
HNSC_SA[is.na(HNSC_SA)]<-0

ggplot(HNSC_SA,aes(x=SA_TCGA,y=SA_cohort))+
  geom_point()+
  stat_cor(method = "kendall",label.x.npc="middle",label.y.npc="top")+
  geom_smooth(method="lm")+
  xlab("Splicing Antigenicity HNSC TCGA")+
  ylab("Splicing Antigenicity HNSC Cohort")
ggplot(HNSC_SA,aes(x=log2(DA_TCGA),y=log2(DA_cohort)))+geom_point()+
  stat_cor(method = "kendall",label.x.npc="middle",label.y.npc="top")+
  geom_smooth(method="lm")+
  xlab("Differential Agretopicity HNSC TCGA")+
  ylab("Differential Agretopicity HNSC Cohort")


```


## Plotting splicing antigenicity give TMB split

```{r}

cancers <- unique(combined_gene_metric_per_sample_all$cancer)
comb_gene_metric_fold_change_cor <- data.frame(cancer=cancers)
rownames(comb_gene_metric_fold_change_cor)<-cancers
comb_gene_metric_pvals <- data.frame(cancer=cancers)
rownames(comb_gene_metric_pvals) <- cancers
counts <- data.frame(cancer=cancers)
row.names(counts) <- cancers

for (cancer in cancers){
  print(cancer)
  # performing TMB type eval
  rows_filt<-!is.na(combined_gene_metric_per_sample_all$non_silent_mutations_per_mb) & combined_gene_metric_per_sample_all$cancer==cancer
  combined_gene_metric_per_sample_filt <- combined_gene_metric_per_sample_all[rows_filt,]
  combined_gene_metric_per_sample_filt$TMB_TYPE <- factor(combined_gene_metric_per_sample_filt$TMB_TYPE,levels=c("TMB LOW","TMB HIGH"))
  HIGH_COUNT=length(which(combined_gene_metric_per_sample_filt$TMB_TYPE=="TMB HIGH"))
  LOW_COUNT=length(which(combined_gene_metric_per_sample_filt$TMB_TYPE=="TMB LOW"))
  counts[cancer,"HIGH_COUNT"] <- HIGH_COUNT
  counts[cancer,"LOW_COUNT"] <- LOW_COUNT
  
  if (HIGH_COUNT > 0 & LOW_COUNT > 0){
    
    wilcox_test_val <- compare_means(splicing_antigenicity ~ TMB_TYPE, p.adjust.method = "BH", 
                             data=combined_gene_metric_per_sample_filt)
    TMB_HIGH_VAL <- mean(combined_gene_metric_per_sample_filt[combined_gene_metric_per_sample_filt$TMB_TYPE=="TMB HIGH","splicing_antigenicity"],na.rm=T)
    TMB_LOW_VAL <- mean(combined_gene_metric_per_sample_filt[combined_gene_metric_per_sample_filt$TMB_TYPE=="TMB LOW","splicing_antigenicity"],na.rm=T)
    comb_gene_metric_fold_change_cor[cancer,"log2_TMB_HIGH_ov_TMB_LOW"] <- log2(TMB_HIGH_VAL/TMB_LOW_VAL)
    comb_gene_metric_pvals[cancer,"BH_pval_wilcoxon"]<-wilcox_test_val$p.adj
  } else {
      comb_gene_metric_fold_change_cor[cancer,"log2_TMB_HIGH_ov_TMB_LOW"] <- 1
      comb_gene_metric_pvals[cancer,"BH_pval_wilcoxon"]<-1
  }
}

comb_gene_metric <- data.frame(cancer=cancers,
                               log2_TMB_HIGH_ov_TMB_LOW=comb_gene_metric_fold_change_cor$log2_TMB_HIGH_ov_TMB_LOW,
                               BH_pval_wilcoxon=comb_gene_metric_pvals$BH_pval_wilcoxon)

cancer_HIGH_ov_LOW <- sprintf("%s:%d/%d",comb_gene_metric$cancer,counts[comb_gene_metric$cancer,"HIGH_COUNT"],counts[comb_gene_metric$cancer,"LOW_COUNT"])
ggplot(comb_gene_metric,aes(x=log2_TMB_HIGH_ov_TMB_LOW,y=-log10(BH_pval_wilcoxon),
                            label=cancer,color=cancer_HIGH_ov_LOW))+
  geom_hline(yintercept=-log10(0.05))+
  geom_point()+
  geom_text_repel(direction="y")+
  xlab("log2(mean HIGH TMB sample SA / mean LOW TMB sample SA)")+
  ylab("-log10(wilcoxon BH adjusted pval)")+
  labs(color='cancer:(TMB High)/(TMB LOW)')

```

## Plotting splicing antigenicity vs all other metrics kendall correlation 

```{r}

# splitting TMB_all 
TMB_all_cor_filt <- t(TMB_all_cor[,seq(2,ncol(TMB_all_cor))])
TMB_all_pvals_filt <- t(TMB_all_pvals[,seq(2,ncol(TMB_all_pvals))])

Heatmap(as.matrix(TMB_all_cor_filt), cell_fun = function(j, i, x, y, w, h, fill) {
  if (is.na(TMB_all_pvals_filt[i,j])){
    grid.text(".",x,y)
  }else if(TMB_all_pvals_filt[i, j] < 0.005) {
      grid.text("**", x, y)
  } else if(TMB_all_pvals_filt[i, j] < 0.05) {
      grid.text("*", x, y)
  }
},cluster_columns=T,name="kendall cor",
clustering_method_rows="ward.D2",
clustering_method_columns="ward.D2")


TMB_all <- cbind(TMB_all_cor,TMB_all_pvals[,seq(2,ncol(TMB_all_pvals))])
TMB_all_cor_cols<-colnames(TMB_all_cor)
TMB_all_pval_cols <- str_replace(TMB_all_cor_cols,"cor","pval")
for (i in seq(nrow(TMB_all_cor_filt))){
  print(ggplot(TMB_all,aes(x=TMB_all[,i+1],y=-log10(TMB_all[,i+1+5]),label=cancers,color=cancers))+
          geom_hline(yintercept=-log10(0.05))+
          geom_text_repel(direction="y")+
          geom_point()+
          xlab(TMB_all_cor_cols[i+1])+
          ylab(TMB_all_pval_cols[i+1])+
          theme(legend.position="none"))
}


```

## Processing CIBERSORT Data

```{r}

cibersort_all_cor_filt<-t(cibersort_all_cor[,seq(2,ncol(cibersort_all_cor))])
cibersort_all_cor_filt[is.na(cibersort_all_cor_filt)]<-0
cibersort_all_pvals_filt<-t(cibersort_all_pvals[,seq(2,ncol(cibersort_all_pvals))])
cibersort_all_pvals_filt[is.na(cibersort_all_pvals_filt)]<-1

print(Heatmap(as.matrix(cibersort_all_cor_filt), cell_fun = function(j, i, x, y, w, h, fill) {
  if (is.na(cibersort_all_pvals_filt[i,j])){
    grid.text(".",x,y)
  }else if(cibersort_all_pvals_filt[i, j] < 0.005) {
      grid.text("**", x, y)
  } else if(cibersort_all_pvals_filt[i, j] < 0.05) {
      grid.text("*", x, y)
  }
},cluster_columns=T,name="kendall cor",
clustering_method_rows="ward.D2",
clustering_method_columns="ward.D2"))

```

## TMB Correlation for all samples

```{r}

HIGH_COUNT=length(which(combined_gene_metric_per_sample_all$TMB_TYPE=="TMB HIGH"))
LOW_COUNT=length(which(combined_gene_metric_per_sample_all$TMB_TYPE=="TMB LOW"))
my_comparisons <- list( c("TMB LOW", "TMB HIGH"))
wilcox_test_val <- compare_means(splicing_antigenicity ~ TMB_TYPE, 
                             comparisons = my_comparisons, p.adjust.method = "BH", data=combined_gene_metric_per_sample_all)
combined_gene_metric_per_sample_all_filt <- combined_gene_metric_per_sample_all[!is.na(combined_gene_metric_per_sample_all$non_silent_mutations_per_mb),]
print(ggplot(combined_gene_metric_per_sample_all_filt,aes(y=splicing_antigenicity,x=TMB_TYPE))+
  geom_boxplot(fill="grey")+stat_pvalue_manual(wilcox_test_val, 
  y.position = max(combined_gene_metric_per_sample_all_filt$splicing_antigenicity)+(max(combined_gene_metric_per_sample_all_filt$splicing_antigenicity)/2),
  label = "p.adj" )+labs(title=sprintf("all cohorts, HIGH: %d, LOW: %d",HIGH_COUNT,LOW_COUNT))+
    ylab(("splicing antigenicity")))

ggplot(combined_gene_metric_per_sample_all_filt,aes(x=splicing_antigenicity,y=log10(non_silent_mutations_per_mb+1)))+
  geom_point()+
  stat_cor(method = "kendall",label.x.npc="middle",label.y.npc="top")+
  geom_smooth(method="lm")+
    labs("All Samples")+
  ylab("log10(TMB)")

ggplot(combined_gene_metric_per_sample_all_filt,aes(x=splicing_antigenicity,y=log10(non_silent_mutations_per_mb+1)))+
  geom_point()+
  stat_cor(method = "kendall",label.x.npc="middle",label.y.npc="top")+
  geom_smooth(method="lm")+
    labs("All Samples")+
  ylab("log10(TMB)")+
  xlim(c(0,0.00005))

combined_gene_metric_per_sample_all_sum <- data.frame(cancer=unique(combined_gene_metric_per_sample_all_filt$cancer))
rownames(combined_gene_metric_per_sample_all_sum) <- combined_gene_metric_per_sample_all_sum$cancer
for (cancer in combined_gene_metric_per_sample_all$cancer){
  combined_gene_metric_per_sample_small <- combined_gene_metric_per_sample_all_filt[combined_gene_metric_per_sample_all_filt$cancer==cancer,]
  combined_gene_metric_per_sample_all_sum[combined_gene_metric_per_sample_all_sum$cancer==cancer,"median_TMB"] <- median(combined_gene_metric_per_sample_small$non_silent_mutations_per_mb,na.rm=T)
  combined_gene_metric_per_sample_all_sum[combined_gene_metric_per_sample_all_sum$cancer==cancer,"splicing_antigenicity"] <- mean(combined_gene_metric_per_sample_small$splicing_antigenicity,na.rm=T)
}
ggplot(combined_gene_metric_per_sample_all_sum,aes(x=splicing_antigenicity,y=median_TMB,label=cancer))+
  stat_cor(method = "kendall",label.x.npc="middle",label.y.npc="top")+
  geom_smooth(method="lm")+
  geom_text_repel()+
  labs("All Samples")+
  xlab("splicing antigenicity")+ylab("median TMB")+
  theme(legend.position="none")


ggplot(combined_gene_metric_DA_all,aes(x=cancer,y=log2(DA)))+
  geom_boxplot()+
  geom_hline(yintercept=0)+
  ylab("Differential Agretopicity")+
  xlab("Cancer Subtype")
ggplot(combined_gene_metric_DA_all,aes(x=cancer,y=log2(DA)))+
  geom_violin()+
  geom_hline(yintercept=0)+
  ylab("Differential Agretopicity")+
  xlab("Cancer Subtype")

```


## TMB, splicing antigenicity, and junction types per cancer

```{r}

tumor_data_file <- mod_path("/mnt/f/TCGA_junctions/TCGA_cancers/JHPCE/cancer_dirs_filt.txt")
tumor_data <- read.table(tumor_data_file)
cancers <- tumor_data$V1
TCGA_ROOT_DIR=mod_path("/mnt/f/TCGA_junctions/TCGA_cancers")
intron_counts_per_cancer <- data.frame(cancers)
rownames(intron_counts_per_cancer)<-intron_counts_per_cancer$cancers
intron_types <- c("annotated","cryptic_fiveprime","cryptic_threeprime","cryptic_unanchored","novel annotated pair")
intron_counts_per_cancer[,intron_types]<-0
intron_counts_per_cancer <- intron_counts_per_cancer[,seq(2,ncol(intron_counts_per_cancer))]

for (cancer in cancers){
  print(cancer)
  cancer_leafcutter_file <- sprintf("%s/%s/JHPCE/leafcutter_out/data.Rdata",TCGA_ROOT_DIR,cancer)
  load(cancer_leafcutter_file)
  introns_tumor <- introns[introns$deltapsi>0,]
  intron_counts_per_cancer[cancer,]<-vapply(intron_types,function(intron_type){
    return(length(which(introns_tumor$verdict==intron_type)))
  },numeric(1))
}

intron_counts_per_cancer_norm <- intron_counts_per_cancer/apply(intron_counts_per_cancer,1,sum)
intron_counts_per_cancer_scaled <- data.frame(apply(intron_counts_per_cancer_norm,2,scale))
rownames(intron_counts_per_cancer_scaled)<-rownames(intron_counts_per_cancer_norm)
colnames(intron_counts_per_cancer_scaled) <- intron_types

print(Heatmap(t(intron_counts_per_cancer_scaled),
        top_annotation = HeatmapAnnotation(median_TMB=anno_barplot(combined_gene_metric_per_sample_all_sum[rownames(intron_counts_per_cancer),"median_TMB"])),
        bottom_annotation = HeatmapAnnotation(splicing_antigenicity=anno_barplot(combined_gene_metric_per_sample_all_sum[rownames(intron_counts_per_cancer),"splicing_antigenicity"])),
        show_row_names=T,
        show_column_names = T,
        cluster_rows=T,
        cluster_columns=T,
        heatmap_legend_param=list(title="zscore"),
clustering_method_rows="ward.D2",
clustering_method_columns="ward.D2"))

print(Heatmap(t(log10(intron_counts_per_cancer)),
        top_annotation = HeatmapAnnotation(median_TMB=anno_barplot(combined_gene_metric_per_sample_all_sum[rownames(intron_counts_per_cancer),"median_TMB"])),
        show_row_names=T,
        show_column_names = T,
        cluster_rows=T,
        cluster_columns=T,
        heatmap_legend_param=list(title="log10(counts)"),
clustering_method_rows="ward.D2",
clustering_method_columns="ward.D2"))

```


# Analyzing Splicing Factor Mutations per cancer type including missense mutations

```{r}

# Reading in supplementary data

TCGA_mutation_representation <- read_excel(mod_path("/mnt/f/splicing_factor_paper/NIHMS958968-supplement-3.xlsx"),skip=2)
TCGA_mutation_representation$sample_ID <- vapply(TCGAbarcode(TCGA_mutation_representation$Tumor_Sample_Barcode,sample=T),
                                       function(val){substr(val,1,nchar(val)-1)},character(1))
TCGA_mutation_representation_filt <- TCGA_mutation_representation[TCGA_mutation_representation$sample_ID %in% combined_gene_metric_per_sample_all$sample_ID,]
cancers <- unique(TCGA_mutation_representation_filt$Cohort)
splicing_factor_genes <- unique(TCGA_mutation_representation_filt$Hugo_Symbol)
for (gene in splicing_factor_genes){
  combined_gene_metric_per_sample_all[,gene]<-"WT"
  TCGA_mutation_representation_filt_small <- TCGA_mutation_representation_filt[TCGA_mutation_representation_filt$Hugo_Symbol==gene,]
  combined_gene_metric_per_sample_all[combined_gene_metric_per_sample_all$sample %in% TCGA_mutation_representation_filt_small$sample_ID,gene]<-"MUT"
}

for (cancer in cancers){
    combined_gene_metric_per_sample_all_small <- combined_gene_metric_per_sample_all[combined_gene_metric_per_sample_all$cancer==cancer,]
    my_comparisons=list(c("WT","MUT"))
    
    MUT_VALS<-length(which(combined_gene_metric_per_sample_all_small$SF3B1=="MUT"))
    WT_VALS<-length(which(combined_gene_metric_per_sample_all_small$SF3B1=="WT"))
    if (MUT_VALS>0 & WT_VALS>0){
      wixoc_test_val <- compare_means(splicing_antigenicity ~ SF3B1,comparisons = my_comparisons, p.adjust.method = "BH", data=combined_gene_metric_per_sample_all_small)
      print(ggplot(combined_gene_metric_per_sample_all_small,aes(y=splicing_antigenicity,x=SF3B1))+
      geom_boxplot(fill="grey")+
        stat_pvalue_manual(wixoc_test_val, 
    y.position = max(combined_gene_metric_per_sample_all_small$splicing_antigenicity)+(max(combined_gene_metric_per_sample_all_small$splicing_antigenicity)/2),
    label = "p.adj" )+
      labs(title=sprintf("%s: MUT=%d,WT=%d",cancer,MUT_VALS,WT_VALS))+
        ylab("splicing antigenicity"))
    } else {
      print(ggplot(combined_gene_metric_per_sample_all_small,aes(y=splicing_antigenicity,x=SF3B1))+
      geom_boxplot(fill="grey")+
      labs(title=sprintf("%s: MUT=%d,WT=%d",cancer,MUT_VALS,WT_VALS))+
        ylab("splicing antigenicity"))
    }

    MUT_VALS<-length(which(combined_gene_metric_per_sample_all_small$U2AF1=="MUT"))
    WT_VALS<-length(which(combined_gene_metric_per_sample_all_small$U2AF1=="WT"))
    if (MUT_VALS>0 & WT_VALS>0){
      wixoc_test_val <- compare_means(splicing_antigenicity ~ U2AF1,comparisons = my_comparisons, p.adjust.method = "BH", data=combined_gene_metric_per_sample_all_small)
      print(ggplot(combined_gene_metric_per_sample_all_small,aes(y=splicing_antigenicity,x=U2AF1))+
      geom_boxplot(fill="grey")+
        stat_pvalue_manual(wixoc_test_val, 
    y.position = max(combined_gene_metric_per_sample_all_small$splicing_antigenicity)+(max(combined_gene_metric_per_sample_all_small$splicing_antigenicity)/2),
    label = "p.adj" )+
      labs(title=sprintf("%s: MUT=%d,WT=%d",cancer,MUT_VALS,WT_VALS))+
        ylab("splicing antigenicity"))
    } else {
      print(ggplot(combined_gene_metric_per_sample_all_small,aes(y=splicing_antigenicity,x=U2AF1))+
      geom_boxplot(fill="grey")+
      labs(title=sprintf("%s: MUT=%d,WT=%d",cancer,MUT_VALS,WT_VALS))+
        ylab("splicing antigenicity"))
    }

        
    MUT_VALS<-length(which(combined_gene_metric_per_sample_all_small$SRSF2=="MUT"))
    WT_VALS<-length(which(combined_gene_metric_per_sample_all_small$SRSF2=="WT"))
    if (MUT_VALS>0 & WT_VALS>0){
      wixoc_test_val <- compare_means(splicing_antigenicity ~ SRSF2,comparisons = my_comparisons, p.adjust.method = "BH", data=combined_gene_metric_per_sample_all_small)
      print(ggplot(combined_gene_metric_per_sample_all_small,aes(y=splicing_antigenicity,x=SRSF2))+
        geom_boxplot(fill="grey")+
        stat_pvalue_manual(wixoc_test_val, 
    y.position = max(combined_gene_metric_per_sample_all_small$splicing_antigenicity)+(max(combined_gene_metric_per_sample_all_small$splicing_antigenicity)/2),
    label = "p.adj" )+
      labs(title=sprintf("%s: MUT=%d,WT=%d",cancer,MUT_VALS,WT_VALS))+
        ylab("splicing antigenicity"))
    } else {
      print(ggplot(combined_gene_metric_per_sample_all_small,aes(y=splicing_antigenicity,x=SRSF2))+
      geom_boxplot(fill="grey")+
      labs(title=sprintf("%s: MUT=%d,WT=%d",cancer,MUT_VALS,WT_VALS))+
        ylab("splicing antigenicity"))
    }
    
    MUT_VALS<-length(which(combined_gene_metric_per_sample_all_small$RBM10=="MUT"))
    WT_VALS<-length(which(combined_gene_metric_per_sample_all_small$RBM10=="WT"))
    if (MUT_VALS>0 & WT_VALS>0){
      wixoc_test_val <- compare_means(splicing_antigenicity ~ RBM10,comparisons = my_comparisons, p.adjust.method = "BH", data=combined_gene_metric_per_sample_all_small)
      print(ggplot(combined_gene_metric_per_sample_all_small,aes(y=splicing_antigenicity,x=RBM10))+
      geom_boxplot(fill="grey")+
        stat_pvalue_manual(wixoc_test_val,
    y.position = max(combined_gene_metric_per_sample_all_small$splicing_antigenicity)+(max(combined_gene_metric_per_sample_all_small$splicing_antigenicity)/2),
    label = "p.adj" )+
      labs(title=sprintf("%s: MUT=%d,WT=%d",cancer,MUT_VALS,WT_VALS))+
        ylab("splicing antigenicity"))
    } else {
      print(ggplot(combined_gene_metric_per_sample_all_small,aes(y=splicing_antigenicity,x=RBM10))+
      geom_boxplot(fill="grey")+
      labs(title=sprintf("%s: MUT=%d,WT=%d",cancer,MUT_VALS,WT_VALS))+
        ylab("splicing antigenicity"))
    }
    
}

```

## Evaluating the 119 splicing factor genes in the splicing paper manner

```{r,eval=F}

splicing_factor_genes<-read_excel("/media/theron/One_Touch/TCGA_junctions/ext_dat/splicing_factor_genes1.xlsx")
splice_maf_file <- read.table("/media/theron/One_Touch/TCGA_junctions/TCGA_cancers/JHPCE/filter_maf/output/filenames.txt",header=F)

for (i in seq(nrow(splice_maf_file))){
  file<-splice_maf_file[i,1]
  gene <- unlist(strsplit(basename(file),"_"))[1]
  gene_maf_dat <- read.table(file,header=T,sep="\t",quote="")
  gene_maf_dat_filt<-gene_maf_dat[gene_maf_dat$Hugo_Symbol==gene,]
  if (i==1){
    splice_maf_data <- gene_maf_dat_filt
  } else {
    splice_maf_data <- rbind(splice_maf_data,gene_maf_dat_filt)
  }
}

saveRDS(splice_maf_data,file="/media/theron/One_Touch/TCGA_junctions/TCGA_cancers/JHPCE/filter_maf/output/splicing_factor_data.maf.rds")
write.table(splice_maf_data,file="/media/theron/One_Touch/TCGA_junctions/TCGA_cancers/JHPCE/filter_maf/output/splicing_factor_data.maf",col.names=T,sep="\t",quote=F)

```

```{r}


splice_vals <- data.frame(cancer=cancers)
rownames(splice_vals) <- cancers

# Ben Ho Park genes: SF3B1, U2AF1 or SRSF2
splice_maf_data <- readRDS(mod_path("/mnt/f/TCGA_junctions/TCGA_cancers/JHPCE/filter_maf/output/splicing_factor_data.maf.rds"))
splice_maf_data$sample_ID <- vapply(TCGAbarcode(splice_maf_data$Tumor_Sample_Barcode,sample=T),
                                       function(val){substr(val,1,nchar(val)-1)},character(1))
splice_maf_data_filt <- splice_maf_data[splice_maf_data$sample_ID %in% combined_gene_metric_per_sample_all$sample,]
splicing_factor_genes <- c(unique(splice_maf_data$Hugo_Symbol),"RBM39")
cancers <- unique(combined_gene_metric_per_sample_all$cancer)
for (gene in splicing_factor_genes){
  combined_gene_metric_per_sample_all[,gene]<-"WT"
  splice_maf_data_filt_small <- splice_maf_data_filt[splice_maf_data_filt$Hugo_Symbol==gene,]
  combined_gene_metric_per_sample_all[combined_gene_metric_per_sample_all$sample %in% splice_maf_data_filt_small$sample_ID,gene]<-"MUT"
}
combined_gene_metric_per_sample_all[,"all_genes"]<-"WT"
combined_gene_metric_per_sample_all[combined_gene_metric_per_sample_all$sample %in% splice_maf_data_filt$sample_ID,"all_genes"]<-"MUT"

all_comps_df <- data.frame(cancer=cancers)
rownames(all_comps_df) <- cancers


for (i in seq(length(cancers))){
  print(i)
  cancer <- cancers[i]
  combined_gene_metric_per_sample_all_small <- combined_gene_metric_per_sample_all[combined_gene_metric_per_sample_all$cancer==cancer,]
  my_comparisons=list(c("WT","MUT"))

  MUT_VALS<-length(which(combined_gene_metric_per_sample_all_small$all_genes=="MUT"))
  WT_VALS<-length(which(combined_gene_metric_per_sample_all_small$all_genes=="WT"))
  if (MUT_VALS>0 & WT_VALS>0){
    wixoc_test_val <- compare_means(splicing_antigenicity ~ all_genes,comparisons = my_comparisons, p.adjust.method = "BH", data=combined_gene_metric_per_sample_all_small)
    MUT_SA <- combined_gene_metric_per_sample_all_small[combined_gene_metric_per_sample_all_small$all_genes=="MUT","splicing_antigenicity"]
    WT_SA <- combined_gene_metric_per_sample_all_small[combined_gene_metric_per_sample_all_small$all_genes=="WT","splicing_antigenicity"]
  splice_vals[cancer,"MUT_ov_WT"]<-mean(MUT_SA,na.rm=T)/mean(WT_SA,na.rm=T)
  splice_vals[cancer,"BH_pval"]<-wixoc_test_val$p.adj
  splice_vals[cancer,"MUT_COUNT"] <- MUT_VALS
  splice_vals[cancer,"WT_COUNT"] <- WT_VALS
    
  } else {
    splice_vals_MUT_ov_WT[cancer,"MUT_ov_WT"]<-NA
    splice_vals_pvals[cancer,"BH_pval"]<-NA
    splice_vals[cancer,"MUT_COUNT"] <- MUT_VALS
    splice_vals[cancer,"WT_COUNT"] <- WT_VALS
  }
  
  wilcox_test_val <- compare_means(splicing_antigenicity ~ all_genes, p.adjust.method = "BH", data=combined_gene_metric_per_sample_all_small)
  print(ggplot(combined_gene_metric_per_sample_all_small,aes(y=splicing_antigenicity,x=all_genes))+
  geom_boxplot()+
    add_pvalue(wilcox_test_val,label = "p.signif", remove.bracket = TRUE,y.position=max(combined_gene_metric_per_sample_all_small$splicing_antigenicity)+0.01)+
    labs(title=cancer)+
    xlab("Mutations in Splicing Machinery")+
    ylab("Splicing Antigenicity"))
    
  for (j in seq(length(splicing_factor_genes))){
    gene <- splicing_factor_genes[j]
    gene_metric_av_gene_df <- data.frame(splicing_antigenicity=combined_gene_metric_per_sample_all_small$splicing_antigenicity,
                                   gene=combined_gene_metric_per_sample_all_small[,gene])
    if (length(unique(gene_metric_av_gene_df$gene))==1){
      a<-c(rep(NA,8),cancer,gene,NA,
           length(gene_metric_av_gene_df$splicing_antigenicity[gene_metric_av_gene_df$gene=="WT"]),
                  length(gene_metric_av_gene_df$splicing_antigenicity[gene_metric_av_gene_df$gene=="MUT"]))
    } else {
      a<-compare_means(splicing_antigenicity~gene,data=gene_metric_av_gene_df)
      a$cancer <- cancer
      a$gene <- gene
      a$mut_ov_wt <- mean(gene_metric_av_gene_df$splicing_antigenicity[gene_metric_av_gene_df$gene=="MUT"],na.rm=T)/mean(gene_metric_av_gene_df$splicing_antigenicity[gene_metric_av_gene_df$gene=="WT"],na.rm=T)
      a$WT_NUM <- length(gene_metric_av_gene_df$splicing_antigenicity[gene_metric_av_gene_df$gene=="WT"])
      a$MUT_NUM <- length(gene_metric_av_gene_df$splicing_antigenicity[gene_metric_av_gene_df$gene=="MUT"])
      a<-data.frame(a)
      
    }
    if (i == 1){
      all_comps <- a
    } else {
      all_comps <- rbind(all_comps,a)
    }
  }
}

for (j in seq(length(splicing_factor_genes))){
  gene <- splicing_factor_genes[j]
  gene_metric_av_gene_df <- data.frame(splicing_antigenicity=combined_gene_metric_per_sample_all$splicing_antigenicity,
                                 gene=combined_gene_metric_per_sample_all[,gene])
  if (length(unique(gene_metric_av_gene_df$gene))==1){
    a<-rep(NA,13)
  } else {
    a<-compare_means(splicing_antigenicity~gene,data=gene_metric_av_gene_df)
    a$cancer <- "all"
    a$gene <- gene
    a$mut_ov_wt <- mean(gene_metric_av_gene_df$splicing_antigenicity[gene_metric_av_gene_df$gene=="MUT"],na.rm=T)/mean(gene_metric_av_gene_df$splicing_antigenicity[gene_metric_av_gene_df$gene=="WT"],na.rm=T)
    a$WT_NUM <- length(gene_metric_av_gene_df$splicing_antigenicity[gene_metric_av_gene_df$gene=="WT"])
    a$MUT_NUM <- length(gene_metric_av_gene_df$splicing_antigenicity[gene_metric_av_gene_df$gene=="MUT"])
    a<-data.frame(a)
  }
  if (j == 1){
    all_comps_all_samples <- a
  } else {
    all_comps_all_samples <- rbind(all_comps_all_samples,a)
  }
}

colnames(splice_vals)<-c("cancer","MUT_ov_WT","BH_pvals","MUT_COUNT","WT_COUNT")

ggplot(splice_vals,aes(x=log2(MUT_ov_WT),y=-log10(BH_pvals),color=sprintf("%s:%d/%d",cancer,MUT_COUNT,WT_COUNT),label=cancer))+
  geom_point()+
  geom_text_repel(direction="y")+
  geom_hline(yintercept=-log10(0.05))+
  labs(color="cancer: MUT_count/WT_count")+
  xlab("log2(mut. sample SA / WT sample SA)")


```

# Citations

1. Immunity. Volume 48 p1-19, 17 April 2018 10.1016/j.immuni.2018.03.023
2. Seiler, Michael et al. “Somatic Mutational Landscape of Splicing Factor Genes and Their Functional Consequences across 33 Cancer Types.” Cell reports vol. 23,1 (2018): 282-296.e4. doi:10.1016/j.celrep.2018.01.088