---
title: "analysis_Theron_08262021"
author: "Theron Palmer"
date: "8/26/2021"
output:
  html_document:
    toc: yes
    toc_float: yes
---


# Loading in necessary libraries

```{r,echo=FALSE}

library(dplyr)
library(knitr)
library(ggplot2)
library(TCGAutils)
library(maftools)
library(ggpubr)
library(stringr)
library(readxl)
library(ComplexHeatmap)
library(recount3)
library(DT)

```

The TCGA MAF summary file

```{r}

maf_file <- "/media/theron/My_Passport/TCGA_junctions/maf_summary.txt"
mc3_maf = read.table(maf_file,header=T)
mc3_maf$Tumor_Sample_ID <- vapply(TCGAbarcode(mc3_maf$Tumor_Sample_Barcode,sample=T),
                                  function(val){substr(val,1,nchar(val)-1)},
                                  character(1))
rownames(mc3_maf) <- mc3_maf$Tumor_Sample_Barcode
mc3_maf$participant_ID <- TCGAbarcode(mc3_maf$Tumor_Sample_Barcode,participant=T)

```


```{r}

mut_sig_perc <- readRDS("/media/theron/My_Passport/TCGA_junctions/TCGA_cancers/mut_sig_percentages.rds")
mut_sig_perc$sample_ID<-vapply(TCGAbarcode(rownames(mut_sig_perc),sample=T),function(sample){
  substr(sample,1,nchar(sample)-1)
},character(1))
apobec <- c("T[C>T]A","T[C>T]T","T[C>G]A","T[C>G]T")

```


# Accumulating mutation data per sample only annotated

## junc summary mean

```{r}

junc_rse_file <- "/media/theron/My_Passport/TCGA_junctions/TCGA_cancers/CHOL/juncrse.rds"
junc_rse <- readRDS(junc_rse_file)
junc_metadata <- as.data.frame(junc_rse@colData@listData)
junc_rse_cols <- colnames(junc_metadata)

tumor_data_file <- "/media/theron/My_Passport/TCGA_junctions/TCGA_cancers/filenames.txt"
tumor_data <- read.table(tumor_data_file)
cancers <- basename(tumor_data$V1)
# TMB<-list()

cluster_metrics_tum <- data.frame(cancers)

TMB_all <- data.frame(cancers)
TMB_all$av_cor <- NA
TMB_all$av_pval <- NA
TMB_all$med_cor <- NA
TMB_all$med_pval <- NA
TMB_all$max_cor <- NA
TMB_all$max_pval <- NA
rownames(TMB_all) <- cancers

for (i in seq(nrow(tumor_data))){
  print(sprintf("%d out of %d",i,nrow(tumor_data)))
  tumor_dir <- tumor_data[i,]
  cancer <- basename(tumor_dir)
  print(cancer)

  tumor_meta_file <- sprintf("%s/%s_metadata.txt",tumor_dir,cancer)
  tumor_meta <- read.table(tumor_meta_file,quote="",sep="\t")
  tumor_meta$participant_ID <- TCGAbarcode(tumor_meta[,4],participant=T)
  tumor_meta$nbases<-tumor_meta[,ncol(tumor_meta)-9]
  mc3_maf_small<-subset(mc3_maf,participant_ID %in% tumor_meta$participant_ID)
  mc3_maf_small <- mc3_maf_small[complete.cases(mc3_maf_small),]
  mc3_maf_small$type <- vapply(rownames(mc3_maf_small),function(barcode){
    type<-as.numeric(substr(strsplit(barcode,"-")[[1]][4],1,2))
    if (type <= 9){
      return("T")
    } else if (type > 9 & type <= 19){
      return ("N")
    } else {
      return ("C")
    }
  },character(1))
  mc3_maf_small$TMB<-log10(mc3_maf_small$total+1)
  mc3_maf_small <- mc3_maf_small %>% dplyr::filter(type == "T")

  tumor_geno_file <- sprintf("%s/%s_genotypes.txt",tumor_dir,cancer)
  tumor_geno <- read.table(tumor_geno_file,header=T)
  splice_mut_file <- sprintf("%s/%s_splice_dat_clusters_filt_ann_mean.rds",tumor_dir,cancer)
  splice_mut_data <- readRDS(splice_mut_file)
  colnames(splice_mut_data) <- vapply(colnames(splice_mut_data),function(col_name){
    col_name<-str_replace(col_name,"X","")
    col_name <- str_replace_all(col_name,"[.]","-")
    tumor_geno$sample_id[which(tumor_geno$external_id == col_name)[1]]
  },character(1))
  
  mc3_maf_small <- mc3_maf_small %>% dplyr::filter(Tumor_Sample_ID %in% colnames(splice_mut_data))
  splice_mut_data<-splice_mut_data[,mc3_maf_small$Tumor_Sample_ID]
  splice_mut_per_sample<-data.frame(colnames(splice_mut_data))
  splice_mut_per_sample$av <- apply(splice_mut_data,2,mean)
  splice_mut_per_sample$med <- apply(splice_mut_data,2,median)
  splice_mut_per_sample$max <- apply(splice_mut_data,2,max)
  splice_mut_per_sample$TMB <- mc3_maf_small$TMB
  splice_mut_per_sample$cancer <- cancer
  colnames(splice_mut_per_sample) <- c("sample","splice_mut_av","splice_mut_med","splice_mut_max","TMB","cancer")
  specific_mut_sig_perc <- data.frame(t(vapply(splice_mut_per_sample$sample,function(samp){
    a<-which(mut_sig_perc$sample_ID == samp)
    if (length(a)==0){
      return(rep(-1,ncol(mut_sig_perc)-1))
    } else {
      return(as.numeric(mut_sig_perc[a,seq(96)]))
    }
  },numeric(ncol(mut_sig_perc)-1))))
  colnames(specific_mut_sig_perc)<-colnames(mut_sig_perc)[seq(96)]
  
  cluster_metrics_tum[cluster_metrics_tum$cancers==cancer,"splice_imm_med"] <- median(splice_mut_per_sample$splice_mut_med[splice_mut_per_sample$splice_mut_med>0])
  cluster_metrics_tum[cluster_metrics_tum$cancers==cancer,"splice_imm_av"] <- mean(splice_mut_per_sample$splice_mut_av)
  cluster_metrics_tum[cluster_metrics_tum$cancers==cancer,"splice_imm_max"] <- mean(splice_mut_per_sample$splice_mut_max)
  cluster_metrics_tum[cluster_metrics_tum$cancers==cancer,"TMB"] <- median(splice_mut_per_sample$TMB)

  if (i == 1){
    splice_mut_per_sample_all <- splice_mut_per_sample
    specific_mut_sig_perc_all <- specific_mut_sig_perc
  } else {
    splice_mut_per_sample_all <- rbind(splice_mut_per_sample_all,splice_mut_per_sample)
    specific_mut_sig_perc_all <- rbind(specific_mut_sig_perc_all,specific_mut_sig_perc)
  }
  
  
  splice_mut_data_mat<-as.matrix(log10(splice_mut_data+1))
  colnames(splice_mut_data_mat)<-colnames(splice_mut_data)
  
  # print(Heatmap(splice_mut_data_mat,
  #         top_annotation = HeatmapAnnotation(TMB=anno_barplot(mc3_maf_small$TMB)),
  #         show_row_names=F,
  #         show_column_names = F,
  #         cluster_rows=T,
  #         cluster_columns=T))
  
  
  a<-cor.test(splice_mut_per_sample$splice_mut_av,splice_mut_per_sample$TMB,method="pearson")
  TMB_all[cancer,"av_cor"]<-a$estimate
  TMB_all[cancer,"av_pval"]<-a$p.value
  
  a<-cor.test(splice_mut_per_sample$splice_mut_med,splice_mut_per_sample$TMB,method="pearson")
  TMB_all[cancer,"med_cor"]<-a$estimate
  TMB_all[cancer,"med_pval"]<-a$p.value
  
  a<-cor.test(splice_mut_per_sample$splice_mut_max,splice_mut_per_sample$TMB,method="pearson")
  TMB_all[cancer,"max_cor"]<-a$estimate
  TMB_all[cancer,"max_pval"]<-a$p.value

  # print(ggplot(splice_mut_per_sample,aes(x=splice_mut_av,y=TMB))+
  #   geom_point()+
  #   stat_cor(method = "pearson")+
  #   geom_smooth(method="lm")+
  #   labs(title=sprintf(cancer)))
  # 
  # print(ggplot(splice_mut_per_sample,aes(x=splice_mut_med,y=TMB))+
  #   geom_point()+
  #   stat_cor(method = "pearson")+
  #   geom_smooth(method="lm")+
  #   labs(title=sprintf(cancer)))
  # 
  # print(ggplot(splice_mut_per_sample,aes(x=splice_mut_max,y=TMB))+
  #   geom_point()+
  #   stat_cor(method = "pearson")+
  #   geom_smooth(method="lm")+
  #   labs(title=sprintf(cancer)))
  
  specific_mut_sig_perc_cor_pval <- data.frame(t(vapply(seq(ncol(mut_sig_perc)-1),function(col_val){
    ret_val <- c()
    mut_col <- as.numeric(specific_mut_sig_perc[,col_val])
    aa<-data.frame(splice_mut_per_sample$splice_mut_av,mut_col)
    colnames(aa)<-c("splice_mut_av","mut_col")
    aa<-aa %>% dplyr::filter(mut_col >= 0)
    a<-cor.test(aa$splice_mut_av,aa$mut_col,method="pearson")
    ret_val <- c(ret_val,a$estimate,a$p.value)
    
    aa<-data.frame(splice_mut_per_sample$splice_mut_med,mut_col)
    colnames(aa)<-c("splice_mut_med","mut_col")
    aa<-aa %>% dplyr::filter(mut_col >= 0)
    a<-cor.test(splice_mut_per_sample$splice_mut_med,mut_col,method="pearson")
    ret_val <- c(ret_val,a$estimate,a$p.value)
    
    aa<-data.frame(splice_mut_per_sample$splice_mut_max,mut_col)
    colnames(aa)<-c("splice_mut_max","mut_col")
    aa<-aa %>% dplyr::filter(mut_col >= 0)
    a<-cor.test(splice_mut_per_sample$splice_mut_max,mut_col,method="pearson")
    ret_val <- c(ret_val,a$estimate,a$p.value)
  },numeric(6))))
  colnames(specific_mut_sig_perc_cor_pval)<-c("av_cor","av_pval","med_cor","med_pval","max_cor","max_pval")
  rownames(specific_mut_sig_perc_cor_pval) <- colnames(specific_mut_sig_perc)
  
  # print(ggplot(splice_mut_per_sample,aes(x=splice_mut_av,y=apobec_prop))+
  #   geom_point()+
  #   stat_cor(method = "pearson")+
  #   geom_smooth(method="lm")+
  #   labs(title=sprintf(cancer)))
  # 
  # print(ggplot(splice_mut_per_sample,aes(x=splice_mut_med,y=apobec_prop))+
  #   geom_point()+
  #   stat_cor(method = "pearson")+
  #   geom_smooth(method="lm")+
  #   labs(title=sprintf(cancer)))
  # 
  # print(ggplot(splice_mut_per_sample,aes(x=splice_mut_max,y=apobec_prop))+
  #   geom_point()+
  #   stat_cor(method = "pearson")+
  #   geom_smooth(method="lm")+
  #   labs(title=sprintf(cancer)))
  
  # print(datatable(specific_mut_sig_perc_cor_pval,caption=sprintf("%s: splicemut vs mutation type",cancer)))
  specific_mut_sig_perc_cor_pval <- specific_mut_sig_perc_cor_pval %>% dplyr::filter(av_pval <= 0.05 | med_pval <= 0.05 | max_pval <= 0.05)
  rownames(specific_mut_sig_perc_cor_pval) <- vapply(rownames(specific_mut_sig_perc_cor_pval),function(rname){
    if (rname %in% apobec){
      return("abobec")
    } else {
      return(rname)
    }
  },character(1))
  specific_mut_sig_perc_cor <- specific_mut_sig_perc_cor_pval[,c("av_cor","med_cor","max_cor")]
  specific_mut_sig_perc_pval <- specific_mut_sig_perc_cor_pval[,c("av_pval","med_pval","max_pval")]
  
  if (nrow(specific_mut_sig_perc_cor_pval) > 0){
    print(Heatmap(as.matrix(specific_mut_sig_perc_cor), cell_fun = function(j, i, x, y, w, h, fill) {
      if(specific_mut_sig_perc_pval[i, j] < 0.005) {
          grid.text("**", x, y)
      } else if(specific_mut_sig_perc_pval[i, j] < 0.05) {
          grid.text("*", x, y)
      }
    },name=sprintf("%s: splicemut vs mutation type",cancer),cluster_columns=F))
  } else {
    print(sprintf("%s: no significant mutation trends",cancer))
  }
}

datatable(TMB_all, caption = "TMB")

specific_mut_sig_perc_cor_pval_all <- data.frame(t(vapply(seq(ncol(mut_sig_perc)-1),function(col_val){
  ret_val <- c()
  mut_col <- as.numeric(specific_mut_sig_perc_all[,col_val])
  aa<-data.frame(splice_mut_per_sample_all$splice_mut_av,mut_col)
  colnames(aa)<-c("splice_mut_av","mut_col")
  aa<-aa %>% dplyr::filter(mut_col >= 0)
  a<-cor.test(aa$splice_mut_av,aa$mut_col,method="pearson")
  ret_val <- c(ret_val,a$estimate,a$p.value)
  
  aa<-data.frame(splice_mut_per_sample_all$splice_mut_med,mut_col)
  colnames(aa)<-c("splice_mut_med","mut_col")
  aa<-aa %>% dplyr::filter(mut_col >= 0)
  a<-cor.test(splice_mut_per_sample_all$splice_mut_med,mut_col,method="pearson")
  ret_val <- c(ret_val,a$estimate,a$p.value)
  
  aa<-data.frame(splice_mut_per_sample_all$splice_mut_max,mut_col)
  colnames(aa)<-c("splice_mut_max","mut_col")
  aa<-aa %>% dplyr::filter(mut_col >= 0)
  a<-cor.test(splice_mut_per_sample_all$splice_mut_max,mut_col,method="pearson")
  ret_val <- c(ret_val,a$estimate,a$p.value)
},numeric(6))))
colnames(specific_mut_sig_perc_cor_pval_all)<-c("av_cor","av_pval","med_cor","med_pval","max_cor","max_pval")
rownames(specific_mut_sig_perc_cor_pval_all) <- colnames(specific_mut_sig_perc_all)

specific_mut_sig_perc_cor_pval_all <- specific_mut_sig_perc_cor_pval_all %>% dplyr::filter(av_pval <= 0.05 | med_pval <= 0.05 | max_pval <= 0.05)
rownames(specific_mut_sig_perc_cor_pval_all) <- vapply(rownames(specific_mut_sig_perc_cor_pval_all),function(rname){
  if (rname %in% apobec){
    return(sprintf("abobec_%d",which(apobec==rname)))
  } else {
    return(rname)
  }
},character(1))
specific_mut_sig_perc_cor_all <- specific_mut_sig_perc_cor_pval_all[,c("av_cor","med_cor","max_cor")]
specific_mut_sig_perc_pval_all <- specific_mut_sig_perc_cor_pval_all[,c("av_pval","med_pval","max_pval")]

Heatmap(as.matrix(specific_mut_sig_perc_cor_all), cell_fun = function(j, i, x, y, w, h, fill) {
  if(specific_mut_sig_perc_pval_all[i, j] < 0.005) {
      grid.text("**", x, y)
  } else if(specific_mut_sig_perc_pval_all[i, j] < 0.05) {
      grid.text("*", x, y)
  }
},name="all samples: splicemut vs mutation type",cluster_columns=F)

ggplot(cluster_metrics_tum,aes(x=log10(splice_imm_med+1),y=TMB,label=cancers))+
  geom_text()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  xlab("splicing antigenicity median")

ggplot(cluster_metrics_tum,aes(x=log10(splice_imm_av+1),y=TMB,label=cancers))+
  geom_text()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  xlab("splicing antigenicity average")

ggplot(cluster_metrics_tum,aes(x=log10(splice_imm_max+1),y=TMB,label=cancers))+
  geom_text()+
  stat_cor(method = "pearson")+
  xlab("splicing antigenicity max")

ggplot(splice_mut_per_sample_all,aes(x=splice_mut_av,y=TMB))+
  geom_point()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  geom_smooth(method="lm")+
    labs("All Samples")

ggplot(splice_mut_per_sample_all,aes(x=splice_mut_med,y=TMB))+
  geom_point()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  geom_smooth(method="lm")+
    labs("All Samples")

ggplot(splice_mut_per_sample_all,aes(x=log10(splice_mut_max),y=TMB))+
  geom_point()+
  stat_cor(method = "pearson")+
  geom_smooth(method="lm")+
    labs("All Samples")


```

# Accumulating mutation data per sample annotated and unannotated

```{r}

junc_rse_file <- "/media/theron/My_Passport/TCGA_junctions/TCGA_cancers/CHOL/juncrse.rds"
junc_rse <- readRDS(junc_rse_file)
junc_metadata <- as.data.frame(junc_rse@colData@listData)
junc_rse_cols <- colnames(junc_metadata)

tumor_data_file <- "/media/theron/My_Passport/TCGA_junctions/TCGA_cancers/filenames.txt"
tumor_data <- read.table(tumor_data_file)
cancers <- basename(tumor_data$V1)
# TMB<-list()

cluster_metrics_tum <- data.frame(cancers)

TMB_all <- data.frame(cancers)
TMB_all$av_cor <- NA
TMB_all$av_pval <- NA
TMB_all$med_cor <- NA
TMB_all$med_pval <- NA
TMB_all$max_cor <- NA
TMB_all$max_pval <- NA
rownames(TMB_all) <- cancers

for (i in seq(nrow(tumor_data))){
  print(sprintf("%d out of %d",i,nrow(tumor_data)))
  tumor_dir <- tumor_data[i,]
  cancer <- basename(tumor_dir)
  print(cancer)

  tumor_meta_file <- sprintf("%s/%s_metadata.txt",tumor_dir,cancer)
  tumor_meta <- read.table(tumor_meta_file,quote="",sep="\t")
  tumor_meta$participant_ID <- TCGAbarcode(tumor_meta[,4],participant=T)
  tumor_meta$nbases<-tumor_meta[,ncol(tumor_meta)-9]
  mc3_maf_small<-subset(mc3_maf,participant_ID %in% tumor_meta$participant_ID)
  mc3_maf_small <- mc3_maf_small[complete.cases(mc3_maf_small),]
  mc3_maf_small$type <- vapply(rownames(mc3_maf_small),function(barcode){
    type<-as.numeric(substr(strsplit(barcode,"-")[[1]][4],1,2))
    if (type <= 9){
      return("T")
    } else if (type > 9 & type <= 19){
      return ("N")
    } else {
      return ("C")
    }
  },character(1))
  mc3_maf_small$TMB<-log10(mc3_maf_small$total+1)
  mc3_maf_small <- mc3_maf_small %>% dplyr::filter(type == "T")

  tumor_geno_file <- sprintf("%s/%s_genotypes.txt",tumor_dir,cancer)
  tumor_geno <- read.table(tumor_geno_file,header=T)
  splice_mut_file <- sprintf("%s/%s_splice_dat_clusters_filt_mean.rds",tumor_dir,cancer)
  splice_mut_data <- readRDS(splice_mut_file)
  colnames(splice_mut_data) <- vapply(colnames(splice_mut_data),function(col_name){
    col_name<-str_replace(col_name,"X","")
    col_name <- str_replace_all(col_name,"[.]","-")
    tumor_geno$sample_id[which(tumor_geno$external_id == col_name)[1]]
  },character(1))
  
  mc3_maf_small <- mc3_maf_small %>% dplyr::filter(Tumor_Sample_ID %in% colnames(splice_mut_data))
  splice_mut_data<-splice_mut_data[,mc3_maf_small$Tumor_Sample_ID]
  splice_mut_per_sample<-data.frame(colnames(splice_mut_data))
  splice_mut_per_sample$av <- apply(splice_mut_data,2,mean)
  splice_mut_per_sample$med <- apply(splice_mut_data,2,median)
  splice_mut_per_sample$max <- apply(splice_mut_data,2,max)
  splice_mut_per_sample$TMB <- mc3_maf_small$TMB
  splice_mut_per_sample$cancer <- cancer
  colnames(splice_mut_per_sample) <- c("sample","splice_mut_av","splice_mut_med","splice_mut_max","TMB","cancer")
  specific_mut_sig_perc <- data.frame(t(vapply(splice_mut_per_sample$sample,function(samp){
    a<-which(mut_sig_perc$sample_ID == samp)
    if (length(a)==0){
      return(rep(-1,ncol(mut_sig_perc)-1))
    } else {
      return(as.numeric(mut_sig_perc[a,seq(96)]))
    }
  },numeric(ncol(mut_sig_perc)-1))))
  colnames(specific_mut_sig_perc)<-colnames(mut_sig_perc)[seq(96)]
  
  cluster_metrics_tum[cluster_metrics_tum$cancers==cancer,"splice_imm_med"] <- median(splice_mut_per_sample$splice_mut_med[splice_mut_per_sample$splice_mut_med>0])
  cluster_metrics_tum[cluster_metrics_tum$cancers==cancer,"splice_imm_av"] <- mean(splice_mut_per_sample$splice_mut_av)
  cluster_metrics_tum[cluster_metrics_tum$cancers==cancer,"splice_imm_max"] <- mean(splice_mut_per_sample$splice_mut_max)
  cluster_metrics_tum[cluster_metrics_tum$cancers==cancer,"TMB"] <- median(splice_mut_per_sample$TMB)

  if (i == 1){
    splice_mut_per_sample_all <- splice_mut_per_sample
    specific_mut_sig_perc_all <- specific_mut_sig_perc
  } else {
    splice_mut_per_sample_all <- rbind(splice_mut_per_sample_all,splice_mut_per_sample)
    specific_mut_sig_perc_all <- rbind(specific_mut_sig_perc_all,specific_mut_sig_perc)
  }
  
  
  splice_mut_data_mat<-as.matrix(log10(splice_mut_data+1))
  colnames(splice_mut_data_mat)<-colnames(splice_mut_data)
  
  # print(Heatmap(splice_mut_data_mat,
  #         top_annotation = HeatmapAnnotation(TMB=anno_barplot(mc3_maf_small$TMB)),
  #         show_row_names=F,
  #         show_column_names = F,
  #         cluster_rows=T,
  #         cluster_columns=T))
  
  
  a<-cor.test(splice_mut_per_sample$splice_mut_av,splice_mut_per_sample$TMB,method="pearson")
  TMB_all[cancer,"av_cor"]<-a$estimate
  TMB_all[cancer,"av_pval"]<-a$p.value
  
  a<-cor.test(splice_mut_per_sample$splice_mut_med,splice_mut_per_sample$TMB,method="pearson")
  TMB_all[cancer,"med_cor"]<-a$estimate
  TMB_all[cancer,"med_pval"]<-a$p.value
  
  a<-cor.test(splice_mut_per_sample$splice_mut_max,splice_mut_per_sample$TMB,method="pearson")
  TMB_all[cancer,"max_cor"]<-a$estimate
  TMB_all[cancer,"max_pval"]<-a$p.value

  # print(ggplot(splice_mut_per_sample,aes(x=splice_mut_av,y=TMB))+
  #   geom_point()+
  #   stat_cor(method = "pearson")+
  #   geom_smooth(method="lm")+
  #   labs(title=sprintf(cancer)))
  # 
  # print(ggplot(splice_mut_per_sample,aes(x=splice_mut_med,y=TMB))+
  #   geom_point()+
  #   stat_cor(method = "pearson")+
  #   geom_smooth(method="lm")+
  #   labs(title=sprintf(cancer)))
  # 
  # print(ggplot(splice_mut_per_sample,aes(x=splice_mut_max,y=TMB))+
  #   geom_point()+
  #   stat_cor(method = "pearson")+
  #   geom_smooth(method="lm")+
  #   labs(title=sprintf(cancer)))
  
  specific_mut_sig_perc_cor_pval <- data.frame(t(vapply(seq(ncol(mut_sig_perc)-1),function(col_val){
    ret_val <- c()
    mut_col <- as.numeric(specific_mut_sig_perc[,col_val])
    aa<-data.frame(splice_mut_per_sample$splice_mut_av,mut_col)
    colnames(aa)<-c("splice_mut_av","mut_col")
    aa<-aa %>% dplyr::filter(mut_col >= 0)
    a<-cor.test(aa$splice_mut_av,aa$mut_col,method="pearson")
    ret_val <- c(ret_val,a$estimate,a$p.value)
    
    aa<-data.frame(splice_mut_per_sample$splice_mut_med,mut_col)
    colnames(aa)<-c("splice_mut_med","mut_col")
    aa<-aa %>% dplyr::filter(mut_col >= 0)
    a<-cor.test(splice_mut_per_sample$splice_mut_med,mut_col,method="pearson")
    ret_val <- c(ret_val,a$estimate,a$p.value)
    
    aa<-data.frame(splice_mut_per_sample$splice_mut_max,mut_col)
    colnames(aa)<-c("splice_mut_max","mut_col")
    aa<-aa %>% dplyr::filter(mut_col >= 0)
    a<-cor.test(splice_mut_per_sample$splice_mut_max,mut_col,method="pearson")
    ret_val <- c(ret_val,a$estimate,a$p.value)
  },numeric(6))))
  colnames(specific_mut_sig_perc_cor_pval)<-c("av_cor","av_pval","med_cor","med_pval","max_cor","max_pval")
  rownames(specific_mut_sig_perc_cor_pval) <- colnames(specific_mut_sig_perc)
  
  # print(ggplot(splice_mut_per_sample,aes(x=splice_mut_av,y=apobec_prop))+
  #   geom_point()+
  #   stat_cor(method = "pearson")+
  #   geom_smooth(method="lm")+
  #   labs(title=sprintf(cancer)))
  # 
  # print(ggplot(splice_mut_per_sample,aes(x=splice_mut_med,y=apobec_prop))+
  #   geom_point()+
  #   stat_cor(method = "pearson")+
  #   geom_smooth(method="lm")+
  #   labs(title=sprintf(cancer)))
  # 
  # print(ggplot(splice_mut_per_sample,aes(x=splice_mut_max,y=apobec_prop))+
  #   geom_point()+
  #   stat_cor(method = "pearson")+
  #   geom_smooth(method="lm")+
  #   labs(title=sprintf(cancer)))
  
  # print(datatable(specific_mut_sig_perc_cor_pval,caption=sprintf("%s: splicemut vs mutation type",cancer)))
  specific_mut_sig_perc_cor_pval <- specific_mut_sig_perc_cor_pval %>% dplyr::filter(av_pval <= 0.05 | med_pval <= 0.05 | max_pval <= 0.05)
  rownames(specific_mut_sig_perc_cor_pval) <- vapply(rownames(specific_mut_sig_perc_cor_pval),function(rname){
    if (rname %in% apobec){
      return("abobec")
    } else {
      return(rname)
    }
  },character(1))
  specific_mut_sig_perc_cor <- specific_mut_sig_perc_cor_pval[,c("av_cor","med_cor","max_cor")]
  specific_mut_sig_perc_pval <- specific_mut_sig_perc_cor_pval[,c("av_pval","med_pval","max_pval")]
  
  if (nrow(specific_mut_sig_perc_cor_pval) > 0){
    print(Heatmap(as.matrix(specific_mut_sig_perc_cor), cell_fun = function(j, i, x, y, w, h, fill) {
      if(specific_mut_sig_perc_pval[i, j] < 0.005) {
          grid.text("**", x, y)
      } else if(specific_mut_sig_perc_pval[i, j] < 0.05) {
          grid.text("*", x, y)
      }
    },name=sprintf("%s: splicemut vs mutation type",cancer),cluster_columns=F))
  } else {
    print(sprintf("%s: no significant mutation trends",cancer))
  }
}

datatable(TMB_all, caption = "TMB")

specific_mut_sig_perc_cor_pval_all <- data.frame(t(vapply(seq(ncol(mut_sig_perc)-1),function(col_val){
  ret_val <- c()
  mut_col <- as.numeric(specific_mut_sig_perc_all[,col_val])
  aa<-data.frame(splice_mut_per_sample_all$splice_mut_av,mut_col)
  colnames(aa)<-c("splice_mut_av","mut_col")
  aa<-aa %>% dplyr::filter(mut_col >= 0)
  a<-cor.test(aa$splice_mut_av,aa$mut_col,method="pearson")
  ret_val <- c(ret_val,a$estimate,a$p.value)
  
  aa<-data.frame(splice_mut_per_sample_all$splice_mut_med,mut_col)
  colnames(aa)<-c("splice_mut_med","mut_col")
  aa<-aa %>% dplyr::filter(mut_col >= 0)
  a<-cor.test(splice_mut_per_sample_all$splice_mut_med,mut_col,method="pearson")
  ret_val <- c(ret_val,a$estimate,a$p.value)
  
  aa<-data.frame(splice_mut_per_sample_all$splice_mut_max,mut_col)
  colnames(aa)<-c("splice_mut_max","mut_col")
  aa<-aa %>% dplyr::filter(mut_col >= 0)
  a<-cor.test(splice_mut_per_sample_all$splice_mut_max,mut_col,method="pearson")
  ret_val <- c(ret_val,a$estimate,a$p.value)
},numeric(6))))
colnames(specific_mut_sig_perc_cor_pval_all)<-c("av_cor","av_pval","med_cor","med_pval","max_cor","max_pval")
rownames(specific_mut_sig_perc_cor_pval_all) <- colnames(specific_mut_sig_perc_all)

specific_mut_sig_perc_cor_pval_all <- specific_mut_sig_perc_cor_pval_all %>% dplyr::filter(av_pval <= 0.05 | med_pval <= 0.05 | max_pval <= 0.05)
rownames(specific_mut_sig_perc_cor_pval_all) <- vapply(rownames(specific_mut_sig_perc_cor_pval_all),function(rname){
  if (rname %in% apobec){
    return(sprintf("abobec_%d",which(apobec==rname)))
  } else {
    return(rname)
  }
},character(1))
specific_mut_sig_perc_cor_all <- specific_mut_sig_perc_cor_pval_all[,c("av_cor","med_cor","max_cor")]
specific_mut_sig_perc_pval_all <- specific_mut_sig_perc_cor_pval_all[,c("av_pval","med_pval","max_pval")]

Heatmap(as.matrix(specific_mut_sig_perc_cor_all), cell_fun = function(j, i, x, y, w, h, fill) {
  if(specific_mut_sig_perc_pval_all[i, j] < 0.005) {
      grid.text("**", x, y)
  } else if(specific_mut_sig_perc_pval_all[i, j] < 0.05) {
      grid.text("*", x, y)
  }
},name="all samples: splicemut vs mutation type",cluster_columns=F)

ggplot(cluster_metrics_tum,aes(x=log10(splice_imm_med+1),y=TMB,label=cancers))+
  geom_text()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  xlab("splicing antigenicity median")

ggplot(cluster_metrics_tum,aes(x=log10(splice_imm_av+1),y=TMB,label=cancers))+
  geom_text()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  xlab("splicing antigenicity average")

ggplot(cluster_metrics_tum,aes(x=log10(splice_imm_max+1),y=TMB,label=cancers))+
  geom_text()+
  stat_cor(method = "pearson")+
  xlab("splicing antigenicity max")

ggplot(splice_mut_per_sample_all,aes(x=splice_mut_av,y=TMB))+
  geom_point()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  geom_smooth(method="lm")+
    labs("All Samples")

ggplot(splice_mut_per_sample_all,aes(x=splice_mut_med,y=TMB))+
  geom_point()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  geom_smooth(method="lm")+
    labs("All Samples")

ggplot(splice_mut_per_sample_all,aes(x=log10(splice_mut_max),y=TMB))+
  geom_point()+
  stat_cor(method = "pearson")+
  geom_smooth(method="lm")+
    labs("All Samples")


```

Looking at cibersort data per sample

```{r}

cibersort_file <- "/media/theron/My_Passport/TCGA_junctions/ext_dat/TCGA.Kallisto.fullIDs.cibersort.relative.tsv"
cibersort_data<-read.table(cibersort_file,header=T)
cibersort_data$sample <- vapply(TCGAbarcode(str_replace_all(cibersort_data$SampleID,"[.]","-"),sample=T),
                                function(sample){substr(sample,1,nchar(sample)-1)},character(1))
rownames(splice_mut_per_sample_all)<-splice_mut_per_sample_all$sample
cibersort_data_filt <- cibersort_data %>% dplyr::filter(sample %in% splice_mut_per_sample_all$sample)
cibersort_append<-lapply(cibersort_data_filt$sample,function(ID){
  a<-which(splice_mut_per_sample_all$sample == ID)
  splice_data_TMB <- as.numeric(splice_mut_per_sample_all[a,c("splice_mut_av","splice_mut_med","splice_mut_max","TMB")])
})
cancers<-vapply(cibersort_data_filt$sample,function(ID){
  a<-which(splice_mut_per_sample_all$sample == ID)
  cancer <- splice_mut_per_sample_all[a,"cancer"]
},character(1))

cibersort_append <- data.frame(matrix(unlist(cibersort_append),byrow=T,nrow=nrow(cibersort_data_filt)))
cibersort_append$cancer <- cancers
colnames(cibersort_append)<-c("splice_mut_av","splice_mut_med","splice_mut_max","TMB","cancer")
cibersort_data_filt <- cbind(cibersort_data_filt,cibersort_append)

cibersort_cor <- data.frame(colnames(cibersort_data_filt)[seq(3,24)])
cibersort_cor$TMB<-NA
cibersort_cor$splice_mut_av<-NA
cibersort_cor$splice_mut_med<-NA
cibersort_cor$splice_mut_max<-NA

cibersort_pval <- data.frame(colnames(cibersort_data_filt)[seq(3,24)])
cibersort_pval$TMB<-NA
cibersort_pval$splice_mut_av<-NA
cibersort_pval$splice_mut_med<-NA
cibersort_pval$splice_mut_max<-NA

```

Naive B-cells
```{r}

ggplot(cibersort_data_filt,aes(x=TMB,y=B.cells.naive))+
  geom_point()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  geom_smooth(method="lm")

a<-cor.test(cibersort_data_filt$B.cells.naive,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[1,"TMB"]<-a$estimate
cibersort_pval[1,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$B.cells.naive,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[1,"splice_mut_av"]<-a$estimate
cibersort_pval[1,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$B.cells.naive,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[1,"splice_mut_med"]<-a$estimate
cibersort_pval[1,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$B.cells.naive,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[1,"splice_mut_max"]<-a$estimate
cibersort_pval[1,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$B.cells.memory,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[2,"TMB"]<-a$estimate
cibersort_pval[2,"TMB"]<-a$p.value


a<-cor.test(cibersort_data_filt$B.cells.memory,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[2,"splice_mut_av"]<-a$estimate
cibersort_pval[2,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$B.cells.memory,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[2,"splice_mut_med"]<-a$estimate
cibersort_pval[2,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$B.cells.memory,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[2,"splice_mut_max"]<-a$estimate
cibersort_pval[2,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$Plasma.cells,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[3,"TMB"]<-a$estimate
cibersort_pval[3,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$Plasma.cells,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[3,"splice_mut_av"]<-a$estimate
cibersort_pval[3,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$Plasma.cells,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[3,"splice_mut_med"]<-a$estimate
cibersort_pval[3,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$Plasma.cells,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[3,"splice_mut_max"]<-a$estimate
cibersort_pval[3,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$T.cells.CD8,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[4,"TMB"]<-a$estimate
cibersort_pval[4,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.CD8,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[4,"splice_mut_av"]<-a$estimate
cibersort_pval[4,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.CD8,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[4,"splice_mut_med"]<-a$estimate
cibersort_pval[4,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.CD8,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[4,"splice_mut_max"]<-a$estimate
cibersort_pval[4,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$T.cells.CD4.naive,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[5,"TMB"]<-a$estimate
cibersort_pval[5,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.CD4.naive,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[5,"splice_mut_av"]<-a$estimate
cibersort_pval[5,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.CD4.naive,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[5,"splice_mut_med"]<-a$estimate
cibersort_pval[5,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.CD4.naive,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[5,"splice_mut_max"]<-a$estimate
cibersort_pval[5,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$T.cells.CD4.memory.resting,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[6,"TMB"]<-a$estimate
cibersort_pval[6,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.CD4.memory.resting,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[6,"splice_mut_av"]<-a$estimate
cibersort_pval[6,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.CD4.memory.resting,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[6,"splice_mut_med"]<-a$estimate
cibersort_pval[6,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.CD4.memory.resting,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[6,"splice_mut_max"]<-a$estimate
cibersort_pval[6,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$T.cells.CD4.memory.activated,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[7,"TMB"]<-a$estimate
cibersort_pval[7,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.CD4.memory.activated,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[7,"splice_mut_av"]<-a$estimate
cibersort_pval[7,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.CD4.memory.activated,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[7,"splice_mut_med"]<-a$estimate
cibersort_pval[7,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.CD4.memory.activated,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[7,"splice_mut_max"]<-a$estimate
cibersort_pval[7,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$T.cells.follicular.helper,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[8,"TMB"]<-a$estimate
cibersort_pval[8,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.follicular.helper,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[8,"splice_mut_av"]<-a$estimate
cibersort_pval[8,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.follicular.helper,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[8,"splice_mut_med"]<-a$estimate
cibersort_pval[8,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.follicular.helper,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[8,"splice_mut_max"]<-a$estimate
cibersort_pval[8,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$T.cells.regulatory..Tregs.,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[9,"TMB"]<-a$estimate
cibersort_pval[9,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.regulatory..Tregs.,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[9,"splice_mut_av"]<-a$estimate
cibersort_pval[9,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.regulatory..Tregs.,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[9,"splice_mut_med"]<-a$estimate
cibersort_pval[9,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.regulatory..Tregs.,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[9,"splice_mut_max"]<-a$estimate
cibersort_pval[9,"splice_mut_max"]<-a$p.value

```


```{r}

a<-cor.test(cibersort_data_filt$T.cells.gamma.delta,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[10,"TMB"]<-a$estimate
cibersort_pval[10,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.gamma.delta,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[10,"splice_mut_av"]<-a$estimate
cibersort_pval[10,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.gamma.delta,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[10,"splice_mut_med"]<-a$estimate
cibersort_pval[10,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$T.cells.gamma.delta,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[10,"splice_mut_max"]<-a$estimate
cibersort_pval[10,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$NK.cells.resting,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[11,"TMB"]<-a$estimate
cibersort_pval[11,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$NK.cells.resting,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[11,"splice_mut_av"]<-a$estimate
cibersort_pval[11,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$NK.cells.resting,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[11,"splice_mut_med"]<-a$estimate
cibersort_pval[11,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$NK.cells.resting,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[11,"splice_mut_max"]<-a$estimate
cibersort_pval[11,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$NK.cells.activated,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[12,"TMB"]<-a$estimate
cibersort_pval[12,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$NK.cells.activated,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[12,"splice_mut_av"]<-a$estimate
cibersort_pval[12,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$NK.cells.activated,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[12,"splice_mut_med"]<-a$estimate
cibersort_pval[12,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$NK.cells.activated,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[12,"splice_mut_max"]<-a$estimate
cibersort_pval[12,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$Monocytes,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[13,"TMB"]<-a$estimate
cibersort_pval[13,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$Monocytes,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[13,"splice_mut_av"]<-a$estimate
cibersort_pval[13,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$Monocytes,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[13,"splice_mut_med"]<-a$estimate
cibersort_pval[13,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$Monocytes,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[13,"splice_mut_max"]<-a$estimate
cibersort_pval[13,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$Macrophages.M0,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[14,"TMB"]<-a$estimate
cibersort_pval[14,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$Macrophages.M0,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[14,"splice_mut_av"]<-a$estimate
cibersort_pval[14,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$Macrophages.M0,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[14,"splice_mut_med"]<-a$estimate
cibersort_pval[14,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$Macrophages.M0,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[14,"splice_mut_max"]<-a$estimate
cibersort_pval[14,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$Macrophages.M1,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[15,"TMB"]<-a$estimate
cibersort_pval[15,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$Macrophages.M1,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[15,"splice_mut_av"]<-a$estimate
cibersort_pval[15,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$Macrophages.M1,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[15,"splice_mut_med"]<-a$estimate
cibersort_pval[15,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$Macrophages.M1,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[15,"splice_mut_max"]<-a$estimate
cibersort_pval[15,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$Macrophages.M2,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[16,"TMB"]<-a$estimate
cibersort_pval[16,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$Macrophages.M2,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[16,"splice_mut_av"]<-a$estimate
cibersort_pval[16,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$Macrophages.M2,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[16,"splice_mut_med"]<-a$estimate
cibersort_pval[16,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$Macrophages.M2,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[16,"splice_mut_max"]<-a$estimate
cibersort_pval[16,"splice_mut_max"]<-a$p.value


```

```{r}

a<-cor.test(cibersort_data_filt$Dendritic.cells.resting,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[17,"TMB"]<-a$estimate
cibersort_pval[17,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$Dendritic.cells.resting,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[17,"splice_mut_av"]<-a$estimate
cibersort_pval[17,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$Dendritic.cells.resting,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[17,"splice_mut_med"]<-a$estimate
cibersort_pval[17,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$Dendritic.cells.resting,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[17,"splice_mut_max"]<-a$estimate
cibersort_pval[17,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$Dendritic.cells.activated,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[18,"TMB"]<-a$estimate
cibersort_pval[18,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$Dendritic.cells.activated,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[18,"splice_mut_av"]<-a$estimate
cibersort_pval[18,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$Dendritic.cells.activated,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[18,"splice_mut_med"]<-a$estimate
cibersort_pval[18,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$Dendritic.cells.activated,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[18,"splice_mut_max"]<-a$estimate
cibersort_pval[18,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$Mast.cells.resting,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[19,"TMB"]<-a$estimate
cibersort_pval[19,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$Mast.cells.resting,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[19,"splice_mut_av"]<-a$estimate
cibersort_pval[19,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$Mast.cells.resting,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[19,"splice_mut_med"]<-a$estimate
cibersort_pval[19,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$Mast.cells.resting,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[19,"splice_mut_max"]<-a$estimate
cibersort_pval[19,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$Mast.cells.activated,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[20,"TMB"]<-a$estimate
cibersort_pval[20,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$Mast.cells.activated,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[20,"splice_mut_av"]<-a$estimate
cibersort_pval[20,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$Mast.cells.activated,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[20,"splice_mut_med"]<-a$estimate
cibersort_pval[20,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$Mast.cells.activated,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[20,"splice_mut_max"]<-a$estimate
cibersort_pval[20,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$Eosinophils,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[21,"TMB"]<-a$estimate
cibersort_pval[21,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$Eosinophils,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[21,"splice_mut_av"]<-a$estimate
cibersort_pval[21,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$Eosinophils,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[21,"splice_mut_med"]<-a$estimate
cibersort_pval[21,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$Eosinophils,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[21,"splice_mut_max"]<-a$estimate
cibersort_pval[21,"splice_mut_max"]<-a$p.value

```

```{r}

a<-cor.test(cibersort_data_filt$Neutrophils,cibersort_data_filt$TMB,method="pearson")
cibersort_cor[22,"TMB"]<-a$estimate
cibersort_pval[22,"TMB"]<-a$p.value

a<-cor.test(cibersort_data_filt$Neutrophils,cibersort_data_filt$splice_mut_av,method="pearson")
cibersort_cor[22,"splice_mut_av"]<-a$estimate
cibersort_pval[22,"splice_mut_av"]<-a$p.value

a<-cor.test(cibersort_data_filt$Neutrophils,cibersort_data_filt$splice_mut_med,method="pearson")
cibersort_cor[22,"splice_mut_med"]<-a$estimate
cibersort_pval[22,"splice_mut_med"]<-a$p.value

a<-cor.test(cibersort_data_filt$Neutrophils,cibersort_data_filt$splice_mut_max,method="pearson")
cibersort_cor[22,"splice_mut_max"]<-a$estimate
cibersort_pval[22,"splice_mut_max"]<-a$p.value

```
Cibersort heatmap
```{r}

rownames(cibersort_cor)<-cibersort_cor[,1]
cibersort_cor_filt <- cibersort_cor[,seq(2,ncol(cibersort_cor))]
cibersort_cor_filt$TMB<-as.numeric(cibersort_cor_filt$TMB)
cibersort_cor_filt$splice_mut_av<-as.numeric(cibersort_cor_filt$splice_mut_av)
cibersort_cor_filt$splice_mut_med<-as.numeric(cibersort_cor_filt$splice_mut_med)
cibersort_cor_filt$splice_mut_max<-as.numeric(cibersort_cor_filt$splice_mut_max)

rownames(cibersort_pval)<-cibersort_pval[,1]
cibersort_pval_filt <- cibersort_pval[,seq(2,ncol(cibersort_pval))]
cibersort_pval_filt$TMB<-as.numeric(cibersort_pval_filt$TMB)
cibersort_pval_filt$splice_mut_av<-as.numeric(cibersort_pval_filt$splice_mut_av)
cibersort_pval_filt$splice_mut_med<-as.numeric(cibersort_pval_filt$splice_mut_med)
cibersort_pval_filt$splice_mut_max<-as.numeric(cibersort_pval_filt$splice_mut_max)

Heatmap(cibersort_cor_filt, cell_fun = function(j, i, x, y, w, h, fill) {
    if(cibersort_pval_filt[i, j] < 0.005) {
        grid.text("**", x, y)
    } else if(cibersort_pval_filt[i, j] < 0.05) {
        grid.text("*", x, y)
    }
})

```

# Analyzing Splicing Factor Mutations per cancer type

```{r,eval=F}

splicing_factor_genes<-read_excel("/media/theron/My_Passport/TCGA_junctions/ext_dat/splicing_factor_genes1.xlsx")
splicing_factor_genes<-toupper(splicing_factor_genes$Gene)
write.table(data.frame(splicing_factor_genes),
            file="/media/theron/My_Passport/TCGA_junctions/ext_dat/splicing_factor_genes.txt",
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)
splice_maf<-read.maf("/media/theron/My_Passport/TCGA_junctions/splice_factor.maf")
splice_maf_samp<-getSampleSummary(splice_maf)
splice_maf_samp$Tumor_Sample_ID <- TCGAbarcode(as.character(splice_maf_samp$Tumor_Sample_Barcode),sample=T)
splice_maf_samp$Tumor_Sample_ID <- vapply(splice_maf_samp$Tumor_Sample_ID,function(samp){
  substr(samp,1,nchar(samp)-1)
},character(1))
splice_maf_samp$participant_ID <- TCGAbarcode(as.character(splice_maf_samp$Tumor_Sample_Barcode),participant=T)
rownames(splice_maf_samp)<-splice_maf_samp$Tumor_Sample_Barcode
splice_maf_samp<-data.frame(splice_maf_samp)

```

# Accumulating splicing factor mutation data per sample

```{r,eval=F}

mc3_maf <- splice_maf_samp

rownames(mc3_maf) <- mc3_maf$Tumor_Sample_Barcode

junc_rse_file <- "/media/theron/My_Passport/TCGA_junctions/TCGA_cancers/CHOL/juncrse.rds"
junc_rse <- readRDS(junc_rse_file)
junc_metadata <- as.data.frame(junc_rse@colData@listData)
junc_rse_cols <- colnames(junc_metadata)

tumor_data_file <- "/media/theron/My_Passport/TCGA_junctions/TCGA_cancers/filenames.txt"
tumor_data <- read.table(tumor_data_file)
cancers <- basename(tumor_data$V1)
# TMB<-list()

cluster_metrics_tum <- data.frame(cancers)

for (i in seq(nrow(tumor_data))){
  print(sprintf("%d out of %d",i,nrow(tumor_data)))
  tumor_dir <- tumor_data[i,]
  cancer <- basename(tumor_dir)
  print(cancer)

  tumor_meta_file <- sprintf("%s/%s_metadata.txt",tumor_dir,cancer)
  tumor_meta <- read.table(tumor_meta_file,quote="",sep="\t")
  tumor_meta$participant_ID <- TCGAbarcode(tumor_meta[,4],participant=T)
  tumor_meta$nbases<-tumor_meta[,ncol(tumor_meta)-9]
  mc3_maf_small<-subset(mc3_maf,participant_ID %in% tumor_meta$participant_ID)
  mc3_maf_small <- mc3_maf_small[complete.cases(mc3_maf_small),]
  mc3_maf_small$type <- vapply(rownames(mc3_maf_small),function(barcode){
    type<-as.numeric(substr(strsplit(barcode,"-")[[1]][4],1,2))
    if (type <= 9){
      return("T")
    } else if (type > 9 & type <= 19){
      return ("N")
    } else {
      return ("C")
    }
  },character(1))
  mc3_maf_small$TMB<-log10(mc3_maf_small$total+1)
  mc3_maf_small <- mc3_maf_small %>% dplyr::filter(type == "T")

  tumor_geno_file <- sprintf("%s/%s_genotypes.txt",tumor_dir,cancer)
  tumor_geno <- read.table(tumor_geno_file,header=T)
  splice_mut_file <- sprintf("%s/%s_splice_dat_clusters_filt.rds",tumor_dir,cancer)
  splice_mut_data <- readRDS(splice_mut_file)
  colnames(splice_mut_data) <- vapply(colnames(splice_mut_data),function(col_name){
    col_name<-str_replace(col_name,"X","")
    col_name <- str_replace_all(col_name,"[.]","-")
    tumor_geno$sample_id[which(tumor_geno$external_id == col_name)[1]]
  },character(1))
  
  mc3_maf_small <- mc3_maf_small %>% dplyr::filter(Tumor_Sample_ID %in% colnames(splice_mut_data))
  splice_mut_data<-splice_mut_data[,mc3_maf_small$Tumor_Sample_ID]
  splice_mut_per_sample<-data.frame(colnames(splice_mut_data))
  splice_mut_per_sample$av <- apply(splice_mut_data,2,mean)
  splice_mut_per_sample$med <- apply(splice_mut_data,2,median)
  splice_mut_per_sample$TMB <- mc3_maf_small$TMB  
  
  # print(Heatmap(splice_mut_data_mat,
  #         top_annotation = HeatmapAnnotation(TMB=anno_barplot(mc3_maf_small$TMB)),
  #         show_row_names=F,
  #         show_column_names = F,
  #         cluster_rows=T,
  #         cluster_columns=T))

  colnames(splice_mut_per_sample) <- c("sample","splice_mut_av","splice_mut_med","TMB")
  
  cluster_metrics_tum[cluster_metrics_tum$cancers==cancer,"splice_imm_med"] <- median(splice_mut_per_sample$splice_mut_med[splice_mut_per_sample$splice_mut_med>0])
  cluster_metrics_tum[cluster_metrics_tum$cancers==cancer,"splice_imm_av"] <- mean(splice_mut_per_sample$splice_mut_av)
  cluster_metrics_tum[cluster_metrics_tum$cancers==cancer,"TMB"] <- median(splice_mut_per_sample$TMB)
  if (i == 1){
    splice_mut_per_sample_all <- splice_mut_per_sample
  } else {
    splice_mut_per_sample_all <- rbind(splice_mut_per_sample_all,splice_mut_per_sample)
  }
  
  splice_mut_data_mat<-as.matrix(log10(splice_mut_data+1))
  colnames(splice_mut_data_mat)<-colnames(splice_mut_data)
  
  # print(Heatmap(splice_mut_data_mat,
  #         top_annotation = HeatmapAnnotation(TMB=anno_barplot(mc3_maf_small$TMB)),
  #         show_row_names=F,
  #         show_column_names = F,
  #         cluster_rows=T,
  #         cluster_columns=T))

  print(ggplot(splice_mut_per_sample,aes(x=splice_mut_av,y=TMB))+
    geom_point()+
    stat_cor(method = "pearson")+
    geom_smooth(method="lm")+
    labs(title=sprintf(cancer)))
  
  splice_mut_per_sample <- splice_mut_per_sample %>% dplyr::filter(splice_mut_med>0)
  
  print(ggplot(splice_mut_per_sample,aes(x=splice_mut_med,y=TMB))+
    geom_point()+
    stat_cor(method = "pearson")+
    geom_smooth(method="lm")+
    labs(title=sprintf(cancer)))
}

ggplot(cluster_metrics_tum,aes(x=log10(splice_imm_med+1),y=TMB,label=cancers))+
  geom_text()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  xlab("splicing antigenicity median")

ggplot(cluster_metrics_tum,aes(x=log10(splice_imm_av+1),y=TMB,label=cancers))+
  geom_text()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  xlab("splicing antigenicity average")

ggplot(splice_mut_per_sample_all,aes(x=splice_mut_av,y=TMB))+
  geom_point()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  geom_smooth(method="lm")+
    labs("All Samples")

ggplot(splice_mut_per_sample_all,aes(x=splice_mut_med,y=TMB))+
  geom_point()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  geom_smooth(method="lm")+
    labs("All Samples")

splice_mut_per_sample_all <- splice_mut_per_sample_all %>% dplyr::filter(splice_mut_med>0)

ggplot(splice_mut_per_sample_all,aes(x=splice_mut_med,y=TMB))+
  geom_point()+
  stat_cor(method = "pearson",
           label.x.npc = "center",
           label.y.npc = "top")+
  geom_smooth(method="lm")+
    labs("All Samples, No Zero Median samples")

```

# splice maf heatmaps

```{r,eval=F}

splice_maf_data <- splice_maf@data
splice_maf_data <- splice_maf_data %>% dplyr::filter(Hugo_Symbol %in% splicing_factor_genes)
splice_maf_data <- splice_maf_data[,c("Hugo_Symbol","Variant_Classification","Tumor_Sample_Barcode")]
splice_maf_data$Tumor_Sample_ID <- vapply(TCGAbarcode(as.character(splice_maf_data$Tumor_Sample_Barcode),sample=T),function(val){substr(val,1,nchar(val)-1)},character(1))
splice_maf_data$participant_ID <- TCGAbarcode(as.character(splice_maf_data$Tumor_Sample_Barcode),participant=T)
splice_maf_data$cancer <- NA
splice_maf_data_fill <- splice_maf_data
splice_maf_data_fill$cancer <- as.character(splice_maf_data_fill$cancer)
splice_maf_data_fill[,c("splice_ant_av","splice_ant_med")] <- data.frame(t(vapply(splice_maf_data_fill$Tumor_Sample_ID,function(ID){
  a<-which(splice_mut_per_sample_all$sample == ID)
  if (length(a)==0){
    return(c(0,0))
  } else {
    return(as.numeric(splice_mut_per_sample_all[a,c("splice_mut_av","splice_mut_med")]))
  }
},numeric(2))))

splice_maf_data_fill$cancer <- vapply(splice_maf_data_fill$Tumor_Sample_ID,function(ID){
  a<-which(splice_mut_per_sample_all$sample == ID)
  if (length(a)==0){
    return("None")
  } else {
    return(splice_mut_per_sample_all[a,"cancer"])
  }
},character(1))


junc_rse_file <- "/media/theron/My_Passport/TCGA_junctions/TCGA_cancers/CHOL/juncrse.rds"
junc_rse <- readRDS(junc_rse_file)
junc_metadata <- as.data.frame(junc_rse@colData@listData)
junc_rse_cols <- colnames(junc_metadata)

tumor_data_file <- "/media/theron/My_Passport/TCGA_junctions/TCGA_cancers/filenames.txt"
tumor_data <- read.table(tumor_data_file)
cancers <- basename(tumor_data$V1)
nogos <- c("ESCA","MESO","PAAD","KIRC","GBM")
# TMB<-list()

splice_maf_data_fill <- splice_maf_data_fill %>% dplyr::filter(cancer != "None")

```

# mutation heatmap

```{r,eval=F}

sample_cancer_dat <- unique(splice_maf_data_fill[,c("Tumor_Sample_ID","cancer","splice_ant_av","splice_ant_med")])
a<-lapply(sample_cancer_dat$Tumor_Sample_ID,function(ID){
  splice_maf_data_fill_small <- splice_maf_data_fill %>% dplyr::filter(Tumor_Sample_ID == ID)
  vapply(splicing_factor_genes,function(gene){
    count <- length(which(splice_maf_data_fill_small$Hugo_Symbol == gene))
  },numeric(1))
})
sample_splice_factor_dat <- data.frame(matrix(unlist(a),nrow=nrow(sample_cancer_dat),byrow=T))
rownames(sample_splice_factor_dat) <- sample_cancer_dat$Tumor_Sample_ID
colnames(sample_splice_factor_dat) <- splicing_factor_genes

sample_splice_factor_dat_muts <- sample_splice_factor_dat[which(apply(sample_splice_factor_dat,1,sd) > 0),]
sample_cancer_dat_muts <- sample_cancer_dat[which(apply(sample_splice_factor_dat,1,sd) > 0),]
cancer_order <- order(sample_cancer_dat_muts$cancer)
sample_splice_factor_dat_muts <- sample_splice_factor_dat_muts[cancer_order,]
sample_cancer_dat_muts <- sample_cancer_dat_muts[cancer_order,]

Heatmap(log2(sample_splice_factor_dat_muts+1),
        right_annotation = rowAnnotation(spliceant = anno_barplot(sample_cancer_dat_muts$splice_ant_av)),
        left_annotation = rowAnnotation(cancer = sample_cancer_dat_muts$cancer),
        show_row_names=F,
        show_column_names = F,
        cluster_rows=T,
        cluster_columns=T)
Heatmap(t(scale(t(sample_splice_factor_dat_muts))),
        right_annotation = rowAnnotation(spliceant = anno_barplot(sample_cancer_dat_muts$splice_ant_med)),
        left_annotation = rowAnnotation(cancer = sample_cancer_dat_muts$cancer),
        show_row_names=F,
        show_column_names = F,
        cluster_rows=T,
        cluster_columns=T)

Heatmap(log2(sample_splice_factor_dat_muts+1),
        right_annotation = rowAnnotation(spliceant = anno_barplot(sample_cancer_dat_muts$splice_ant_av)),
        left_annotation = rowAnnotation(cancer = sample_cancer_dat_muts$cancer),
        show_row_names=F,
        show_column_names = F,
        cluster_rows=F,
        cluster_columns=T)

Heatmap(t(scale(t(sample_splice_factor_dat_muts))),
        right_annotation = rowAnnotation(spliceant = anno_barplot(sample_cancer_dat_muts$splice_ant_med)),
        left_annotation = rowAnnotation(cancer = sample_cancer_dat_muts$cancer),
        show_row_names=F,
        show_column_names = F,
        cluster_rows=F,
        cluster_columns=T)


```

# sample_splice_factor_dat_muts summarized

```{r,eval=F}

sample_cancer_dat_muts$sum <- apply(sample_splice_factor_dat_muts,1,sum)
for (i in unique(sample_cancer_dat_muts$cancer)){
  specific_cancer <- sample_cancer_dat_muts %>% dplyr::filter(cancer == i)
  print(ggplot(specific_cancer,aes(x=log10(splice_ant+1),y=log2(sum+1)))+geom_point()+labs(title=i))
}


ggplot(sample_cancer_dat_muts,aes(x=cancer,y=log2(sum+1)))+geom_boxplot()
ggplot(sample_cancer_dat_muts,aes(x=cancer,y=log10(splice_ant+1)))+geom_boxplot()

ggplot(sample_cancer_dat_muts,aes(x=log10(splice_ant+1),y=log10(sum+1)))+
  geom_point()+
  stat_cor(method = "pearson")+
    geom_smooth(method="lm")

```

# Determining per gene metrics for each cancer type

```{r,eval=F}

split_num <- function(vals){
  a<-data.frame(matrix(as.numeric(unlist(str_split(vals,"/"))),byrow=T,nrow=length(vals)))[,1]
  return(a)
}
split_dom <- function(vals){
  a<-data.frame(matrix(as.numeric(unlist(str_split(vals,"/"))),byrow=T,nrow=length(vals)))[,2]
  return(a)
}

eval_clusters <- function(vals){
  
}
  
# splicemutr data needed per cancer type
# per-line counts needed per cancer type
# only keep those proteins that are longer than 9 kmers

tumor_data_file <- "/media/theron/My_Passport/TCGA_junctions/TCGA_cancers/filenames.txt"
tumor_data <- read.table(tumor_data_file)
cancers <- basename(tumor_data$V1)
nogos <- c("ESCA","MESO","PAAD","KIRC","GBM")

for (i in seq(nrow(tumor_data))){
  print(sprintf("%d out of %d",i,nrow(tumor_data)))
  tumor_dir <- tumor_data[i,]
  cancer <- basename(tumor_dir)
  print(cancer)
  if (cancer %in% nogos){next}
  
  splice_dat_file <- sprintf("%s/%s_splicemutr_dat.txt",tumor_dir,cancer)
  splice_dat <- read.table(splice_dat_file,header=T,sep="\t")
  tumor_geno_file <- sprintf("%s/%s_genotypes.txt",tumor_dir,cancer)
  tumor_geno <- read.table(tumor_geno_file,header=T)
  # tumor_geno <- tumor_geno %>% dplyr::filter(type == "T")
  summary_file <- sprintf("%s/summaries.txt",tumor_dir)
  summaries <- read.table(summary_file)
  summaries <- summaries$V1
  summaries<-unname(vapply(summaries,function(summ){
    str_replace(summ,"kmers_summary","persamp_line")
  },character(1)))
  meta_file <- sprintf("%s/%s_metadata.rds",tumor_dir,cancer)
  meta_dat <- readRDS(meta_file)
  rownames(meta_dat) <- meta_dat$external_id
  psi_file <- sprintf("%s/data_perind.counts",tumor_dir)
  psi_dat <- read.table(psi_file,header=T,check.names=F)
  psi_dat <- psi_dat[,c("chrom",tumor_geno$external_id)]
  sample_names <- colnames(psi_dat)[seq(2,ncol(psi_dat))]
  sample_names <- meta_dat[sample_names,"tcga.tcga_barcode"]
  colnames(psi_dat)[seq(2,ncol(psi_dat))] <- sample_names

  for (summ in seq(length(summaries))){
    if (summ == 1){
      summaries_combined <- read.table(summaries[summ],header=F,sep="\t")
      if (length(summaries)>1){
        summaries_combined <- summaries_combined[,seq(ncol(summaries_combined)-2)]
      }
    } else if (summ == length(summaries)){
      summaries_fill <- read.table(summaries[summ],header=F,sep="\t")
      summaries_combined <- cbind(summaries_combined,summaries_fill)
    } else {
      summaries_fill <- read.table(summaries[summ],header=F,sep="\t")
      summaries_fill <- summaries_fill[,seq(ncol(summaries_fill)-2)]
      summaries_combined <- cbind(summaries_combined,summaries_fill)
    }
  }
  
  sample_types <- sprintf("%s_%s",tumor_geno$sample_id,tumor_geno$type)
  sample_types<-c(sample_types,"row","cluster")
  colnames(summaries_combined)<-sample_types
  tumor_cols <- which(str_detect(sample_types,"_T"))
  summaries_combined <- summaries_combined[,c(tumor_cols,length(sample_types)-1,length(sample_types))]

  num <- data.frame(apply(psi_dat[,seq(2,ncol(psi_dat))],2,split_num))
  denom <- data.frame(apply(psi_dat[,seq(2,ncol(psi_dat))],2,split_dom))
  psi <- num/denom
  is.nan.data.frame <- function(x){do.call(cbind, lapply(x, is.nan))}
  psi[is.nan(psi)]<-0
  psi$chrom <- psi_dat$chrom
  colnames(psi)[seq(1,ncol(psi)-1)] <- colnames(psi_dat)[seq(2,ncol(psi_dat))]
  tumor_cols <- which(tumor_geno$type == "T")
  psi <- psi[,c(tumor_cols,ncol(psi))]
  psi_summary<-psi[summaries_combined$row+1,]
  
  summaries_combined_psi <- psi_summary[,seq(1,ncol(psi_summary)-1)]*summaries_combined[,seq(1,ncol(summaries_combined)-2)]
  
  splice_dat_specific <- splice_dat[summaries_combined$row+1,]
  rows_to_keep <- which(!is.na(splice_dat_specific$peptide))
  
  summaries_combined_psi <- summaries_combined_psi[rows_to_keep,]
  summaries_combined_psi[is.na(summaries_combined_psi)]<-0
  splice_dat_specific <- splice_dat_specific[rows_to_keep,]
  rows_to_keep <- !(splice_dat_specific$verdict == "annotated" & splice_dat_specific$modified == "changed")
  summaries_combined_psi <- summaries_combined_psi[rows_to_keep,]
  summaries_combined <- summaries_combined[rows_to_keep,]
  splice_dat_specific <- splice_dat_specific[rows_to_keep,]
  rows_to_keep <- which(splice_dat_specific$deltapsi > 0)
  summaries_combined_psi <- summaries_combined_psi[rows_to_keep,]
  summaries_combined <- summaries_combined[rows_to_keep,]
  splice_dat_specific <- splice_dat_specific[rows_to_keep,]
  
  clusters <- data.frame(table(splice_dat_specific$cluster))
  rownames(clusters)<-clusters$Var1
  clusters$Var1<-as.character(clusters$Var1)
  
  clusters$genes <- vapply(clusters$Var1,function(clu){
    splice_dat_small <- splice_dat_specific %>% dplyr::filter(cluster == clu)
    gene<-paste(unique(splice_dat_small$gene),collapse=":")
  },character(1))
    
  splice_dat_clusters <- data.frame(t(vapply(clusters$Var1,function(clu){
    cluster_rows <- which(splice_dat_specific$cluster == clu)
    summaries_combined_small <- summaries_combined_psi[cluster_rows,]
    apply(summaries_combined_small,2,sum)/clusters[clu,"Freq"]
  },numeric(ncol(summaries_combined_psi)))))
  
  saveRDS(splice_dat_clusters,file=sprintf("%s/%s_splice_dat_clusters.rds",tumor_dir,cancer))
  saveRDS(clusters,file=sprintf("%s/%s_clusters.rds",tumor_dir,cancer))
  
  rm(clusters,
     splice_dat_clusters,
     splice_dat_specific,
     rows_to_keep,
     summaries_combined,
     summaries_combined_psi,
     psi_summary,psi,
     tumor_cols,
     sample_types,
     summaries_fill,
     psi_dat,
     psi_file,
     meta_dat,
     meta_file,
     summaries,
     summary_file,
     tumor_geno,
     tumor_geno_file,
     splice_dat,
     splice_dat_file,
     cancer,
     tumor_dir)
}

```

# Compiling per-gene metrics

```{r,eval=F}

tumor_data_file <- "/media/theron/My_Passport/TCGA_junctions/TCGA_cancers/filenames.txt"
tumor_data <- read.table(tumor_data_file)
cancers <- basename(tumor_data$V1)
nogos <- c("ESCA","MESO","PAAD","KIRC","GBM")

genes <- c()
for (i in seq(nrow(tumor_data))){
  print(sprintf("%d out of %d",i,nrow(tumor_data)))
  tumor_dir <- tumor_data[i,]
  cancer <- basename(tumor_dir)
  print(cancer)
  if (cancer %in% nogos){next}
  
  
  per_gene_data<-readRDS(file=sprintf("%s/%s_per_gene_data.rds",tumor_dir,cancer))
  genes <- unique(c(genes,per_gene_data$genes))
  
}

per_gene_data_tot_binders <- data.frame(genes)
per_gene_data_tot_prop <- data.frame(genes)

rownames(per_gene_data_tot_binders) <- per_gene_data_tot_binders$genes
rownames(per_gene_data_tot_prop) <- per_gene_data_tot_prop$genes

cancers <- c()
for (i in seq(nrow(tumor_data))){
  print(sprintf("%d out of %d",i,nrow(tumor_data)))
  tumor_dir <- tumor_data[i,]
  cancer <- basename(tumor_dir)
  print(cancer)
  if (cancer %in% nogos){next}
  cancers <- c(cancers,cancer)
  
  per_gene_data<-readRDS(file=sprintf("%s/%s_per_gene_data.rds",tumor_dir,cancer))
  rownames(per_gene_data)<-per_gene_data$genes
  per_gene_data_tot_binders[,cancer]<- -1
  per_gene_data_tot_prop[,cancer]<- -1
  per_gene_data_tot_binders[per_gene_data$genes,cancer] <- per_gene_data$median_binders
  per_gene_data_tot_prop[per_gene_data$genes,cancer] <- per_gene_data$ann_prop
  
}
row_ann <- unname(vapply(genes,function(gene){
  if (str_detect(gene,"-")){
    return("Fusion")
  } else {
    return("Single")
  }
},character(1)))
row_ann <- data.frame(row_ann)
rownames(row_ann)<-genes

per_gene_data_tot_binders <- per_gene_data_tot_binders[,seq(2,ncol(per_gene_data_tot_binders))]
per_gene_data_tot_prop <- per_gene_data_tot_prop[,seq(2,ncol(per_gene_data_tot_prop))]

per_gene_comp <- per_gene_data_tot_binders*per_gene_data_tot_prop

Heatmap(log10(per_gene_comp+1),
        right_annotation = rowAnnotation(df=row_ann),
        show_row_names=F,
        show_column_names = T,
        cluster_rows=T,
        cluster_columns=T)

```

