---
title: "analysis_Theron"
author: "Theron Palmer"
date: "8/17/2021"
output:
  html_document:
    toc: yes
    toc_float: yes
---

```{r setup, include=FALSE} 
knitr::opts_chunk$set(warning = FALSE, message = FALSE) 
```


# Libraries used in the analysis

```{r}

library(ggplot2)
library(knitr)
library(dplyr)
library(org.Mm.eg.db)
library(DT)
library(GO.db)
library(KEGGREST)
# library(KEGG.db)
library(fgsea)
library(maftools)
library(pheatmap)
library(ensembldb)
library(stringr)
library(ensembldb)
library(BSgenome.Mmusculus.GENCODE.GRCm38.102)
library(biomaRt)
library(ggpubr)
library(DESeq2)
library(pheatmap)
library(ComplexHeatmap)

```

# Loading in the necessary data

```{r,echo=FALSE}

base_dir <- "/media/theron/My_Passport/Ali_data/HTMBmice_RNAseq_Data/MB01JHU504/MB01JHU504_000_analysis/DESeq2/refseq_genes"
PD1_vs_combo <- read.table(sprintf("%s/%s",base_dir,"PD1_vs_Combo_genes_DESeq2.txt"),header=T)
untreated_vs_combo <- read.table(sprintf("%s/%s",base_dir,"Untreated_vs_Combo_genes_DESeq2.txt"),header=T)
untreated_vs_PD1 <- read.table(sprintf("%s/%s",base_dir,"Untreated_vs_PD1_genes_DESeq2.txt"),header=T)

```

# PD1 vs Combo (PD1/Combo)

## Differential Expression Data 

RSEM was run on each sample. Using the RSEM-quantified gene expression, DESeq2 was run between arms. The fold change order is as follows: A_vs_B => numerator_vs_denominator. The following table is interactive and shows the differential expression results for PD1 vs combination therapy samples. The magnitude log2 fold change is very small towards PD1 and towards Combo. 

```{r,echo=FALSE}

which_sig <- vapply(as.numeric(PD1_vs_combo$PD1_vs_Combo_pvalue),function(val){
  if (is.na(val)){
    return("Insig")
  }
  if (val <= 0.05){
    return("Sig")
  } else {
    return("Insig")
  }},character(1))
PD1_vs_combo$significance_pval <- which_sig

which_sig <- vapply(PD1_vs_combo$PD1_vs_Combo_padj,function(val){
  if (is.na(val)){
    return("Insig")
  }
  if (val <= 0.05){
    return("Sig")
  } else {
    return("Insig")
  }},character(1))
PD1_vs_combo$significance_padj <- which_sig

PD1_vs_combo_table <- PD1_vs_combo[complete.cases(PD1_vs_combo),]
datatable(PD1_vs_combo_table)

```


## Volcano Plots

```{r,echo=F}
ggplot(PD1_vs_combo_table,aes(x=PD1_vs_Combo_log2FoldChange, y=-log10(PD1_vs_Combo_pvalue), color = significance_pval))+
  geom_text(label=PD1_vs_combo_table$gene_symbol)+
  labs(title="unadjusted pvalues")
```


```{r,echo=F}

ggplot(PD1_vs_combo_table,aes(x=PD1_vs_Combo_log2FoldChange, y=-log10(PD1_vs_Combo_padj), color = significance_padj))+
  geom_text(label=PD1_vs_combo_table$gene_symbol)+
  labs(title="BH-adjusted pvalues")

```

## gsea results on GO and KEGG terms

Gene-set enrichment analysis within the GO pathways show enrichment for ribosome pathways, protein synthesis and catalysis, innate and adaptive immune response, B-cell proliferation, an antiviral immune response, interferon production, cellular response to interferon, RNA translation, and RNA and mRNA binding.

```{r,echo=F}

symbol_to_entrez <- as.list(org.Mm.egALIAS2EG)
PD1_vs_combo_table$entrez <- symbol_to_entrez[PD1_vs_combo_table$gene_symbol]

SYMBOLToGO <- mapIds(org.Mm.eg.db,keys=PD1_vs_combo_table$gene_symbol,column='GO',keytype = 'SYMBOL',multiVals = list)
GOToSYMBOL <- sapply(reverseSplit(SYMBOLToGO),unique)
GOToSYMBOL <- GOToSYMBOL[sapply(GOToSYMBOL,length)>5]

SYMBOLToKEGG <- mapIds(org.Mm.eg.db,keys=PD1_vs_combo_table$gene_symbol,column='PATH',keytype = 'SYMBOL',multiVals = list)
KEGGToSYMBOL <- sapply(reverseSplit(SYMBOLToKEGG),unique)
KEGGToSYMBOL <- KEGGToSYMBOL[sapply(KEGGToSYMBOL,length)>5]

diff_dat_filt <- PD1_vs_combo_table %>% dplyr::filter(!is.na(PD1_vs_Combo_log2FoldChange) & !is.na(PD1_vs_Combo_pvalue))
ranks <- diff_dat_filt$PD1_vs_Combo_log2FoldChange*-log10(diff_dat_filt$PD1_vs_Combo_pvalue)
names(ranks)<-diff_dat_filt$gene_symbol
ranks<-ranks[unname(!is.na(ranks))]
ranks_pos<-ranks[ranks>=0]
ranks_neg<-ranks[ranks<0]
ranks_pos<-rank(ranks_pos,ties.method = "random")
ranks_neg<--1*rank(-1*ranks_neg,ties.method = "random")
ranks<-c(ranks_pos,ranks_neg)
# ranks<-rank(ranks,ties.method = "random")
fgseaRes_GO <- fgsea(GOToSYMBOL, ranks, eps=0,scoreType="pos")
fgseaRes_GO<-cbind(sapply(c('ONTOLOGY','TERM','DEFINITION'),
                        function(x){mapIds(GO.db,keys=fgseaRes_GO$pathway,keytype='GOID',column=x)}),fgseaRes_GO)
fgseaRes_KEGG <- fgsea(KEGGToSYMBOL, ranks, eps=0,scoreType="pos")
fgseaRes_KEGG <- cbind(KEGG.Name=unlist(as.list(KEGGPATHID2NAME)[fgseaRes_KEGG$pathway]),
                   fgseaRes_KEGG)

pathways<- readRDS("/media/theron/My_Passport/LM_LMT_LUCIANE_cellranger/references/gene_sets/mouse_hallmark.rds")
pathways_list <- lapply(unique(pathways$gs_name),function(pathway){
  pathway_small <- pathways %>% dplyr::filter(gs_name == pathway)
  return(pathway_small$gene_symbol)
})
names(pathways_list)<-unique(pathways$gs_name)

fgseaRes_HALLMARK <- fgsea(pathways_list, ranks, eps=0)
fgseaRes_HALLMARK <- fgseaRes_HALLMARK %>% dplyr::filter(padj <= 0.05)

```

### GO GSEA

```{r,echo=F}

GO_table <- fgseaRes_GO %>% dplyr::filter(padj <= 0.05)
GO_table<-GO_table[,c("TERM","DEFINITION","padj","NES","size")]
datatable(GO_table)

```

### KEGG GSEA

```{r,echo=F}

KEGG_table <- fgseaRes_KEGG %>% dplyr::filter(padj <= 0.05)
KEGG_table<-KEGG_table[,c("KEGG.Name","padj","NES","size")]
datatable(KEGG_table)

```

### HALLMARK GSEA
```{r,echo=F}

datatable(fgseaRes_HALLMARK[,c(1,3,6,7)])

```


# Untreated vs Combo (Untreated/Combo) 

## Differential Expression Data

```{r,echo=F}

which_sig <- vapply(as.numeric(untreated_vs_combo$Untreated_vs_Combo_pvalue),function(val){
  if (is.na(val)){
    return("Insig")
  }
  if (val <= 0.05){
    return("Sig")
  } else {
    return("Insig")
  }},character(1))
untreated_vs_combo$significance_pval <- which_sig

which_sig <- vapply(untreated_vs_combo$Untreated_vs_Combo_padj,function(val){
  if (is.na(val)){
    return("Insig")
  }
  if (val <= 0.05){
    return("Sig")
  } else {
    return("Insig")
  }},character(1))
untreated_vs_combo$significance_padj <- which_sig

untreated_vs_combo_table <- untreated_vs_combo[complete.cases(untreated_vs_combo),]
datatable(untreated_vs_combo_table)

```

## Volcano Plots

```{r,echo=F}

ggplot(untreated_vs_combo_table,aes(x=Untreated_vs_Combo_log2FoldChange, y=-log10(Untreated_vs_Combo_pvalue), color = significance_pval))+
  geom_text(label=untreated_vs_combo_table$gene_symbol)
ggplot(untreated_vs_combo_table,aes(x=Untreated_vs_Combo_log2FoldChange, y=-log10(Untreated_vs_Combo_padj), color = significance_padj))+
  geom_text(label=untreated_vs_combo_table$gene_symbol)

```

## gsea results on GO and KEGG terms

```{r,echo=F}

symbol_to_entrez <- as.list(org.Mm.egALIAS2EG)
untreated_vs_combo_table$entrez <- symbol_to_entrez[untreated_vs_combo_table$gene_symbol]

SYMBOLToGO <- mapIds(org.Mm.eg.db,keys=untreated_vs_combo_table$gene_symbol,column='GO',keytype = 'SYMBOL',multiVals = list)
GOToSYMBOL <- sapply(reverseSplit(SYMBOLToGO),unique)
GOToSYMBOL <- GOToSYMBOL[sapply(GOToSYMBOL,length)>5]

SYMBOLToKEGG <- mapIds(org.Mm.eg.db,keys=untreated_vs_combo_table$gene_symbol,column='PATH',keytype = 'SYMBOL',multiVals = list)
KEGGToSYMBOL <- sapply(reverseSplit(SYMBOLToKEGG),unique)
KEGGToSYMBOL <- KEGGToSYMBOL[sapply(KEGGToSYMBOL,length)>5]

diff_dat_filt <- untreated_vs_combo_table %>% dplyr::filter(!is.na(Untreated_vs_Combo_log2FoldChange) & !is.na(Untreated_vs_Combo_pvalue))
ranks <- diff_dat_filt$Untreated_vs_Combo_log2FoldChange*-log10(diff_dat_filt$Untreated_vs_Combo_pvalue)
names(ranks)<-diff_dat_filt$gene_symbol
ranks<-ranks[unname(!is.na(ranks))]
ranks_pos<-ranks[ranks>=0]
ranks_neg<-ranks[ranks<0]
ranks_pos<-rank(ranks_pos,ties.method = "random")
ranks_neg<--1*rank(-1*ranks_neg,ties.method = "random")
ranks<-c(ranks_pos,ranks_neg)
fgseaRes_GO <- fgsea(GOToSYMBOL, ranks, eps=0,scoreType="pos")
fgseaRes_GO<-cbind(sapply(c('ONTOLOGY','TERM','DEFINITION'),
                        function(x){mapIds(GO.db,keys=fgseaRes_GO$pathway,keytype='GOID',column=x)}),fgseaRes_GO)
fgseaRes_KEGG <- fgsea(KEGGToSYMBOL, ranks, eps=0,scoreType="pos")
fgseaRes_KEGG <- cbind(KEGG.Name=unlist(as.list(KEGGPATHID2NAME)[fgseaRes_KEGG$pathway]),
                   fgseaRes_KEGG)

pathways_list <- readRDS("/media/theron/My_Passport/LM_LMT_LUCIANE_cellranger/references/gene_sets/pathways_list_ensembl.rds")
fgseaRes_HALLMARK <- fgsea(pathways_list, ranks, eps=0)
fgseaRes_HALLMARK <- fgseaRes_HALLMARK %>% dplyr::filter(padj <= 0.05)

```

### GO GSEA
```{r,echo=F}

GO_table <- fgseaRes_GO %>% dplyr::filter(padj <= 0.05)
GO_table<-GO_table[,c("TERM","DEFINITION","padj","NES","size")]
datatable(GO_table)

```

### KEGG GSEA
```{r,echo=F}

KEGG_table <- fgseaRes_KEGG %>% dplyr::filter(padj <= 0.05)
KEGG_table<-KEGG_table[,c("KEGG.Name","padj","NES","size")]
datatable(KEGG_table)

```

### HALLMARK Pathway GSEA
```{r,echo=F}

datatable(fgseaRes_HALLMARK[,c(1,3,6,7)])

```

# Untreated vs PD1 (Untreated/PD1) 

## Differential Expression Data

```{r,echo=F}

which_sig <- vapply(as.numeric(untreated_vs_PD1$Untreated_vs_PD1_pvalue),function(val){
  if (is.na(val)){
    return("Insig")
  }
  if (val <= 0.05){
    return("Sig")
  } else {
    return("Insig")
  }},character(1))
untreated_vs_PD1$significance_pval <- which_sig

which_sig <- vapply(untreated_vs_PD1$Untreated_vs_PD1_padj,function(val){
  if (is.na(val)){
    return("Insig")
  }
  if (val <= 0.05){
    return("Sig")
  } else {
    return("Insig")
  }},character(1))
untreated_vs_PD1$significance_padj <- which_sig

untreated_vs_PD1_table <- untreated_vs_PD1[complete.cases(untreated_vs_PD1),]

datatable(untreated_vs_PD1_table)

```


## Volcano Plots

```{r,echo=F}

ggplot(untreated_vs_PD1_table,aes(x=Untreated_vs_PD1_log2FoldChange, y=-log10(Untreated_vs_PD1_pvalue), color = significance_pval))+
  geom_text(label=untreated_vs_PD1_table$gene_symbol)
ggplot(untreated_vs_PD1_table,aes(x=Untreated_vs_PD1_log2FoldChange, y=-log10(Untreated_vs_PD1_padj), color = significance_padj))+
  geom_text(label=untreated_vs_PD1_table$gene_symbol)

```


## gsea results on GO,KEGG, and Hallmark Pathway terms

```{r,echo=F}

symbol_to_entrez <- as.list(org.Mm.egALIAS2EG)
untreated_vs_PD1_table$entrez <- symbol_to_entrez[untreated_vs_PD1_table$gene_symbol]

SYMBOLToGO <- mapIds(org.Mm.eg.db,keys=untreated_vs_PD1_table$gene_symbol,column='GO',keytype = 'SYMBOL',multiVals = list)
GOToSYMBOL <- sapply(reverseSplit(SYMBOLToGO),unique)
GOToSYMBOL <- GOToSYMBOL[sapply(GOToSYMBOL,length)>5]

SYMBOLToKEGG <- mapIds(org.Mm.eg.db,keys=untreated_vs_PD1_table$gene_symbol,column='PATH',keytype = 'SYMBOL',multiVals = list)
KEGGToSYMBOL <- sapply(reverseSplit(SYMBOLToKEGG),unique)
KEGGToSYMBOL <- KEGGToSYMBOL[sapply(KEGGToSYMBOL,length)>5]

diff_dat_filt <- untreated_vs_PD1_table %>% dplyr::filter(!is.na(Untreated_vs_PD1_log2FoldChange) & !is.na(Untreated_vs_PD1_pvalue))
ranks <- diff_dat_filt$Untreated_vs_PD1_log2FoldChange*-log10(diff_dat_filt$Untreated_vs_PD1_pvalue)
names(ranks)<-diff_dat_filt$gene_symbol
ranks<-ranks[unname(!is.na(ranks))]
ranks_pos<-ranks[ranks>=0]
ranks_neg<-ranks[ranks<0]
ranks_pos<-rank(ranks_pos,ties.method = "random")
ranks_neg<--1*rank(-1*ranks_neg,ties.method = "random")
ranks<-c(ranks_pos,ranks_neg)
fgseaRes_GO <- fgsea(GOToSYMBOL, ranks, eps=0,scoreType="pos")
fgseaRes_GO<-cbind(sapply(c('ONTOLOGY','TERM','DEFINITION'),
                        function(x){mapIds(GO.db,keys=fgseaRes_GO$pathway,keytype='GOID',column=x)}),fgseaRes_GO)
fgseaRes_KEGG <- fgsea(KEGGToSYMBOL, ranks, eps=0,scoreType="pos")
fgseaRes_KEGG <- cbind(KEGG.Name=unlist(as.list(KEGGPATHID2NAME)[fgseaRes_KEGG$pathway]),
                   fgseaRes_KEGG)

pathways_list <- readRDS("/media/theron/My_Passport/LM_LMT_LUCIANE_cellranger/references/gene_sets/pathways_list_ensembl.rds")
fgseaRes_HALLMARK <- fgsea(pathways_list, ranks, eps=0)
fgseaRes_HALLMARK <- fgseaRes_HALLMARK %>% dplyr::filter(padj <= 0.05)

```

### GO GSEA
```{r,echo=F}

GO_table <- fgseaRes_GO %>% dplyr::filter(padj <= 0.05)
GO_table<-GO_table[,c("TERM","DEFINITION","padj","NES","size")]
datatable(GO_table)

```

### KEGG GSEA
```{r,echo=F}

KEGG_table <- fgseaRes_KEGG %>% dplyr::filter(padj <= 0.05)
KEGG_table<-KEGG_table[,c("KEGG.Name","padj","NES","size")]
datatable(KEGG_table)

```

### HALLMARK Pathway GSEA
```{r,echo=F}

datatable(fgseaRes_HALLMARK[,c(1,3,6,7)])

```


```{r,echo=F,eval=F}

maf_file <- "/media/theron/My_Passport/Ali_data/WES Data for 4T1MIS/MB01JHU503/MB01JHU503_000_analysis/MAF/DNA_4TIMSI_haplotypecaller.maf"
ali_maf = read.maf(maf_file)
ali_gene_summ = getGeneSummary(ali_maf)
rownames(ali_gene_summ) <- ali_gene_summ$Hugo_Symbol

rownames(PD1_vs_combo) <- PD1_vs_combo$gene_symbol
rownames(untreated_vs_combo) <- untreated_vs_combo$gene_symbol
rownames(untreated_vs_PD1) <- untreated_vs_PD1$gene_symbol

ali_gene_summ <- cbind(ali_gene_summ,PD1_vs_combo[ali_gene_summ$Hugo_Symbol,c("PD1_vs_Combo_log2FoldChange","significance_pval","significance_padj")])
ali_gene_summ <- cbind(ali_gene_summ,untreated_vs_combo[ali_gene_summ$Hugo_Symbol,c("Untreated_vs_Combo_log2FoldChange","significance_pval","significance_padj")])
ali_gene_summ <- cbind(ali_gene_summ,untreated_vs_PD1[ali_gene_summ$Hugo_Symbol,c("Untreated_vs_PD1_log2FoldChange","significance_pval","significance_padj")])

cols <- colnames(ali_gene_summ)
cols[seq(15,22)]<-c("pc_significance_pval","pc_significance_padj",
                    "Untreated_vs_Combo_log2FoldChange","uc_significance_pval","uc_significance_padj",
                    "Untreated_vs_PD1_log2FoldChange","up_significance_pval","up_significance_padj")
colnames(ali_gene_summ) <- cols
datatable(ali_gene_summ)

```

# Creating conglomerate gene and junction expression per sample


```{r,eval=F}
## combining and normalizing the gene expression matrix per sample

gene_expression_file <- read.table("/media/theron/My_Passport/Ali_data/HTMBmice_RNAseq_Data/MB01JHU504/MB01JHU504_000_analysis/RSEM/tables/genes/filenames.txt")
for (i in seq(nrow(gene_expression_file))){
  file<-gene_expression_file[i,]
  sample <- str_replace(basename(file),"_rsem_mm10.genes.results","")
  if (i == 1){
    expr_fill <- read.table(file,header=T)
    genes <- expr_fill$gene_id
    gene_expression<-data.frame(genes)
    gene_expression[,sample] <- expr_fill$expected_count
  } else {
    expr_fill <- read.table(file,header=T)
    gene_expression[,sample] <- expr_fill$expected_count
  }
}
rownames(gene_expression)<-gene_expression$genes
gene_expression<-gene_expression[,seq(2,ncol(gene_expression))]
gene_expression <- t(apply(gene_expression,1,round))
gene_expression_vst <- as.data.frame(varianceStabilizingTransformation(gene_expression))
gene_expression_rlog <- as.data.frame(rlog(gene_expression))

saveRDS(gene_expression_vst,file="/media/theron/My_Passport/Ali_data/HTMBmice_RNAseq_Data/MB01JHU504/MB01JHU504_000_analysis/RSEM/tables/genes/gene_expression_vst.rds")
gene_expression <- as.data.frame(gene_expression)
saveRDS("/media/theron/My_Passport/Ali_data/HTMBmice_RNAseq_Data/MB01JHU504/MB01JHU504_000_analysis/RSEM/tables/genes/gene_expression_vst.rds")

```


```{r,eval=F,echo=F}

## combining and normalizing the isoform expression matrix per sample

isoform_expression_file <- read.table("/media/theron/My_Passport/Ali_data/HTMBmice_RNAseq_Data/MB01JHU504/MB01JHU504_000_analysis/RSEM/tables/isoforms/filenames.txt")
for (i in seq(nrow(isoform_expression_file))){
  file<-isoform_expression_file[i,]
  sample <- str_replace(basename(file),"_rsem_mm10.isoforms.results","")
  if (i == 1){
    expr_fill <- read.table(file,header=T)
    transcripts <- expr_fill$transcript_id
    isoform_expression<-data.frame(transcripts)
    isoform_expression[,sample] <- expr_fill$expected_count
  } else {
    expr_fill <- read.table(file,header=T)
    isoform_expression[,sample] <- expr_fill$expected_count
  }
}
rownames(isoform_expression) <- isoform_expression$transcripts
isoform_expression[,seq(2,ncol(isoform_expression))] <- t(apply(isoform_expression[,seq(2,ncol(isoform_expression))],1,round))
isoform_expression_vst <- as.data.frame(varianceStabilizingTransformation(as.matrix(isoform_expression[,seq(2,ncol(isoform_expression))])))
isoform_expression_vst<-cbind(isoform_expression[,1],isoform_expression_vst)
colnames(isoform_expression_vst)<-colnames(isoform_expression)
saveRDS(isoform_expression_vst,file="/media/theron/My_Passport/Ali_data/HTMBmice_RNAseq_Data/MB01JHU504/MB01JHU504_000_analysis/RSEM/tables/genes/isoform_expression_vst.rds")

```

```{bash,eval=F}

./media/theron/My_Passport/scripts/Ali_data/analysis/bash/create_junc_files.sh

```


```{r,echo=F}
# Exploring outlier splice junctions Combo vs PD1 and PD1 vs Combo

## Filtering for significant junctions per sample

C1_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/C1P_pVals.txt"
C1_clusters <- read.table(C1_clusters_file,header=T)
C1_clusters$juncs <- vapply(rownames(C1_clusters),function(val){
  all_info <- strsplit(val,":")[[1]]
  strand <- strsplit(all_info[4],"_")[[1]][3]
  paste(c(all_info[seq(3)],strand),collapse=":")
  },character(1))
C1_clusters_sig <- C1_clusters %>% dplyr::filter(C1 <= 0.05)


C2_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/C2P_pVals.txt"
C2_clusters <- read.table(C2_clusters_file,header=T)
C2_clusters$juncs <- vapply(rownames(C2_clusters),function(val){
  all_info <- strsplit(val,":")[[1]]
  strand <- strsplit(all_info[4],"_")[[1]][3]
  paste(c(all_info[seq(3)],strand),collapse=":")
  },character(1))
C2_clusters_sig <- C2_clusters %>% dplyr::filter(C2 <= 0.05)

C3_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/C3P_pVals.txt"
C3_clusters <- read.table(C3_clusters_file,header=T)
C3_clusters$juncs <- vapply(rownames(C3_clusters),function(val){
  all_info <- strsplit(val,":")[[1]]
  strand <- strsplit(all_info[4],"_")[[1]][3]
  paste(c(all_info[seq(3)],strand),collapse=":")
  },character(1))
C3_clusters_sig <- C3_clusters %>% dplyr::filter(C3 <= 0.05)

C5_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/C5P_pVals.txt"
C5_clusters <- read.table(C5_clusters_file,header=T)
C5_clusters$juncs <- vapply(rownames(C5_clusters),function(val){
  all_info <- strsplit(val,":")[[1]]
  strand <- strsplit(all_info[4],"_")[[1]][3]
  paste(c(all_info[seq(3)],strand),collapse=":")
  },character(1))
C5_clusters_sig <- C5_clusters %>% dplyr::filter(C5 <= 0.05)

C6_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/C6P_pVals.txt"
C6_clusters <- read.table(C6_clusters_file,header=T)
C6_clusters$juncs <- vapply(rownames(C6_clusters),function(val){
  all_info <- strsplit(val,":")[[1]]
  strand <- strsplit(all_info[4],"_")[[1]][3]
  paste(c(all_info[seq(3)],strand),collapse=":")
  },character(1))
C6_clusters_sig <- C6_clusters %>% dplyr::filter(C6 <= 0.05)

C7_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/C7P_pVals.txt"
C7_clusters <- read.table(C7_clusters_file,header=T)
C7_clusters_sig <- C7_clusters %>% dplyr::filter(C7 <= 0.05)
C7_clusters_sig$juncs <- vapply(rownames(C7_clusters_sig),function(val){
  all_info <- strsplit(val,":")[[1]]
  strand <- strsplit(all_info[4],"_")[[1]][3]
  paste(c(all_info[seq(3)],strand),collapse=":")
},character(1))

P1_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/P1C_pVals.txt"
P1_clusters <- read.table(P1_clusters_file,header=T)
P1_clusters$juncs <- vapply(rownames(P1_clusters),function(val){
  all_info <- strsplit(val,":")[[1]]
  strand <- strsplit(all_info[4],"_")[[1]][3]
  paste(c(all_info[seq(3)],strand),collapse=":")
  },character(1))
P1_clusters_sig <- P1_clusters %>% dplyr::filter(P1 <= 0.05)

P2_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/P2C_pVals.txt"
P2_clusters <- read.table(P2_clusters_file,header=T)
P2_clusters$juncs <- vapply(rownames(P2_clusters),function(val){
  all_info <- strsplit(val,":")[[1]]
  strand <- strsplit(all_info[4],"_")[[1]][3]
  paste(c(all_info[seq(3)],strand),collapse=":")
  },character(1))
P2_clusters_sig <- P2_clusters %>% dplyr::filter(P2 <= 0.05)

P3_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/P3C_pVals.txt"
P3_clusters <- read.table(P3_clusters_file,header=T)
P3_clusters$juncs <- vapply(rownames(P3_clusters),function(val){
  all_info <- strsplit(val,":")[[1]]
  strand <- strsplit(all_info[4],"_")[[1]][3]
  paste(c(all_info[seq(3)],strand),collapse=":")
  },character(1))
P3_clusters_sig <- P3_clusters %>% dplyr::filter(P3 <= 0.05)

P5_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/P5C_pVals.txt"
P5_clusters <- read.table(P5_clusters_file,header=T)
P5_clusters$juncs <- vapply(rownames(P5_clusters),function(val){
  all_info <- strsplit(val,":")[[1]]
  strand <- strsplit(all_info[4],"_")[[1]][3]
  paste(c(all_info[seq(3)],strand),collapse=":")
},character(1))
P5_clusters_sig <- P5_clusters %>% dplyr::filter(P5 <= 0.05)

P6_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/P6C_pVals.txt"
P6_clusters <- read.table(P6_clusters_file,header=T)
P6_clusters$juncs <- vapply(rownames(P6_clusters),function(val){
  all_info <- strsplit(val,":")[[1]]
  strand <- strsplit(all_info[4],"_")[[1]][3]
  paste(c(all_info[seq(3)],strand),collapse=":")
},character(1))
P6_clusters_sig <- P6_clusters %>% dplyr::filter(P6 <= 0.05)

P7_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/P7C_pVals.txt"
P7_clusters <- read.table(P7_clusters_file,header=T)
P7_clusters$juncs <- vapply(rownames(P7_clusters),function(val){
  all_info <- strsplit(val,":")[[1]]
  strand <- strsplit(all_info[4],"_")[[1]][3]
  paste(c(all_info[seq(3)],strand),collapse=":")
},character(1))
P7_clusters_sig <- P7_clusters %>% dplyr::filter(P7 <= 0.05)

juncs_sig <- unique(c(C1_clusters_sig$juncs,C2_clusters_sig$juncs,
           C3_clusters_sig$juncs,C5_clusters_sig$juncs,
           C6_clusters_sig$juncs,C7_clusters_sig$juncs,
           P1_clusters_sig$juncs,P2_clusters_sig$juncs,
           P3_clusters_sig$juncs,P5_clusters_sig$juncs,
           P6_clusters_sig$juncs,P7_clusters_sig$juncs))
juncs_sig_C <- unique(c(C1_clusters_sig$juncs,C2_clusters_sig$juncs,
           C3_clusters_sig$juncs,C5_clusters_sig$juncs,
           C6_clusters_sig$juncs,C7_clusters_sig$juncs))
juncs_sig_P <- unique(c(P1_clusters_sig$juncs,P2_clusters_sig$juncs,
           P3_clusters_sig$juncs,P5_clusters_sig$juncs,
           P6_clusters_sig$juncs,P7_clusters_sig$juncs))
juncs_all <- unique(c(C1_clusters$juncs,C2_clusters$juncs,
           C3_clusters$juncs,C5_clusters$juncs,
           C6_clusters$juncs,C7_clusters$juncs,
           P1_clusters$juncs,P2_clusters$juncs,
           P3_clusters$juncs,P5_clusters$juncs,
           P6_clusters$juncs,P7_clusters$juncs))

```

```{r,eval=F,echo=F}

juncs_sig_df <- data.frame(matrix(unlist(strsplit(juncs_sig,":")),nrow=length(juncs_sig),ncol=4,byrow=T))
colnames(juncs_sig_df)<-c("chrom","start","end","strand")

write.table(juncs_sig_df,
            file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs_filt/juncs_sig_CP_PC.txt",
            sep="\t",
            quote=F,
            col.names=T,
            row.names=F)
saveRDS(juncs_sig_df,
        file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs_filt/juncs_sig_CP_PC.rds")

juncs_all_df <- data.frame(matrix(unlist(strsplit(juncs_all,":")),nrow=length(juncs_all),ncol=4,byrow=T))
colnames(juncs_all_df)<-c("chrom","start","end","strand")

write.table(juncs_all_df,
            file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs_filt/juncs_all_CP_PC.txt",
            sep="\t",
            quote=F,
            col.names=T,
            row.names=F)

```


```{r,eval=F,echo=F}
# splicemutr junction annotations run on juncs_sig_df

txdb_file<-"/media/theron/My_Passport/reference_genomes/GTF_GFF/ENSEMBL/Mus_musculus.GRCm38.102.chr.sqlite"
txdb<-loadDb(txdb_file) # making the txdb from gtf

introns_by_tx<-data.frame(unlist(intronsByTranscript(txdb,use.names=T)))
juncs_test<-sprintf("%s:%s:%s:%s",introns_by_tx$seqnames,introns_by_tx$start-1,introns_by_tx$end+1,introns_by_tx$strand)
introns_by_tx$juncs <- juncs_test
C1_SJ<-read.table("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/C1SJ.out.tab")
C1_SJ_strand <- vapply(C1_SJ$V4,function(val){
  if (val == 0){
    return("*")
  } else if (val == 1){
    return("+")
  } else if (val == 2){
    return("-")
  }
},character(1))
juncs<-sprintf("%s:%s:%s:%s",C1_SJ$V1,C1_SJ$V2,C1_SJ$V3,C1_SJ_strand)

annotated_outlier_junctions<-juncs_all %in% introns_by_tx$juncs
table(annotated_outlier_junctions)

```

# Mutation immunogenicity

```{r,echo=F}

## reading mut data and filtering for missense mutations

maf_file <- "/media/theron/My_Passport/Ali_data/WES Data for 4T1MIS/MB01JHU503/MB01JHU503_000_analysis/MAF/DNA_4TIMSI_haplotypecaller.maf"
ali_maf = read.maf(maf_file)
ali_maf_df <- data.frame(ali_maf@data)
ali_maf_df_miss <- ali_maf_df %>% dplyr::filter(Variant_Type == "SNP" & Variant_Classification=="Missense_Mutation")
ali_maf_df_miss$row<-seq(nrow(ali_maf_df_miss))
ali_maf_df_miss$mut_list <- sprintf("%s:%s:%s:%s",ali_maf_df_miss$Chromosome,ali_maf_df_miss$Start_Position,ali_maf_df_miss$Reference_Allele,ali_maf_df_miss$Allele)

```


```{r,echo=F}

## extracting transcript and gene information

refseq_transcripts <- lapply(ali_maf_df_miss$all_effects,function(val){
  val_split <- strsplit(val,";")[[1]]
  val_split_split <- strsplit(val_split,",")
  data.frame(t(vapply(seq(length(val_split_split)),function(val_val){
    val_val <- val_split_split[[val_val]]
    if (length(val_val)<5){
      return(c(val_val[1],val_val[4],""))
    } else {
      return(c(val_val[1],val_val[4],val_val[5]))
    }
  },character(3))))
})

for (i in seq(length(refseq_transcripts))){
  if (i == 1){
    refseq_transcripts_df <- refseq_transcripts[[i]]
  } else {
    refseq_transcripts_df <- rbind(refseq_transcripts_df,refseq_transcripts[[i]])
  }
}

ens_refseq<-refseq_transcripts_df
colnames(ens_refseq)<-c("gene","ensembl","refseq")
ens_refseq<-unique(ens_refseq)

rm(refseq_transcripts_df)
rm(refseq_transcripts)

```

```{r,eval=FALSE,echo=F}

# Processing mutations

source("/media/theron/My_Passport/splicemute/R/functions.R")
locate_mutation_in_cds <- function(cds_tx_sort,mutation_chr,
                                   mutation_start,mutation_end){
  width <- running_sum(cds_tx_sort$width)
  for (i in seq(nrow(cds_tx_sort))){
    if(mutation_end <= cds_tx_sort[i,"end"] & mutation_end >= cds_tx_sort[i,"start"]){
      return(width[i] - (cds_tx_sort[i,"end"]-mutation_end))
    }
  }
  return(-1)
}

extract_portion <- function(trans_str,mut_loc_prot){
  trans_str_split <- strsplit(trans_str,"")[[1]]
  trans_str_split[mut_loc_prot] <- tolower(trans_str_split[mut_loc_prot])
  start <- mut_loc_prot - 8
  if (start < 1){
    start<-1
  }
  end <- mut_loc_prot + 8
  if (end > nchar(trans_str)){
    end <- nchar(trans_str)
  }
  return(paste(trans_str_split[seq(start,end)],collapse=""))
}

conjugate <- function(nuc){
  if (nuc == "A"){
    return("T")
  } else if (nuc == "T"){
    return("A")
  } else if (nuc == "G"){
    return("C")
  } else if (nuc == "C"){
    return("G")
  }
}

BSgenome <- BSgenome.Mmusculus.GENCODE.GRCm38.102
txdb<-loadDb("/media/theron/My_Passport/reference_genomes/GTF_GFF/ENSEMBL/Mus_musculus.GRCm38.102.chr.sqlite")
cds_by_tx<-cdsBy(txdb,by="tx",use.names=T)
protein_coding_txs <- names(cds_by_tx)
tots<-nrow(ali_maf_df_miss)
ali_maf_dat_muts <- as.data.frame(t(vapply(seq(tots),function(i){
  if (i %% 100 == 0){print(sprintf("%d out of%d",i,tots))}
  txs <- transcripts[[i]]
  tx_record <- c()
  mut_loc <- vapply(txs,function(val){
    tx_record <<-c(tx_record,val)
    if (val %in% protein_coding_txs){
      cds_tx <- cds_by_tx[[val]]
      cds_tx_sort <- as.data.frame(sort(cds_tx))
      mutation_chr <- ali_maf_df_miss$Chromosome[i]
      mutation_start <- ali_maf_df_miss$Start_Position[i]
      mutation_end <- ali_maf_df_miss$End_Position[i]
      mutation_strand <- ali_maf_df_miss$Strand[i]
      mutation_allele <- ali_maf_df_miss$Allele[i]
      mutation_allele_norm <- ali_maf_df_miss$Reference_Allele[i]
      mut_loc<-locate_mutation_in_cds(cds_tx_sort,mutation_chr,
                                   mutation_start,mutation_end)
      sequence<-getSeq(BSgenome,sort(cds_tx))
      sequ<-paste(as.character(sequence[1:length(sequence)]), collapse="")
      if(!check_tx(sequ)){return("untrans")}
      if (mut_loc == -1){
        return("outside_cds")
      }
      sequ_split<-strsplit(sequ,"")[[1]]
      if (mutation_strand != cds_tx_sort$strand[1]){
        sequ_split[mut_loc]<-conjugate(mutation_allele)
        sequ_mut <- paste(sequ_split,collapse="")
      } else if (mutation_strand == cds_tx_sort$strand[1]){
        sequ_split[mut_loc]<-mutation_allele
        sequ_mut <- paste(sequ_split,collapse="")
      }
      trans<-translate(DNAStringSet(sequ_mut),no.init.codon = F)
      mut_loc_prot <- floor(mut_loc/3)
      trans_str <- as.character(trans)
      portion <- extract_portion(trans_str,mut_loc_prot)
      return(portion)
    } else {
      return("outside_cds")
    }
  },character(1))
  mut_loc_names <- paste(names(mut_loc),collapse=":")
  mut_loc_vals <- paste(unname(mut_loc),collapse=":")
  return(c(mut_loc_names,mut_loc_vals))
},character(2))))

write.table(ali_maf_dat_muts,
            file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs_filt/ali_maf_muts.txt",
            sep="\t",
            quote=F,
            col.names=T,
            row.names=F)
saveRDS(ali_maf_dat_muts,file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs_filt/ali_maf_muts.rds")

```


```{r,eval=F,echo=F}

# creating fasta file

ali_maf_dat_muts <- read.table("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/ali_maf_muts.txt",header = T)
sequences<-lapply(seq(nrow(ali_maf_dat_muts)),function(row_val){
  transcripts<-strsplit(ali_maf_dat_muts$V1[row_val],":")[[1]]
  sequences<-strsplit(ali_maf_dat_muts$V2[row_val],":")[[1]]
  keep<-!(sequences %in% c("outside_cds","untrans"))
  if (!any(keep)){return(NA)}
  tx<-transcripts[keep]
  seq<-sequences[keep]
  names(seq)<-sprintf("%s_%d",tx,rep(row_val,length(which(keep))))
  return(seq)
})
sequences<-unlist(sequences)
sequences<-sequences[!is.na(sequences)]
tx_name <- names(sequences)
sequences <- vapply(sequences,function(seq){
  seq<-toupper(seq)
  seq<-str_replace_all(seq,"[*]","")
},character(1))
names(sequences)<-seq(length(tx_name))
sequences<-AAStringSet(sequences)
out_fasta<-"/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs_filt/ali_maf_muts.fa"
writeXStringSet(sequences,out_fasta)
saveRDS(tx_name,file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs_filt/tx_names.rds")

```

```{r,eval=F,echo=F}

breaks<-seq(1,length(sequences),5000)

for (i in seq(length(breaks))){
  if (i == length(breaks)){
    sequences_small<-AAStringSet(sequences[seq(breaks[i],length(sequences))])
    out_fasta<-sprintf("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs_filt/ali_maf_muts_%d.fa",i)
    writeXStringSet(sequences_small,out_fasta,append=FALSE)
  } else {
    sequences_small<-AAStringSet(sequences[seq(breaks[i],breaks[i+1]-1)])
    out_fasta<-sprintf("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs_filt/ali_maf_muts_%d.fa",i)
    writeXStringSet(sequences_small,out_fasta,append=FALSE)
  }
}

```


```{r,echo=F}

## combining netmhc output for mutations

mut1<-read.table("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/immunogenicity/mut1_NetMHC.txt",header=T)
mut2<-read.table("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/immunogenicity/mut2_NetMHC.txt",header=T)
mut3<-read.table("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/immunogenicity/mut3_NetMHC.txt",header=T)
mut4<-read.table("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/immunogenicity/mut4_NetMHC.txt",header=T)
mut_imm <- rbind(mut1,mut2,mut3,mut4)
mut_imm_binding <- mut_imm %>% dplyr::filter(N_binders>0)
tx_names<-readRDS("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/tx_names.rds")
tx_names<-data.frame(t(vapply(tx_names,function(tx){str_split(tx,"_")[[1]]},character(2))))
mut_imm_binding$tx<-NA
mut_imm_binding$row_val<-NA
mut_imm_binding[,c("tx","row_val")]<-tx_names[mut_imm_binding$ID,c("X1","X2")]

mut1<-read.table("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/immunogenicity/mut1_NetMHC.txt",header=T)
mut2<-read.table("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/immunogenicity/mut2_NetMHC.txt",header=T)
mut3<-read.table("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/immunogenicity/mut3_NetMHC.txt",header=T)
mut4<-read.table("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/immunogenicity/mut4_NetMHC.txt",header=T)
mut_imm <- rbind(mut1,mut2,mut3,mut4)
mut_imm_binding <- mut_imm %>% dplyr::filter(N_binders>0)
tx_names<-readRDS("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/tx_names.rds")
tx_names<-data.frame(t(vapply(tx_names,function(tx){str_split(tx,"_")[[1]]},character(2))))
mut_imm_binding$tx<-NA
mut_imm_binding$row_val<-NA # row val comes from ali_maf_dat_muts row
mut_imm_binding[,c("tx","row_val")]<-tx_names[mut_imm_binding$ID,c("X1","X2")]
mut_imm_binding$row_val <- as.numeric(mut_imm_binding$row_val)
  
mut_imm_stats_per_row <- data.frame(t(vapply(unique(mut_imm_binding$row_val),function(row){
  mut_imm_small <- mut_imm_binding %>% dplyr::filter(row_val == row)
  ranks <- c(mut_imm_small$Rank,mut_imm_small$Rank.1,mut_imm_small$Rank.2)
  SB <- length(which(ranks <= 2))
  WB <- length(which(ranks > 2 & ranks <= 10))
  total <- WB+SB
  return(c(row,SB,WB,total))
},numeric(4))))
colnames(mut_imm_stats_per_row)<-c("row_val","SB_count","WB_count","total")
rownames(mut_imm_stats_per_row) <- mut_imm_stats_per_row$row_val
mut_imm_stats_per_row <- mut_imm_stats_per_row[,c("SB_count","WB_count","total")]

```


```{r,echo=F}
## Creating fold change log per transcript and arm

isoform_diff_expr <- "/media/theron/My_Passport/Ali_data/HTMBmice_RNAseq_Data/MB01JHU504/MB01JHU504_000_analysis/DESeq2/isoforms"
pd1_vs_combo <- read.table(sprintf("%s/PD1_vs_Combo_isoforms_DESeq2.txt",isoform_diff_expr),header=T)
pd1_vs_combo <- pd1_vs_combo %>% dplyr::filter(PD1_vs_Combo_pvalue <= 0.05)
untreated_vs_combo <- read.table(sprintf("%s/Untreated_vs_Combo_isoforms_DESeq2.txt",isoform_diff_expr),header=T)
untreated_vs_combo <- untreated_vs_combo %>% dplyr::filter(Untreated_vs_Combo_pvalue <= 0.05)
untreated_vs_pd1 <- read.table(sprintf("%s/Untreated_vs_PD1_isoforms_DESeq2.txt",isoform_diff_expr),header=T)
untreated_vs_pd1 <- untreated_vs_pd1 %>% dplyr::filter(Untreated_vs_PD1_pvalue <= 0.05)
diff_data <- data.frame(c(pd1_vs_combo$transcript_id,untreated_vs_combo$transcript_id,untreated_vs_pd1$transcript_id),
                               c(pd1_vs_combo$gene_id,untreated_vs_combo$gene_id,untreated_vs_pd1$gene_id))
colnames(diff_data)<-c("transcript","gene")
diff_dat <- unique(diff_data)

```

```{r,echo=F}

## Creating fold change log per gene and arm

gene_diff_expr <- "/media/theron/My_Passport/Ali_data/HTMBmice_RNAseq_Data/MB01JHU504/MB01JHU504_000_analysis/DESeq2/refseq_genes"
pd1_vs_combo <- read.table(sprintf("%s/PD1_vs_Combo_genes_DESeq2.txt",gene_diff_expr),header=T)
pd1_vs_combo <- pd1_vs_combo %>% dplyr::filter(PD1_vs_Combo_pvalue <= 0.05)
untreated_vs_combo <- read.table(sprintf("%s/Untreated_vs_Combo_genes_DESeq2.txt",gene_diff_expr),header=T)
untreated_vs_combo <- untreated_vs_combo %>% dplyr::filter(Untreated_vs_Combo_pvalue <= 0.05)
untreated_vs_pd1 <- read.table(sprintf("%s/Untreated_vs_PD1_genes_DESeq2.txt",gene_diff_expr),header=T)
untreated_vs_pd1 <- untreated_vs_pd1 %>% dplyr::filter(Untreated_vs_PD1_pvalue <= 0.05)
diff_data_genes <- data.frame(c(pd1_vs_combo$gene_symbol,untreated_vs_combo$gene_symbol,untreated_vs_pd1$gene_symbol))
colnames(diff_data_genes)<-c("gene")
diff_dat_genes <- unique(diff_data_genes)

```

```{r,echo=F}
## Filtering reference and tumor vcf files for tumor mutations
### reading in reference and tumor vcf, takes a while

ref_vcf_file <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/VCF/BALB_cJ.mgp.v5.snps.dbSNP142.vcf"
ref_vcf <- read.table(ref_vcf_file,header=T)
tumor_vcf_file <- "/media/theron/My_Passport/Ali_data/WES Data for 4T1MIS/MB01JHU503/MB01JHU503_000_analysis/HaplotypeCaller/DNA_4TIMSI_haplotypecaller.vcf"
tumor_vcf <- read.table(tumor_vcf_file,header=T)

```

```{r,echo=F}

### finding tumor-specific mutations

ref_vcf_list <- sprintf("%s:%s:%s:%s",ref_vcf$CHROM,ref_vcf$POS,ref_vcf$REF,ref_vcf$ALT)
tumor_vcf_list <- sprintf("%s:%s:%s:%s",str_remove(tumor_vcf$CHROM,"chr"),tumor_vcf$POS,tumor_vcf$REF,tumor_vcf$ALT)
somatic_muts <- tumor_vcf_list[which(!(tumor_vcf_list %in% ref_vcf_list))]
ali_maf_df_miss_som<-ali_maf_df_miss[which(ali_maf_df_miss$mut_list %in% somatic_muts),]
rows_to_keep <- ali_maf_df_miss_som$row

```

```{r,eval=F,echo=F}
### Creating BED file for mutations to be used by samtools mpileup

chrom<-sprintf("chr%s",ali_maf_df_miss_som$Chromosome)
start<-ali_maf_df_miss_som$Start_Position-1
end<-ali_maf_df_miss_som$End_Position

maf_bed <- data.frame(chrom,start,end)
write.table(maf_bed, file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/maf.bed",
            sep="\t",
            quote=F,
            col.names=F,
            row.names=F)

```


```{r,echo=F}

## Processing the all samples pileup

arm_samples<-c("chr","pos","ref_base",
               "C1_total_reads","C1_base","C1_quality",
               "C2_total_reads","C2_base","C2_quality",
               "C3_total_reads","C3_base","C3_quality",
               "C5_total_reads","C5_base","C5_quality",
               "C6_total_reads","C6_base","C6_quality",
               "C7_total_reads","C7_base","C7_quality",
               "P1_total_reads","P1_base","P1_quality",
               "P2_total_reads","P2_base","P2_quality",
               "P3_total_reads","P3_base","P3_quality",
               "P5_total_reads","P5_base","P5_quality",
               "P6_total_reads","P6_base","P6_quality",
               "P7_total_reads","P7_base","P7_quality",
               "U1_total_reads","U1_base","U1_quality",
               "U2_total_reads","U2_base","U2_quality",
               "U3_total_reads","U3_base","U3_quality")
total_reads <- arm_samples[seq(4,length(arm_samples),3)]
base <- arm_samples[seq(5,length(arm_samples),3)]
quality <- arm_samples[seq(6,length(arm_samples),3)]
all_samples_pileup_file <- "/media/theron/My_Passport/Ali_data/HTMBmice_RNAseq_Data/MB01JHU504/MB01JHU504_000_analysis/RSEM/alignments/all_samples.pileup"
all_samples_pileup <- read.table(all_samples_pileup_file)
colnames(all_samples_pileup)<-arm_samples
total_reads_samples <- all_samples_pileup[,total_reads]
base_samples <- all_samples_pileup[,base]
quality <- all_samples_pileup[,quality]
chr_pos <- sprintf("%s:%s",all_samples_pileup$chr,all_samples_pileup$pos)
all_samples_pileup$chr_pos <- chr_pos

ali_maf_chr_pos <- sprintf("chr%s:%s",ali_maf_df_miss_som$Chromosome,ali_maf_df_miss_som$Start_Position)
ali_maf_df_miss_som$ali_maf_chr_pos <- ali_maf_chr_pos
rownames(ali_maf_df_miss_som) <- ali_maf_chr_pos

ali_maf_pileup <- ali_maf_df_miss_som[chr_pos,]
variant <- ali_maf_pileup$Allele
base_samples$variant <- variant
variant_count <- function(row){
  str_count(toupper(row[seq(length(row)-1)]),row[length(row)])
}

base_samples_variant_count<-data.frame(t(apply(base_samples,1,variant_count)))
base_samples_variant_freq <- base_samples_variant_count/total_reads_samples
colnames(base_samples_variant_freq) <- str_replace_all(base,"_base","_variant_freq")

is.nan.data.frame <- function(x){
  do.call(cbind, lapply(x, is.nan))
}

base_samples_variant_freq[is.nan(base_samples_variant_freq)] <- -1 +1
rownames(base_samples_variant_freq)<-seq(nrow(base_samples_variant_freq))
ali_maf_pileup[,c("SB_count","WB_count","total")] <- mut_imm_stats_per_row[as.character(ali_maf_pileup$row),]
ali_maf_pileup[is.na(ali_maf_pileup$total),"SB_count"]<-0
ali_maf_pileup[is.na(ali_maf_pileup$total),"WB_count"]<-0
ali_maf_pileup[is.na(ali_maf_pileup$total),"total"]<-0
total<-ali_maf_pileup$total

# Heatmap(base_samples_variant_freq,
#         right_annotation = rowAnnotation(total = anno_barplot(cbind(log2(as.numeric(ali_maf_pileup$SB_count)+1),
#                                                                     log2(as.numeric(ali_maf_pileup$WB_count)+1)),gp = gpar(fill = 3:4, col = 3:4))),
#         show_row_names=F,
#         show_column_names = T)

```


## Mutation expression per sample

The majority of tumor-specific mutations are depleted in the combination therapy treatment. The combination therapy showed the greatest response. Therefore, we think that the tumor clones that contain immunogenic mutations are killed off in the combination therapy. 

```{r,echo=F}

gene_expression_vst <- readRDS("/media/theron/My_Passport/Ali_data/HTMBmice_RNAseq_Data/MB01JHU504/MB01JHU504_000_analysis/RSEM/tables/genes/gene_expression_vst.rds")
gene_expression_vst <- cbind(rownames(gene_expression_vst),gene_expression_vst)
colnames(gene_expression_vst)<-c("gene",colnames(gene_expression_vst)[seq(2,ncol(gene_expression_vst))])
mut_imm_binding_tum <- mut_imm_binding %>% dplyr::filter(row_val %in% rows_to_keep)
gene_expression_vst<-gene_expression_vst %>% dplyr::filter(gene %in% diff_data_genes$gene)

ens_refseq$gene <- str_replace(ens_refseq$gene,"[.]","")
gene_val <- vapply(mut_imm_binding_tum$tx,function(ID){
  a <- which(ens_refseq$ensembl==ID)
  if (length(a) == 0){
    return("-1")
  } else {
    gene_val<-ens_refseq$gene[a[1]]
    return(gene_val)
  }
},character(1))
N_binders<-mut_imm_binding_tum$N_binders
row_val<-mut_imm_binding_tum$row_val


gene_expression_vst<-cbind(gene_expression_vst[unname(gene_val),],N_binders,row_val)
gene_expression_complete <- gene_expression_vst[!is.na(gene_expression_vst[,1]),]
gene_expression_complete[is.na(gene_expression_complete)]<-0
gene_expression_complete <- cbind(gene_expression_complete,mut_imm_stats_per_row[gene_expression_complete$row_val,])


C_mut_av <- gene_expression_complete[,seq(2,7)]
C_mut_vals<-vapply(seq(nrow(C_mut_av)),function(row_val){
  median(as.numeric(C_mut_av[row_val,]),na.rm = T)
},numeric(1))
P_mut_av <- gene_expression_complete[,seq(8,13)]
P_mut_vals<-vapply(seq(nrow(P_mut_av)),function(row_val){
  median(as.numeric(P_mut_av[row_val,]),na.rm = T)
},numeric(1))
U_mut_av <- gene_expression_complete[,seq(14,16)]
U_mut_vals<-vapply(seq(nrow(U_mut_av)),function(row_val){
  median(as.numeric(U_mut_av[row_val,]),na.rm = T)
},numeric(1))
gene_expression_average <- data.frame(C_mut_vals,
                                  P_mut_vals,
                                  U_mut_vals,
                                  gene_expression_complete$N_binders,
                                  gene_expression_complete$gene,
                                  gene_expression_complete$row_val,
                                  gene_expression_complete$SB_count,
                                  gene_expression_complete$WB_count,
                                  gene_expression_complete$total)
colnames(gene_expression_average)<-c("C_TPM_AV","P_TPM_AV","U_TPM_AV",
                                 "N_binders","gene_id",
                                 "row_val","SB_count","WB_count","total")

gene_expression_average_linear <- data.frame(c(gene_expression_average$C_TPM_AV,gene_expression_average$P_TPM_AV,gene_expression_average$U_TPM_AV), c(rep("combo",nrow(gene_expression_average)),rep("pd1",nrow(gene_expression_average)),rep("untreated",nrow(gene_expression_average))),
                                             rep(gene_expression_average$gene_id,3),
                                         rep(gene_expression_average$row_val,3),rep(gene_expression_average$N_binders,3))
colnames(gene_expression_average_linear)<-c("vst","arm","gene_id","row_val","N_binders")
gene_expression_average_linear$unique_id<-sprintf("%s_%s",gene_expression_average$gene_id,gene_expression_average$row_val)

```

Here the average variance-stabilized gene expression of genes harboring mutations are shown for each treatment type. There is a significant difference in average between combination and untreated.

```{r,echo=F}

my_comparisons <- list( c("combo", "pd1"), c("combo", "untreated"), c("pd1", "untreated") )
ggplot(gene_expression_average_linear,aes(x=arm,y=vst))+geom_violin(aes(fill=arm))+
  stat_compare_means(comparisons = my_comparisons)

```

This tracks the variance-stabilized gene expression of specific mutation-harboring genes across treatment type.

```{r,echo=F}

ggplot(gene_expression_average_linear,aes(x=arm,y=vst,group=unique_id,color=unique_id))+
  geom_point(size=0.5)+geom_line(size=0.1)+
  theme(legend.position = "none")

```

A plot showing the variance-stabilized gene expression across treatment type. Each point on the x-axis is a mutation-harboring gene. 

```{r,echo=F}

ggplot(gene_expression_average_linear,aes(x=unique_id,y=vst,group=arm,color=arm))+
  geom_point(shape=1)+geom_line()+
  theme(legend.position = "none",
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

```


```{r,echo=F}
gene_expression_average$unique_id<-sprintf("%s_%s",gene_expression_average$gene_id,gene_expression_average$row_val)
gene_expression_average$expression_order <- vapply(seq(nrow(gene_expression_average)),function(row_val){
  vals<-gene_expression_average[row_val, c("C_TPM_AV","P_TPM_AV","U_TPM_AV")]
  if (vals$C_TPM_AV < vals$P_TPM_AV & vals$P_TPM_AV < vals$U_TPM_AV){
    return(c("combo<pd1<untreated"))
  } else if (vals$C_TPM_AV < vals$U_TPM_AV & vals$U_TPM_AV < vals$P_TPM_AV) {
    return("combo<untreated<pd1")
  } else if (vals$P_TPM_AV < vals$C_TPM_AV & vals$C_TPM_AV < vals$U_TPM_AV){
    return("pd1<combo<untreated")
  } else if (vals$P_TPM_AV < vals$U_TPM_AV & vals$U_TPM_AV < vals$C_TPM_AV){
    return("pd1<untreated<combo")
  } else if (vals$U_TPM_AV < vals$P_TPM_AV & vals$P_TPM_AV < vals$C_TPM_AV) {
    return("untreated<pd1<combo")
  } else if (vals$U_TPM_AV < vals$C_TPM_AV & vals$C_TPM_AV < vals$P_TPM_AV) {
    return("untreated<combo<pd1")
  } else if (sd(vals)==0 & round(sum(vals))==14){
    return("no expression")
  } else {
    return("something's equal")
  }
},character(1))

```

This plot shows how many mutation-harboring genes fall within a specific order of variance-stabilized gene expression across treatment type. The majority of mutation-harboring genes show depletion in combination therapy and higher expression in the pd1 and untreated arms. 

```{r,echo=F}
gene_expression_average <- unique(gene_expression_average)
ggplot(gene_expression_average,aes(x=expression_order,fill=expression_order))+
  geom_bar(aes(y=(..count..)/sum(..count..)))+
  theme(axis.text.x=element_blank())

```

This plot shows the distribution after subsetting to the "unique" set of genes. Genes can hardbor multiple mutations, affecting the immunogenicity score for that gene. No gene is repeated in this frequency plot. 

```{r,echo=F}

gene_expression_average_unique <- unique(gene_expression_average[,c(-6,-7,-8,-9,-10)])
ggplot(gene_expression_average_unique,aes(x=expression_order,fill=expression_order))+
  geom_bar(aes(y=(..count..)/sum(..count..)))+
  theme(axis.text.x=element_blank())

```

The following three plots each show that there is no significant difference in immunogenic kmer binding strength across expression order type.

```{r,echo=F}
my_comparisons<-list( c("combo<pd1<untreated", "combo<untreated<pd1"), c("combo<pd1<untreated", "pd1<combo<untreated"), c("combo<pd1<untreated", "pd1<untreated<combo"), c("combo<pd1<untreated","untreated<pd1<combo"),c("combo<pd1<untreated","untreated<combo<pd1"),
                      c("pd1<combo<untreated","pd1<untreated<combo"),c("untreated<pd1<combo","untreated<combo<pd1"))
ggplot(gene_expression_average,aes(x=expression_order,y=log2(as.numeric(total)),fill=expression_order))+
  geom_violin()+
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))+
  stat_compare_means(comparisons=my_comparisons)+ylab("log2(total binder count)")

```


```{r,echo=F}
my_comparisons<-list( c("combo<pd1<untreated", "combo<untreated<pd1"), c("combo<pd1<untreated", "pd1<combo<untreated"), c("combo<pd1<untreated", "pd1<untreated<combo"), c("combo<pd1<untreated","untreated<pd1<combo"),c("combo<pd1<untreated","untreated<combo<pd1"),
                      c("pd1<combo<untreated","pd1<untreated<combo"),c("untreated<pd1<combo","untreated<combo<pd1"))
ggplot(gene_expression_average,aes(x=expression_order,y=log2(as.numeric(SB_count)),fill=expression_order))+
  geom_violin()+
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))+
  stat_compare_means(comparisons=my_comparisons)+ylab("log2(SB_count)")

```


```{r,echo=F}

ggplot(gene_expression_average,aes(x=expression_order,y=log2(as.numeric(WB_count)),fill=expression_order))+
  geom_violin()+
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))+
    stat_compare_means(comparisons=my_comparisons)+ylab("log2(WB_count)")


```


```{r,echo=F}

base_samples_variant_count<-data.frame(t(apply(base_samples,1,variant_count)))
base_samples_variant_freq <- base_samples_variant_count/total_reads_samples
colnames(base_samples_variant_freq) <- str_replace_all(base,"_base","_variant_freq")

is.nan.data.frame <- function(x){
  do.call(cbind, lapply(x, is.nan))
}
base_samples_variant_freq[is.nan(base_samples_variant_freq)] <- -1
base_samples_variant_freq <- base_samples_variant_freq + 1
rownames(base_samples_variant_freq)<-seq(nrow(base_samples_variant_freq))
row <- ali_maf_pileup$row
base_samples_variant_freq <- cbind(base_samples_variant_freq,row)
base_samples_variant_freq <- base_samples_variant_freq %>% dplyr::filter(row %in% gene_expression_average$row_val)
rownames(base_samples_variant_freq)<-base_samples_variant_freq$row

gene_expression_average_var_freq <- cbind(gene_expression_average,base_samples_variant_freq[as.character(gene_expression_average$row_val),seq(ncol(base_samples_variant_freq)-1)])

```

## Variant Allele Frequency Heatmaps

The variant allele frequency (VAF) is calculated as follows: $VAF=\frac{R_{v}}{R_{t}}$ where $R_{v}$ is the number of reads in sample sequencing library that show expression of the variant and $R_{t}$ is the total number of reads in the sample sequencing library that span the variant location. In the following heatmaps, any sample with no expression ($R_{t}=0$) associated with a variant is given a vaf value of -1. Across the heatmap 1 is added to the vaf values, scaling the vaf range from 0-2, where 0 indicates no expression, and 2 indicates complete expression of the variant. There are very few samples with mixed variant and non-variant expression. There are two sets of 3 heatmaps, the first unscaled and the second scaled across rows. Each set of figures is shown in three; one where both columns and rows are clustered, one where columns are not clustered and rows are clustered, and one where neither are clustered. There is no apparent relationship between variant allele frequency and expression order when looking at the heatmaps. The samples do not cluster by treatment type. 

```{r,echo=F}
gene_expression_average_var_freq <- gene_expression_average_var_freq[order(gene_expression_average_var_freq$expression_order),]
var_freq_heatmap <- as.matrix(gene_expression_average_var_freq[,seq(ncol(gene_expression_average_var_freq)-14,ncol(gene_expression_average_var_freq))])
row_berid <- apply(var_freq_heatmap,1,sd)!=0
var_freq_heatmap<-var_freq_heatmap[row_berid,]
var_freq_heatmap_scaled <- t(apply(var_freq_heatmap,1,scale))
colnames(var_freq_heatmap)<-colnames(gene_expression_average_var_freq[,seq(ncol(gene_expression_average_var_freq)-14,ncol(gene_expression_average_var_freq))])
expr_order <- gene_expression_average_var_freq$expression_order[row_berid]

Heatmap(var_freq_heatmap,
        right_annotation = rowAnnotation(expression_order = expr_order),
        show_row_names=F,
        show_column_names = T,
        cluster_rows = T,
        cluster_columns = T)

```


```{r,echo=F}

Heatmap(var_freq_heatmap,
        right_annotation = rowAnnotation(expression_order = expr_order),
        show_row_names=F,
        show_column_names = T,
        cluster_rows = T,
        cluster_columns = F)

```

```{r,echo=F}

Heatmap(var_freq_heatmap,
        right_annotation = rowAnnotation(expression_order = expr_order),
        show_row_names=F,
        show_column_names = T,
        cluster_rows = F,
        cluster_columns = F)

```


```{r,echo=F}

Heatmap(var_freq_heatmap_scaled,
        right_annotation = rowAnnotation(expression_order = expr_order),
        show_row_names=F,
        show_column_names = T,
        cluster_rows = T,
        cluster_columns = T)

```


```{r,echo=F}

Heatmap(var_freq_heatmap_scaled,
        right_annotation = rowAnnotation(expression_order = expr_order),
        show_row_names=F,
        show_column_names = T,
        cluster_rows = T,
        cluster_columns = F)

```


```{r,echo=F}

Heatmap(var_freq_heatmap_scaled,
        right_annotation = rowAnnotation(expression_order = expr_order),
        show_row_names=F,
        show_column_names = T,
        cluster_rows = F,
        cluster_columns = F)

# C_var_freq_av <- apply(gene_expression_average_var_freq[,seq(12,17)],1,mean)
# P_var_freq_av <- apply(gene_expression_average_var_freq[,seq(18,23)],1,mean)
# U_var_freq_av <- apply(gene_expression_average_var_freq[,seq(24,26)],1,mean)
# gene_expression_average_var_freq_linear <- data.frame(rep(gene_expression_average_var_freq$expression_order),
#                                                       C_var_freq_av,P_var_freq_av,U_var_freq_av)
# colnames(gene_expression_average_var_freq_linear) <- c("expression_order","C_var_freq_av","P_var_freq_av","U_var_freq_av")
# gene_expression_average_var_freq_linear <- gene_expression_average_var_freq_linear[order(gene_expression_average_var_freq_linear$expression_order),]
# Heatmap(as.matrix(gene_expression_average_var_freq_linear[,seq(2,4)]),
#         right_annotation = rowAnnotation(expression_order = gene_expression_average_var_freq_linear$expression_order),
#         show_row_names=F,
#         show_column_names = T,
#         cluster_rows = F,
#         cluster_columns = F)

```

## Variant Allele Frequency Across Expression Order

The following plots show the variant allele frequency across expression order groups for each sample. The first plot contains sample-mutation-gene points that do not have expression at the variant location. The second plot has these points filtered out. Of note, there are significant differences between the combo<pd1<untreated group and the untreated<pd1<combo and untreated<pd1<combo groups. 

```{r,echo=F}
## Variant Allele frequency linearized

gene_expression_average_var_freq_linear <- data.frame(rep(gene_expression_average$expression_order,15),
                                                  c(gene_expression_average_var_freq$C1_variant_freq,
                                                    gene_expression_average_var_freq$C2_variant_freq,
                                                    gene_expression_average_var_freq$C3_variant_freq,
                                                    gene_expression_average_var_freq$C5_variant_freq,
                                                    gene_expression_average_var_freq$C6_variant_freq,
                                                    gene_expression_average_var_freq$C7_variant_freq,
                                                    gene_expression_average_var_freq$P1_variant_freq,
                                                    gene_expression_average_var_freq$P2_variant_freq,
                                                    gene_expression_average_var_freq$P3_variant_freq,
                                                    gene_expression_average_var_freq$P5_variant_freq,
                                                    gene_expression_average_var_freq$P6_variant_freq,
                                                    gene_expression_average_var_freq$P7_variant_freq,
                                                    gene_expression_average_var_freq$U1_variant_freq,
                                                    gene_expression_average_var_freq$U2_variant_freq,
                                                    gene_expression_average_var_freq$U3_variant_freq),
                                                  c(rep("C1_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("C2_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("C3_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("C5_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("C6_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("C7_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("P1_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("P2_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("P3_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("P5_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("P6_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("P7_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("U1_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("U2_variant_freq",nrow(gene_expression_average_var_freq)),
                                                    rep("U3_variant_freq",nrow(gene_expression_average_var_freq))),
                                                    rep(gene_expression_average_var_freq$SB_count,15),
                                                    rep(gene_expression_average_var_freq$WB_count,15),
                                                    rep(gene_expression_average_var_freq$total,15))
colnames(gene_expression_average_var_freq_linear)<-c("expression_order","variant_freq","sample","SB_count","WB_count","total")
gene_expression_average_var_freq_linear$sample <- vapply(gene_expression_average_var_freq_linear$sample,function(samp){
  if (str_detect(samp,"C")){
    return("combo")
  } else if (str_detect(samp,"P")){
    return("pd1")
  } else if (str_detect(samp,"U")){
    return("untreated")
  }
},character(1))

my_comparisons<-list( c("combo<pd1<untreated", "combo<untreated<pd1"), c("combo<pd1<untreated", "pd1<combo<untreated"), c("combo<pd1<untreated", "pd1<untreated<combo"), c("combo<pd1<untreated","untreated<pd1<combo"),c("combo<pd1<untreated","untreated<combo<pd1"),
                      c("pd1<combo<untreated","pd1<untreated<combo"),c("untreated<pd1<combo","untreated<combo<pd1"))
ggplot(gene_expression_average_var_freq_linear,aes(x=expression_order,y=variant_freq,fill=expression_order))+
  geom_violin()+
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))+
  stat_compare_means(comparisons=my_comparisons,method="wilcox.test")

```


```{r,echo=F}

gene_expression_average_var_freq_linear_filt <- gene_expression_average_var_freq_linear %>% dplyr::filter(variant_freq >= 1)
ggplot(gene_expression_average_var_freq_linear_filt,aes(x=expression_order,y=variant_freq,fill=expression_order))+
  geom_violin()+
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))+
  stat_compare_means(comparisons=my_comparisons,method="wilcox.test")

```

## Variant Allele Frequency With Respect to binder count

The following plots show statistically significant, yet biologically irrelevant, negative correlation between binder count and variant allele frequency. 

```{r,echo=F}

ggplot(gene_expression_average_var_freq_linear,aes(y=variant_freq,x=(as.numeric(SB_count))))+
  geom_point(aes=0.3,shape=1)+
  stat_cor(method = "spearman",
           label.x.npc = 0.75,
           label.y.npc = "top")+
  geom_smooth(method="lm")

```


```{r,echo=F}
ggplot(gene_expression_average_var_freq_linear,aes(y=variant_freq,x=(as.numeric(WB_count))))+
  geom_point(aes=0.3,shape=1)+
  stat_cor(method = "spearman",
           label.x.npc = 0.75,
           label.y.npc = "top")+
  geom_smooth(method="lm")
```


```{r,echo=F}

ggplot(gene_expression_average_var_freq_linear,aes(y=variant_freq,x=(as.numeric(total))))+
  geom_point(aes=0.3,shape=1)+
  stat_cor(method = "spearman",
           label.x.npc = 0.75,
           label.y.npc = "top")+
  geom_smooth(method="lm")

```

## Variant Allele Frequency With Respect to binder count, vaf >=0 

```{r,echo=F}
ggplot(gene_expression_average_var_freq_linear_filt,aes(y=variant_freq,x=(as.numeric(SB_count))))+
  geom_point(aes=0.3,shape=1)+
  stat_cor(method = "spearman",
           label.x.npc = 0.75,
           label.y.npc = "top")+
  geom_smooth(method="lm")

```


```{r,echo=F}

ggplot(gene_expression_average_var_freq_linear_filt,aes(y=variant_freq,x=(as.numeric(WB_count))))+
  geom_point(aes=0.3,shape=1)+
  stat_cor(method = "spearman",
           label.x.npc = 0.75,
           label.y.npc = "top")+
  geom_smooth(method="lm")

```


```{r,echo=F}

ggplot(gene_expression_average_var_freq_linear_filt,aes(y=variant_freq,x=(as.numeric(total))))+
  geom_point(aes=0.3,shape=1)+
  stat_cor(method = "spearman",
           label.x.npc = 0.75,
           label.y.npc = "top")+
  geom_smooth(method="lm")

```

# Splicemutr Immunogenicity

The splicing immunogenicity is calculated by identifying outlier splicing events between the combination and pd1 treatment groups. These outlier splicing events are then used to create transcripts which are translated, kmerized into 9mers, and run through mhc binding affinity prediction based on Balb/cJ mhc class 1 alleles. A gene splicing antigenicity metric is calculated as follows: $G=\frac{\sum_{j \in J}\frac{R_{j}}{R_G}*k_{j}}{|J|}$ where $R_{j}$ is the variance stabilized read count for the outlier splice-junction j, $k_{j}$ is the number of immunogenic kmers predicted from the transcript associated with the junction j, $R_{G}$ is the variance-stabilized read count for the specific gene G, and J is the set of junctions associated with the gene G. This metric serves as a weighted sum for the splicing-based immunogenic impact a gene has. Using this metric, calculated on combination and pd1 outlier junctions, we show that there are specific combination and pd1 immunogenecity signatures. We also show that overall, immunogenic scores are significantly lower in the combination associated genes than in the pd1 associated genes, supporting the idea of depletion of tumor clones with more immunogenic signatures.

```{r,eval=F,echo=F}

## creating immunogenic kmers per sample file and specific_splicemutr_dat

out_splice<-read.table("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/immunogenicity/outlier_splicing_results_CP_PC_NetMHC.txt",header=T)
splicemutr_dat <- read.table("/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/formed_transcripts_CP_PC/data_splicemutr.txt",header=T)
juncs <- sprintf("%s:%s:%s:%s",splicemutr_dat$chr,splicemutr_dat$start,splicemutr_dat$end,splicemutr_dat$cluster)
splicemutr_dat$juncs <- juncs
splicemutr_dat$rows<-seq(nrow(splicemutr_dat))

IDs <- unique(out_splice$ID)
ID_counts <- data.frame(IDs)
ID_counts$counts <- vapply(IDs,function(splice_ID){
  out_splice_small <- out_splice %>% dplyr::filter(ID == splice_ID)
  return(nrow(out_splice_small))
},numeric(1))
ID_counts$rows <- as.numeric(str_replace(ID_counts$IDs,"protein",""))
rownames(ID_counts)<-ID_counts$rows
ID_counts[as.character(which(!(seq(nrow(splicemutr_dat)) %in% ID_counts$rows))),]<-0
ID_counts$rows <- rownames(ID_counts)

ID_counts <- ID_counts[,c(3,1,rep(2,15))]
colnames(ID_counts) <- c("rows","juncs","C1","C2","C3","C5","C6","C7","P1","P2","P3","P5","P6","P7","U1","U2","U3")
ID_counts$juncs <- splicemutr_dat[as.numeric(ID_counts$rows),"juncs"]
ID_counts_C <- ID_counts %>% dplyr::filter(juncs %in% juncs_sig_C)
ID_counts_P <- ID_counts %>% dplyr::filter(juncs %in% juncs_sig_P)

saveRDS(ID_counts,file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/immunogenicity/kmers_specfic_CP_PC.rds")
saveRDS(ID_counts_C,file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/immunogenicity/kmers_specfic_C.rds")
saveRDS(ID_counts_P,file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/immunogenicity/kmers_specfic_P.rds")

genes<-unique(splicemutr_dat$gene)
mgi_symbols <- c("Ifi203",
                 "C920025E04Rik",
                 "Gsn",
                 "Cd44",
                 "Macf1",
                 "Ndufs1",
                 "Rbm26",
                 "H2-D1-H2-Q4",
                 "Prc1",
                 "Tspan4",
                 "Col6a3",
                 "Gm47287-Gm47295",
                 "Hist1h2bk-Hist1h2bj",
                 "H2-DMb2",
                 "Myof",
                 "Mtf2",
                 "Sirt2",
                 "Gbp8",
                 "Myo5a",
                 "Rtn4",
                 "Ighm-Ighj1",
                 "H2-D1",
                 "H2-Q4-H2-Q6",
                 "Ms4a6b",
                 "Ms4a6b-Ms4a6e",
                 "Tnc",
                 "Gbp4",
                 "Myl6",
                 "Gm30211",
                 "Igkj2-Igkc",
                 "Prrc2c",
                 "Uap1",
                 "Ptk2",
                 "1110038B12Rik",
                 "Srsf3",
                 "Mbnl1",
                 "Gbp2",
                 "Gbp2-Gbp2-ps",
                 "Ubr4",
                 "Asph",
                 "Arhgef10l",
                 "Gbp9",
                 "Gm43302",
                 "Gbp6",
                 "Dmwd",
                 "Picalm",
                 "Hif1a",
                 "H2-T22-H2-T10",
                 "H2-T22-Gm7030",
                 "Evi5",
                 "Hnrnpf",
                 "Lamc2",
                 "Gm49339",
                 "Gp49a",
                 "Scrib",
                 "Cpne1",
                 "Gm28036",
                 "Arhgef1",
                 "Cpsf6",
                 "Gbp9-Gm43302",
                 "Gbp8-Gm43302",
                 "Gbp4-Gm43302",
                 "Gbp9-Gbp6",
                 "Gbp8-Gbp6",
                 "Gbp4-Gbp6",
                 "Gprasp1",
                 "Armcx5")
ens_mgi_symbols<-data.frame(genes,mgi_symbols)
rownames(ens_mgi_symbols)<-ens_mgi_symbols$genes
saveRDS(splicemutr_dat,file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/formed_transcripts_CP_PC/data_splicemutr_CP_PC_ens.rds")
splicemutr_dat_C <- splicemutr_dat %>% dplyr::filter(juncs %in% juncs_sig_C)
splicemutr_dat_P <- splicemutr_dat %>% dplyr::filter(juncs %in% juncs_sig_P)
saveRDS(splicemutr_dat_C,file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/formed_transcripts_CP_PC/data_splicemutr_C_ens.rds")
saveRDS(splicemutr_dat_P,file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/formed_transcripts_CP_PC/data_splicemutr_P_ens.rds")

splicemutr_dat$gene<-ens_mgi_symbols[splicemutr_dat$gene,"mgi_symbols"]
splicemutr_dat_C <- splicemutr_dat %>% dplyr::filter(juncs %in% juncs_sig_C)
splicemutr_dat_P <- splicemutr_dat %>% dplyr::filter(juncs %in% juncs_sig_P)
saveRDS(splicemutr_dat,file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/formed_transcripts_CP_PC/data_splicemutr_CP_PC_mgi.rds")
saveRDS(splicemutr_dat_C,file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/formed_transcripts_CP_PC/data_splicemutr_C_mgi.rds")
saveRDS(splicemutr_dat_P,file="/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/formed_transcripts_CP_PC/data_splicemutr_P_mgi.rds")

```

## calculating per-gene metric

```{bash,eval=F}

./media/theron/My_Passport/scripts/Ali_data/analysis/bash/calc_gene_metric.sh

```

## Splicemutr data

```{r,echo=F}
gene_metric_file_C <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/formed_transcripts/C_gene_metric_mean.rds"
gene_metric_file_P <- "/media/theron/My_Passport/Ali_data/analysis/star_to_leaf_juncs/formed_transcripts/P_gene_metric_mean.rds"

gene_metric_C <- readRDS(gene_metric_file_C)
gene_metric_C <- gene_metric_C[,which(str_detect(colnames(gene_metric_C),"C"))]
gene_metric_P <- readRDS(gene_metric_file_P)
gene_metric_P <- gene_metric_P[,which(str_detect(colnames(gene_metric_P),"P"))]
genes <- unique(c(rownames(gene_metric_C),rownames(gene_metric_P)))

gene_metric <- data.frame(genes)
rownames(gene_metric) <- gene_metric$genes
gene_metric[rownames(gene_metric_C),colnames(gene_metric_C)]<-gene_metric_C
gene_metric[rownames(gene_metric_P),colnames(gene_metric_P)]<-gene_metric_P
gene_metric <- gene_metric[,seq(2,ncol(gene_metric))]
gene_metric[is.na(gene_metric)]<-0

gene_metric <- gene_metric[complete.cases(gene_metric),]
gene_metric<- gene_metric[!is.infinite(rowSums(gene_metric)),]
gene_metric <- gene_metric[apply(gene_metric,1,sd)!=0,]
gene_metric_scaled <- t(apply(gene_metric,1,scale))
colnames(gene_metric_scaled)<-colnames(gene_metric)

```


```{r,echo=F}
Heatmap(gene_metric_scaled,cluster_columns = F)
Heatmap(gene_metric_scaled,cluster_columns = T,cluster_rows = T)
gene_metric_log10 <- log10(gene_metric+1)
gene_metric_log10 <- as.matrix(gene_metric_log10)
Heatmap(gene_metric_log10,cluster_columns = F)
Heatmap(gene_metric_log10,cluster_columns = T,cluster_rows = T)

```


```{r,echo=F}
c_samples <- which(str_detect(colnames(gene_metric),"C"))
p_samples <- which(str_detect(colnames(gene_metric),"P"))

gene_metric_scaled <- data.frame(gene_metric)
gene_metric_summ <- data.frame(rownames(gene_metric))
gene_metric_summ$C_AV <- apply(gene_metric[,c_samples],1,mean)
gene_metric_summ$P_AV <- apply(gene_metric[,p_samples],1,mean)
colnames(gene_metric_summ)<-c("gene","C_AV","P_AV")
rownames(gene_metric_summ)<-gene_metric_summ$gene
gene_metric_summ <- gene_metric_summ[,c(2,3)]
samps <- c("C","P")
gene_metric_summ$order <- vapply(rownames(gene_metric_summ),function(gene){
  if (length(unique(unname(as.numeric(gene_metric_summ[gene,]))))<2){
    return("something's equal")
  }
  paste(as.character(samps[order(unname(as.numeric(gene_metric_summ[gene,])),decreasing=F)]),collapse="<")
},character(1))

ggplot(gene_metric_summ,aes(x=order,fill=order))+geom_bar(aes(y=(..count..)/sum(..count..)))+labs(xlab="splice_antigenicity order")

```


```{r,echo=F}
gene_metric_linear <- data.frame(rep(rownames(gene_metric_summ),2),
                                 c(gene_metric_summ$C_AV,gene_metric_summ$P_AV),
                                 c(rep("C",nrow(gene_metric_summ)),rep("P",nrow(gene_metric_summ))))
colnames(gene_metric_linear)<-c("genes","gene_metric","sample")
ggplot(gene_metric_linear,aes(x=genes,y=log10(gene_metric+1),color=sample))+
  geom_point()+
  geom_line()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

```


```{r,echo=F}

cf <- coef(lm(log10(gene_metric+1)~as.numeric(factor(sample)), data=gene_metric_linear))
my_comparisons=list(c("C","P"))
ggplot(gene_metric_linear,aes(x=sample,y=log10(gene_metric+1),groups=sample,color=sample))+
  geom_violin()+
  stat_compare_means(comparisons = my_comparisons,method="wilcox.test")+
  stat_summary(aes(group=1),fun=mean, geom="point", color="red", size=1) +
  geom_abline(slope=cf[2], intercept=cf[1], lwd=.5)

```




