---
title: "analysis_Theron"
author: "Theron Palmer"
date: "8/17/2021"
output: html_document
---

```{r setup, include=FALSE} 
knitr::opts_chunk$set(warning = FALSE, message = FALSE) 
```


# Libraries used in the analysis

```{r}

library(ggplot2)
library(knitr)
library(dplyr)
library(org.Mm.eg.db)
library(DT)
library(GeneOverlap)
library(GO.db)
library(KEGGREST)
library(KEGG.db)
library(fgsea)
library(maftools)
library(pheatmap)

```

# Loading in the necessary data

```{r}

base_dir <- "/media/theron/My_Passport/Ali_data/HTMBmice_RNAseq_Data/MB01JHU504/MB01JHU504_000_analysis/DESeq2/refseq_genes"
PD1_vs_combo <- read.table(sprintf("%s/%s",base_dir,"PD1_vs_Combo_genes_DESeq2.txt"),header=T)
untreated_vs_combo <- read.table(sprintf("%s/%s",base_dir,"Untreated_vs_Combo_genes_DESeq2.txt"),header=T)
untreated_vs_PD1 <- read.table(sprintf("%s/%s",base_dir,"Untreated_vs_PD1_genes_DESeq2.txt"),header=T)

```

# PD1 vs Combo (PD1/Combo)

## Differential Expression Data

```{r}

which_sig <- vapply(as.numeric(PD1_vs_combo$PD1_vs_Combo_pvalue),function(val){
  if (is.na(val)){
    return("Insig")
  }
  if (val <= 0.05){
    return("Sig")
  } else {
    return("Insig")
  }},character(1))
PD1_vs_combo$significance_pval <- which_sig

which_sig <- vapply(PD1_vs_combo$PD1_vs_Combo_padj,function(val){
  if (is.na(val)){
    return("Insig")
  }
  if (val <= 0.05){
    return("Sig")
  } else {
    return("Insig")
  }},character(1))
PD1_vs_combo$significance_padj <- which_sig

PD1_vs_combo_table <- PD1_vs_combo %>% dplyr::filter(!is.na(PD1_vs_Combo_pvalue) | !is.na(PD1_vs_Combo_padj))
datatable(PD1_vs_combo_table)

```


## Volcano Plots

```{r}

ggplot(PD1_vs_combo,aes(x=PD1_vs_Combo_log2FoldChange, y=-log10(PD1_vs_Combo_pvalue), color = significance_pval))+
  geom_text(label=PD1_vs_combo$gene_symbol)
ggplot(PD1_vs_combo,aes(x=PD1_vs_Combo_log2FoldChange, y=-log10(PD1_vs_Combo_padj), color = significance_padj))+
  geom_text(label=PD1_vs_combo$gene_symbol)

```
## gsea results on GO and KEGG terms

```{r}

symbol_to_entrez <- as.list(org.Mm.egALIAS2EG)
PD1_vs_combo$entrez <- symbol_to_entrez[PD1_vs_combo$gene_symbol]

SYMBOLToGO <- mapIds(org.Mm.eg.db,keys=PD1_vs_combo$gene_symbol,column='GO',keytype = 'SYMBOL',multiVals = list)
GOToSYMBOL <- sapply(reverseSplit(SYMBOLToGO),unique)
GOToSYMBOL <- GOToSYMBOL[sapply(GOToSYMBOL,length)>5]

SYMBOLToKEGG <- mapIds(org.Mm.eg.db,keys=PD1_vs_combo$gene_symbol,column='PATH',keytype = 'SYMBOL',multiVals = list)
KEGGToSYMBOL <- sapply(reverseSplit(SYMBOLToKEGG),unique)
KEGGToSYMBOL <- KEGGToSYMBOL[sapply(KEGGToSYMBOL,length)>5]

diff_dat_filt <- PD1_vs_combo %>% dplyr::filter(!is.na(PD1_vs_Combo_log2FoldChange) & !is.na(PD1_vs_Combo_pvalue))
ranks <- diff_dat_filt$PD1_vs_Combo_log2FoldChange*-log10(diff_dat_filt$PD1_vs_Combo_pvalue)
names(ranks)<-diff_dat_filt$gene_symbol
ranks<-ranks[unname(!is.na(ranks))]
# ranks<-rank(ranks,ties.method = "random")
fgseaRes_GO <- fgsea(GOToSYMBOL, ranks, eps=0,scoreType="pos")
fgseaRes_GO<-cbind(sapply(c('ONTOLOGY','TERM','DEFINITION'),
                        function(x){mapIds(GO.db,keys=fgseaRes_GO$pathway,keytype='GOID',column=x)}),fgseaRes_GO)
fgseaRes_KEGG <- fgsea(KEGGToSYMBOL, ranks, eps=0,scoreType="pos")
fgseaRes_KEGG <- cbind(KEGG.Name=unlist(as.list(KEGGPATHID2NAME)[fgseaRes_KEGG$pathway]),
                   fgseaRes_KEGG)

pathways<- readRDS("/media/theron/My_Passport/LM_LMT_LUCIANE_cellranger/references/gene_sets/mouse_hallmark.rds")
pathways_list <- lapply(unique(pathways$gs_name),function(pathway){
  pathway_small <- pathways %>% dplyr::filter(gs_name == pathway)
  return(pathway_small$gene_symbol)
})
names(pathways_list)<-unique(pathways$gs_name)

fgseaRes_HALLMARK <- fgsea(pathways_list, ranks, eps=0)
fgseaRes_HALLMARK <- fgseaRes_HALLMARK %>% dplyr::filter(padj <= 0.05)

```

### GO GSEA
```{r}

GO_table <- fgseaRes_GO %>% dplyr::filter(padj <= 0.05)
GO_table<-GO_table[,c("TERM","DEFINITION","padj","NES","size")]
datatable(GO_table)

```

### KEGG GSEA
```{r}

KEGG_table <- fgseaRes_KEGG %>% dplyr::filter(padj <= 0.05)
KEGG_table<-KEGG_table[,c("KEGG.Name","padj","NES","size")]
datatable(KEGG_table)

```

### HALLMARK GSEA
```{r}

datatable(fgseaRes_HALLMARK[,c(1,3,6,7)])

```


# Untreated vs Combo (Untreated/Combo) 

## Differential Expression Data

```{r}

which_sig <- vapply(as.numeric(untreated_vs_combo$Untreated_vs_Combo_pvalue),function(val){
  if (is.na(val)){
    return("Insig")
  }
  if (val <= 0.05){
    return("Sig")
  } else {
    return("Insig")
  }},character(1))
untreated_vs_combo$significance_pval <- which_sig

which_sig <- vapply(untreated_vs_combo$Untreated_vs_Combo_padj,function(val){
  if (is.na(val)){
    return("Insig")
  }
  if (val <= 0.05){
    return("Sig")
  } else {
    return("Insig")
  }},character(1))
untreated_vs_combo$significance_padj <- which_sig

untreated_vs_combo_table <- untreated_vs_combo %>% dplyr::filter(!is.na(Untreated_vs_Combo_pvalue) | !is.na(Untreated_vs_Combo_padj))
datatable(untreated_vs_combo_table)

```

## Volcano Plots

```{r}

ggplot(untreated_vs_combo,aes(x=Untreated_vs_Combo_log2FoldChange, y=-log10(Untreated_vs_Combo_pvalue), color = significance_pval))+
  geom_text(label=untreated_vs_combo$gene_symbol)
ggplot(untreated_vs_combo,aes(x=Untreated_vs_Combo_log2FoldChange, y=-log10(Untreated_vs_Combo_padj), color = significance_padj))+
  geom_text(label=untreated_vs_combo$gene_symbol)

```

## gsea results on GO and KEGG terms

```{r}

symbol_to_entrez <- as.list(org.Mm.egALIAS2EG)
untreated_vs_combo$entrez <- symbol_to_entrez[untreated_vs_combo$gene_symbol]

SYMBOLToGO <- mapIds(org.Mm.eg.db,keys=untreated_vs_combo$gene_symbol,column='GO',keytype = 'SYMBOL',multiVals = list)
GOToSYMBOL <- sapply(reverseSplit(SYMBOLToGO),unique)
GOToSYMBOL <- GOToSYMBOL[sapply(GOToSYMBOL,length)>5]

SYMBOLToKEGG <- mapIds(org.Mm.eg.db,keys=untreated_vs_combo$gene_symbol,column='PATH',keytype = 'SYMBOL',multiVals = list)
KEGGToSYMBOL <- sapply(reverseSplit(SYMBOLToKEGG),unique)
KEGGToSYMBOL <- KEGGToSYMBOL[sapply(KEGGToSYMBOL,length)>5]

diff_dat_filt <- untreated_vs_combo %>% dplyr::filter(!is.na(Untreated_vs_Combo_log2FoldChange) & !is.na(Untreated_vs_Combo_pvalue))
ranks <- diff_dat_filt$Untreated_vs_Combo_log2FoldChange*-log10(diff_dat_filt$Untreated_vs_Combo_pvalue)
names(ranks)<-diff_dat_filt$gene_symbol
ranks<-ranks[unname(!is.na(ranks))]
# ranks<-rank(ranks,ties.method = "random")
fgseaRes_GO <- fgsea(GOToSYMBOL, ranks, eps=0,scoreType="pos")
fgseaRes_GO<-cbind(sapply(c('ONTOLOGY','TERM','DEFINITION'),
                        function(x){mapIds(GO.db,keys=fgseaRes_GO$pathway,keytype='GOID',column=x)}),fgseaRes_GO)
fgseaRes_KEGG <- fgsea(KEGGToSYMBOL, ranks, eps=0,scoreType="pos")
fgseaRes_KEGG <- cbind(KEGG.Name=unlist(as.list(KEGGPATHID2NAME)[fgseaRes_KEGG$pathway]),
                   fgseaRes_KEGG)

pathways_list <- readRDS("/media/theron/My_Passport/LM_LMT_LUCIANE_cellranger/references/gene_sets/pathways_list_ensembl.rds")
fgseaRes_HALLMARK <- fgsea(pathways_list, ranks, eps=0)
fgseaRes_HALLMARK <- fgseaRes_HALLMARK %>% dplyr::filter(padj <= 0.05)

```

### GO GSEA
```{r}

GO_table <- fgseaRes_GO %>% dplyr::filter(padj <= 0.05)
GO_table<-GO_table[,c("TERM","DEFINITION","padj","NES","size")]
datatable(GO_table)

```

### KEGG GSEA
```{r}

KEGG_table <- fgseaRes_KEGG %>% dplyr::filter(padj <= 0.05)
KEGG_table<-KEGG_table[,c("KEGG.Name","padj","NES","size")]
datatable(KEGG_table)

```

### HALLMARK Pathway GSEA
```{r}
datatable(fgseaRes_HALLMARK[,c(1,3,6,7)])

```

# Untreated vs PD1 (Untreated/PD1) 

## Differential Expression Data

```{r}

which_sig <- vapply(as.numeric(untreated_vs_PD1$Untreated_vs_PD1_pvalue),function(val){
  if (is.na(val)){
    return("Insig")
  }
  if (val <= 0.05){
    return("Sig")
  } else {
    return("Insig")
  }},character(1))
untreated_vs_PD1$significance_pval <- which_sig

which_sig <- vapply(untreated_vs_PD1$Untreated_vs_PD1_padj,function(val){
  if (is.na(val)){
    return("Insig")
  }
  if (val <= 0.05){
    return("Sig")
  } else {
    return("Insig")
  }},character(1))
untreated_vs_PD1$significance_padj <- which_sig
untreated_vs_PD1_table <- untreated_vs_PD1 %>% dplyr::filter(!is.na(Untreated_vs_PD1_pvalue) | !is.na(Untreated_vs_PD1_padj))
datatable(untreated_vs_PD1_table)

```


## Volcano Plots

```{r}

ggplot(untreated_vs_PD1,aes(x=Untreated_vs_PD1_log2FoldChange, y=-log10(Untreated_vs_PD1_pvalue), color = significance_pval))+
  geom_text(label=untreated_vs_PD1$gene_symbol)
ggplot(untreated_vs_PD1,aes(x=Untreated_vs_PD1_log2FoldChange, y=-log10(Untreated_vs_PD1_padj), color = significance_padj))+
  geom_text(label=untreated_vs_PD1$gene_symbol)

```


## gsea results on GO,KEGG, and Hallmark Pathway terms

```{r}

symbol_to_entrez <- as.list(org.Mm.egALIAS2EG)
untreated_vs_PD1$entrez <- symbol_to_entrez[untreated_vs_PD1$gene_symbol]

SYMBOLToGO <- mapIds(org.Mm.eg.db,keys=untreated_vs_PD1$gene_symbol,column='GO',keytype = 'SYMBOL',multiVals = list)
GOToSYMBOL <- sapply(reverseSplit(SYMBOLToGO),unique)
GOToSYMBOL <- GOToSYMBOL[sapply(GOToSYMBOL,length)>5]

SYMBOLToKEGG <- mapIds(org.Mm.eg.db,keys=untreated_vs_PD1$gene_symbol,column='PATH',keytype = 'SYMBOL',multiVals = list)
KEGGToSYMBOL <- sapply(reverseSplit(SYMBOLToKEGG),unique)
KEGGToSYMBOL <- KEGGToSYMBOL[sapply(KEGGToSYMBOL,length)>5]

diff_dat_filt <- untreated_vs_PD1 %>% dplyr::filter(!is.na(Untreated_vs_PD1_log2FoldChange) & !is.na(Untreated_vs_PD1_pvalue))
ranks <- diff_dat_filt$Untreated_vs_PD1_log2FoldChange*-log10(diff_dat_filt$Untreated_vs_PD1_pvalue)
names(ranks)<-diff_dat_filt$gene_symbol
ranks<-ranks[unname(!is.na(ranks))]
# ranks<-rank(ranks,ties.method = "random")
fgseaRes_GO <- fgsea(GOToSYMBOL, ranks, eps=0,scoreType="pos")
fgseaRes_GO<-cbind(sapply(c('ONTOLOGY','TERM','DEFINITION'),
                        function(x){mapIds(GO.db,keys=fgseaRes_GO$pathway,keytype='GOID',column=x)}),fgseaRes_GO)
fgseaRes_KEGG <- fgsea(KEGGToSYMBOL, ranks, eps=0,scoreType="pos")
fgseaRes_KEGG <- cbind(KEGG.Name=unlist(as.list(KEGGPATHID2NAME)[fgseaRes_KEGG$pathway]),
                   fgseaRes_KEGG)

pathways_list <- readRDS("/media/theron/My_Passport/LM_LMT_LUCIANE_cellranger/references/gene_sets/pathways_list_ensembl.rds")
fgseaRes_HALLMARK <- fgsea(pathways_list, ranks, eps=0)
fgseaRes_HALLMARK <- fgseaRes_HALLMARK %>% dplyr::filter(padj <= 0.05)

```

### GO GSEA
```{r}
GO_table <- fgseaRes_GO %>% dplyr::filter(padj <= 0.05)
GO_table<-GO_table[,c("TERM","DEFINITION","padj","NES","size")]
datatable(GO_table)
```

### KEGG GSEA
```{r}
KEGG_table <- fgseaRes_KEGG %>% dplyr::filter(padj <= 0.05)
KEGG_table<-KEGG_table[,c("KEGG.Name","padj","NES","size")]
datatable(KEGG_table)
```

### HALLMARK Pathway GSEA
```{r}

datatable(fgseaRes_HALLMARK[,c(1,3,6,7)])

```

# Exploring Whole Exome mutation data- maf format

## Those mutations with differential expression per comparison

```{r}

maf_file <- "/media/theron/My_Passport/Ali_data/WES Data for 4T1MIS/MB01JHU503/MB01JHU503_000_analysis/MAF/DNA_4TIMSI_haplotypecaller.maf"
ali_maf = read.maf(maf_file)
ali_gene_summ = getGeneSummary(ali_maf)
rownames(ali_gene_summ) <- ali_gene_summ$Hugo_Symbol

rownames(PD1_vs_combo) <- PD1_vs_combo$gene_symbol
rownames(untreated_vs_combo) <- untreated_vs_combo$gene_symbol
rownames(untreated_vs_PD1) <- untreated_vs_PD1$gene_symbol

ali_gene_summ <- cbind(ali_gene_summ,PD1_vs_combo[ali_gene_summ$Hugo_Symbol,c("PD1_vs_Combo_log2FoldChange","significance_pval","significance_padj")])
ali_gene_summ <- cbind(ali_gene_summ,untreated_vs_combo[ali_gene_summ$Hugo_Symbol,c("Untreated_vs_Combo_log2FoldChange","significance_pval","significance_padj")])
ali_gene_summ <- cbind(ali_gene_summ,untreated_vs_PD1[ali_gene_summ$Hugo_Symbol,c("Untreated_vs_PD1_log2FoldChange","significance_pval","significance_padj")])

cols <- colnames(ali_gene_summ)
cols[seq(15,22)]<-c("pc_significance_pval","pc_significance_padj",
                    "Untreated_vs_Combo_log2FoldChange","uc_significance_pval","uc_significance_padj",
                    "Untreated_vs_PD1_log2FoldChange","up_significance_pval","up_significance_padj")
colnames(ali_gene_summ) <- cols
datatable(ali_gene_summ)

```

## PD1 vs Combo 
```{r}

ali_gene_summ_PD1 <- ali_gene_summ %>% dplyr::filter(pc_significance_pval=="Sig")
ggplot(ali_gene_summ_PD1,aes(y=log2(Missense_Mutation),x=PD1_vs_Combo_log2FoldChange))+
  geom_text(label=ali_gene_summ_PD1$Hugo_Symbol)+labs(title="pval Significant Genes")
datatable(ali_gene_summ_PD1)

ali_gene_summ_PD1 <- ali_gene_summ %>% dplyr::filter(pc_significance_padj=="Sig")
ggplot(ali_gene_summ_PD1,aes(y=log2(Missense_Mutation),x=PD1_vs_Combo_log2FoldChange))+
  geom_text(label=ali_gene_summ_PD1$Hugo_Symbol)+labs(title="padj Significant Genes")
datatable(ali_gene_summ_PD1)

```

## Untreated vs Combo

```{r}
ali_gene_summ_uc <- ali_gene_summ %>% dplyr::filter(uc_significance_pval=="Sig")
ggplot(ali_gene_summ_uc,aes(y=log2(Missense_Mutation),x=Untreated_vs_Combo_log2FoldChange))+
  geom_text(label=ali_gene_summ_uc$Hugo_Symbol)+labs(title="pval Significant Genes")
datatable(ali_gene_summ_uc)

ali_gene_summ_uc <- ali_gene_summ %>% dplyr::filter(uc_significance_padj=="Sig")
ggplot(ali_gene_summ_uc,aes(y=log2(Missense_Mutation),x=Untreated_vs_Combo_log2FoldChange))+
  geom_text(label=ali_gene_summ_uc$Hugo_Symbol)+labs(title="padj Significant Genes")
datatable(ali_gene_summ_uc)

```

## Untreated vs PD1
```{r}

ali_gene_summ_up <- ali_gene_summ %>% dplyr::filter(up_significance_pval=="Sig")
ggplot(ali_gene_summ_up,aes(y=log2(Missense_Mutation),x=Untreated_vs_PD1_log2FoldChange))+
  geom_text(label=ali_gene_summ_up$Hugo_Symbol)+labs(title="pval Significant Genes")
datatable(ali_gene_summ_up)

ali_gene_summ_up <- ali_gene_summ %>% dplyr::filter(up_significance_padj=="Sig")
ggplot(ali_gene_summ_up,aes(y=log2(Missense_Mutation),x=Untreated_vs_PD1_log2FoldChange))+
  geom_text(label=ali_gene_summ_up$Hugo_Symbol)+labs(title="padj Significant Genes")
datatable(ali_gene_summ_up)

```

## Gene expression of Mutated Genes Per Sample

```{r}

tpm_gene_expression_file <- "/media/theron/My_Passport/Ali_data/HTMBmice_RNAseq_Data/MB01JHU504/MB01JHU504_000_analysis/RSEM/tables/genes/genes_tpm_all_samples_norm.txt"
tpm_gene_expression <- read.table(tpm_gene_expression_file,header=T,sep="\t")

maf_file <- "/media/theron/My_Passport/Ali_data/WES Data for 4T1MIS/MB01JHU503/MB01JHU503_000_analysis/MAF/DNA_4TIMSI_haplotypecaller.maf"
ali_maf = read.maf(maf_file)
ali_gene_summ = as.data.frame(getGeneSummary(ali_maf))
rownames(ali_gene_summ) <- ali_gene_summ$Hugo_Symbol
missense_dat <- ali_gene_summ[tpm_gene_expression$gene_id,]
missense_dat <- missense_dat[complete.cases(missense_dat),]

rownames(tpm_gene_expression) <- tpm_gene_expression$gene_id
Missense_Mutation<-missense_dat$Missense_Mutation
tpm_gene_expression <- cbind(tpm_gene_expression[rownames(missense_dat),seq(2,ncol(tpm_gene_expression))],Missense_Mutation)
tpm_gene_expression <- tpm_gene_expression[complete.cases(tpm_gene_expression),]


combo<-apply(tpm_gene_expression[,seq(6)],1,mean)
pd1<-apply(tpm_gene_expression[,seq(7,12)],1,mean)
untreated<-apply(tpm_gene_expression[,seq(12,ncol(tpm_gene_expression))],1,mean)

Missense_Mutation<-tpm_gene_expression$Missense_Mutation
average_expression<-data.frame(c(combo,pd1,untreated),
                               rep(Missense_Mutation,3),
                               c(rep("combo",length(combo)),
                               rep("pd1",length(pd1)),
                               rep("untreated",length(untreated))))
colnames(average_expression)<-c("TPM_expression","Missense_Mutation","type")

average_expression_compared<-data.frame(combo,pd1,untreated,Missense_Mutation)

ggplot(average_expression,aes(x=log2(Missense_Mutation+1),y=log2(TPM_expression+1),color=type))+geom_point(shape=1)

```

# Exploring outlier splice junctions

## Filtering for significant junctions per sample

```{r}

C1_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/C1_pVals.txt"
C1_clusters <- read.table(C1_clusters_file,header=T)
C1_clusters_sig <- C1_clusters %>% dplyr::filter(C1 <= 0.05)
C1_clusters_sig$juncs <- vapply(rownames(C1_clusters_sig),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(C1_clusters)

C2_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/C2_pVals.txt"
C2_clusters <- read.table(C2_clusters_file,header=T)
C2_clusters_sig <- C2_clusters %>% dplyr::filter(C2 <= 0.05)
C2_clusters_sig$juncs <- vapply(rownames(C2_clusters_sig),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(C2_clusters)

C3_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/C3_pVals.txt"
C3_clusters <- read.table(C3_clusters_file,header=T)
C3_clusters_sig <- C3_clusters %>% dplyr::filter(C3 <= 0.05)
C3_clusters_sig$juncs <- vapply(rownames(C3_clusters_sig),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(C3_clusters)

C5_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/C5_pVals.txt"
C5_clusters <- read.table(C5_clusters_file,header=T)
C5_clusters_sig <- C5_clusters %>% dplyr::filter(C5 <= 0.05)
C5_clusters_sig$juncs <- vapply(rownames(C5_clusters_sig),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(C5_clusters)

C6_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/C6_pVals.txt"
C6_clusters <- read.table(C6_clusters_file,header=T)
C6_clusters_sig <- C6_clusters %>% dplyr::filter(C6 <= 0.05)
C6_clusters_sig$juncs <- vapply(rownames(C6_clusters_sig),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(C6_clusters)

C7_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/C7_pVals.txt"
C7_clusters <- read.table(C7_clusters_file,header=T)
C7_clusters_sig <- C7_clusters %>% dplyr::filter(C7 <= 0.05)
C7_clusters_sig$juncs <- vapply(rownames(C7_clusters_sig),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(C7_clusters)

P1_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/P1_pVals.txt"
P1_clusters <- read.table(P1_clusters_file,header=T)
P1_clusters_sig <- P1_clusters %>% dplyr::filter(P1 <= 0.05)
P1_clusters_sig$juncs <- vapply(rownames(P1_clusters_sig),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(P1_clusters)

P2_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/P2_pVals.txt"
P2_clusters <- read.table(P2_clusters_file,header=T)
P2_clusters_sig <- P2_clusters %>% dplyr::filter(P2 <= 0.05)
P2_clusters_sig$juncs <- vapply(rownames(P2_clusters_sig),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(P2_clusters)

P3_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/P3_pVals.txt"
P3_clusters <- read.table(P3_clusters_file,header=T)
P3_clusters_sig <- P3_clusters %>% dplyr::filter(P3 <= 0.05)
P3_clusters_sig$juncs <- vapply(rownames(P3_clusters_sig),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(P3_clusters)

P5_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/P5_pVals.txt"
P5_clusters <- read.table(P5_clusters_file,header=T)
P5_clusters_sig <- P5_clusters %>% dplyr::filter(P5 <= 0.05)
P5_clusters_sig$juncs <- vapply(rownames(P5_clusters_sig),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(P5_clusters)

P6_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/P6_pVals.txt"
P6_clusters <- read.table(P6_clusters_file,header=T)
P6_clusters_sig <- P6_clusters %>% dplyr::filter(P6 <= 0.05)
P6_clusters_sig$juncs <- vapply(rownames(P6_clusters_sig),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(P6_clusters)

P7_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/P7_pVals.txt"
P7_clusters <- read.table(P7_clusters_file,header=T)
P7_clusters_sig <- P7_clusters %>% dplyr::filter(P7 <= 0.05)
P7_clusters_sig$juncs <- vapply(rownames(P7_clusters_sig),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(P7_clusters)

juncs <- unique(c(C1_clusters_sig$juncs,C2_clusters_sig$juncs,
           C3_clusters_sig$juncs,C5_clusters_sig$juncs,
           C6_clusters_sig$juncs,C7_clusters_sig$juncs,
           P1_clusters_sig$juncs,P2_clusters_sig$juncs,
           P3_clusters_sig$juncs,P5_clusters_sig$juncs,
           P7_clusters_sig$juncs))

```

## Combining the outlier splicing junctions for each type together

```{r}

all_sample_pvals <- data.frame(juncs)
rownames(all_sample_pvals) <- juncs
all_sample_pvals$C1 <- NA
all_sample_pvals[C1_clusters_sig$juncs,"C1"]<-C1_clusters_sig$C1
all_sample_pvals$C2 <- NA
all_sample_pvals[C2_clusters_sig$juncs,"C2"]<-C2_clusters_sig$C2
all_sample_pvals$C3 <- NA
all_sample_pvals[C3_clusters_sig$juncs,"C3"]<-C3_clusters_sig$C3
all_sample_pvals$C5 <- NA
all_sample_pvals[C5_clusters_sig$juncs,"C5"]<-C5_clusters_sig$C5
all_sample_pvals$C6 <- NA
all_sample_pvals[C6_clusters_sig$juncs,"C6"]<-C6_clusters_sig$C6
all_sample_pvals$C7 <- NA
all_sample_pvals[C7_clusters_sig$juncs,"C7"]<-C7_clusters_sig$C7
all_sample_pvals$P1 <- NA
all_sample_pvals[P1_clusters_sig$juncs,"P1"]<-P1_clusters_sig$P1
all_sample_pvals$P2 <- NA
all_sample_pvals[P2_clusters_sig$juncs,"P2"]<-P2_clusters_sig$P2
all_sample_pvals$P3 <- NA
all_sample_pvals[P3_clusters_sig$juncs,"P3"]<-P3_clusters_sig$P3
all_sample_pvals$P5 <- NA
all_sample_pvals[P5_clusters_sig$juncs,"P5"]<-P5_clusters_sig$P5
all_sample_pvals$P7 <- NA
all_sample_pvals[P7_clusters_sig$juncs,"P7"]<-P7_clusters_sig$P7

```

## Using all significant junctions, extracting psi per leafcutter cluster


```{r}

C1_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/C1_perind.counts.gz"
C1_clusters <- read.table(C1_clusters_file,header=T)
rownames(C1_clusters) <- C1_clusters$chrom
C1_clusters_psi <- C1_clusters[rownames(C1_clusters_sig),]
C1_clusters_psi$juncs <- vapply(rownames(C1_clusters_psi),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(C1_clusters)
rownames(C1_clusters_psi) <- C1_clusters_psi

C2_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/C2_perind.counts.gz"
C2_clusters <- read.table(C2_clusters_file,header=T)
rownames(C2_clusters) <- C2_clusters$chrom
C2_clusters_psi <- C2_clusters[rownames(C2_clusters_sig),]
C2_clusters_psi$juncs <- vapply(rownames(C2_clusters_psi),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(C2_clusters)

C3_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/C3_perind.counts.gz"
C3_clusters <- read.table(C3_clusters_file,header=T)
rownames(C3_clusters) <- C3_clusters$chrom
C3_clusters_psi <- C3_clusters[rownames(C3_clusters_sig),]
C3_clusters_psi$juncs <- vapply(rownames(C3_clusters_psi),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(C3_clusters)

C5_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/C5_perind.counts.gz"
C5_clusters <- read.table(C5_clusters_file,header=T)
rownames(C5_clusters) <- C5_clusters$chrom
C5_clusters_psi <- C5_clusters[rownames(C5_clusters_sig),]
C5_clusters_psi$juncs <- vapply(rownames(C5_clusters_psi),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(C5_clusters)

C6_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/C6_perind.counts.gz"
C6_clusters <- read.table(C6_clusters_file,header=T)
rownames(C6_clusters) <- C6_clusters$chrom
C6_clusters_psi <- C6_clusters[rownames(C6_clusters_sig),]
C6_clusters_psi$juncs <- vapply(rownames(C6_clusters_psi),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(C6_clusters)

C7_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/C7_perind.counts.gz"
C7_clusters <- read.table(C7_clusters_file,header=T)
rownames(C7_clusters) <- C7_clusters$chrom
C7_clusters_psi <- C7_clusters[rownames(C7_clusters_sig),]
C7_clusters_psi$juncs <- vapply(rownames(C7_clusters_psi),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(C7_clusters)

P1_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/P1_perind.counts.gz"
P1_clusters <- read.table(P1_clusters_file,header=T)
rownames(P1_clusters) <- P1_clusters$chrom
P1_clusters_psi <- P1_clusters[rownames(P1_clusters_sig),]
P1_clusters_psi$juncs <- vapply(rownames(P1_clusters_psi),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(P1_clusters)

P2_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/P2_perind.counts.gz"
P2_clusters <- read.table(P2_clusters_file,header=T)
rownames(P2_clusters) <- P2_clusters$chrom
P2_clusters_psi <- P2_clusters[rownames(P2_clusters_sig),]
P2_clusters_psi$juncs <- vapply(rownames(P2_clusters_psi),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(P2_clusters)

P3_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/P3_perind.counts.gz"
P3_clusters <- read.table(P3_clusters_file,header=T)
rownames(P3_clusters) <- P3_clusters$chrom
P3_clusters_psi <- P3_clusters[rownames(P3_clusters_sig),]
P3_clusters_psi$juncs <- vapply(rownames(P3_clusters_psi),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(P3_clusters)

P5_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/P5_perind.counts.gz"
P5_clusters <- read.table(P5_clusters_file,header=T)
rownames(P5_clusters) <- P5_clusters$chrom
P5_clusters_psi <- P5_clusters[rownames(P5_clusters_sig),]
P5_clusters_psi$juncs <- vapply(rownames(P5_clusters_psi),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(P5_clusters)

P6_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/P6_perind.counts.gz"
P6_clusters <- read.table(P6_clusters_file,header=T)
rownames(P6_clusters) <- P6_clusters$chrom
P6_clusters_psi <- P6_clusters[rownames(P6_clusters_sig),]
P6_clusters_psi$juncs <- vapply(rownames(P6_clusters_psi),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(P6_clusters)

P7_clusters_file <- "/media/theron/My_Passport/Ali_data/analysis/P7_perind.counts.gz"
P7_clusters <- read.table(P7_clusters_file,header=T)
rownames(P7_clusters) <- P7_clusters$chrom
P7_clusters_psi <- P7_clusters[rownames(P7_clusters_sig),]
P7_clusters_psi$juncs <- vapply(rownames(P7_clusters_psi),function(val){
  paste(strsplit(val,":")[[1]][seq(3)],collapse=":")
},character(1))
rm(P7_clusters)

```
